---
title: "ST paper"
author: "Joao Carlos Gomes-Neto and Natasha Pavlovikj"
date: "1/18/2021"
output: html_document
---

Load packages 

```{r, include = FALSE}

library(tidyverse)
library(vegan)
library(skimr)
library(stringr)
library(RColorBrewer)
library(visdat)
library(assertive)
library(GGally)
library(plotly)
library(factoextra)
```

Enter the data and quality control

```{r, include = FALSE}

# enter the data
data1 <- read_csv('stringmlst_mlst_stats.csv')
# skim the data
skim(data1)
# replace NA with zeros
data2 <- data1 %>% replace(is.na(.), 0)
# transform potential extraneous characters into NA - hyphen and ?
data2 <- data2 %>%
                    mutate_all(na_if, "-") %>%
                        mutate_all(na_if, "?") 
data2 <- data2 %>% replace(is.na(.), 0)
# drop all ST equals zero
data3 <- data2 %>% filter(st != 0)
# after filter out NAs and ST numbered zero we have left 
exc <- dim(data2)[1] - dim(data3)[1]
exc
# percentage of excluded rows 
(exc/dim(data2)[1])*100
# 10.1% 
# check for NA
sum(is.na(data3))
# filter out saintpaul
data3 <- data3 %>% filter(serovar != "saintpaul")


```
Distribution of individual variables 

```{r, echo = FALSE}
# get the name of the variables
colnames(data3)
# distribution of genomes per species
d1 <- data3 %>% select(species) %>% drop_na() %>% group_by(species) %>% count()
# Fig S3-A
p1 <- ggplot(d1, aes(x = species, y = n)) + geom_col(fill = "darkblue") +
  theme_bw() + 
  theme(legend.position = "none") +
  xlab("") +
  ylab("Frequency of genomes") +
  theme(axis.text.y = element_text(size = 35, face = "italic")) +
  theme(axis.title.y = element_text(size = 40, face = "bold")) +
  theme(axis.title.x = element_text(size = 40, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 35)) +
  coord_flip() 
 p1
 
############################################################################
# distribution of genomes per serovars
d1 <- data3 %>% select(serovar) %>% drop_na() %>% group_by(serovar) %>% count()
d2 <- d1 %>% filter(serovar != "saureus" & serovar != "listeria" & serovar != "jejuni")
d3 <- d2 %>% mutate(serovar = str_to_title(serovar))
# Fig S4-A
p2 <- ggplot(d3, aes(x = serovar, y = n)) + geom_col(fill = "darkblue") +
  theme_bw() + 
  theme(legend.position = "none") +
  xlab("") +
  ylab("Frequency of genomes") +
  theme(axis.text.y = element_text(size = 35)) +
  theme(axis.title.y = element_text(size = 40, face = "bold")) +
  theme(axis.title.x = element_text(size = 40, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 35)) +
  coord_flip() 
 p2 
############################################################################
 
# distribution of genomes per species per batch

d1 <- data3 %>% select(batch, species) %>% drop_na() %>% group_by(batch, species) %>% count()
d1 <- d1 %>% mutate(batch = as.factor(batch))
# Fig S3-B
p3 <- ggplot(d1, aes(x = species, y = n, fill = batch)) + geom_col(position = "dodge") +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("") +
  ylab("Frequency of genomes") +
  theme(axis.text.y = element_text(size = 35, face = "italic")) +
  theme(axis.title.y = element_text(size = 40, face = "bold")) +
  theme(axis.title.x = element_text(size = 40, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 35)) +
  coord_flip() +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=25)) 
 p3
############################################################################
 
# distribution of genomes per serovars per batch

d1 <- data3 %>% select(batch, serovar) %>% drop_na() %>% group_by(batch, serovar) %>% count()
d2 <- d1 %>% filter(serovar != "saureus" & serovar != "listeria" & serovar != "jejuni")
d3 <- d2 %>% mutate(serovar = str_to_title(serovar))
# Fig S4-B
p4 <- ggplot(d3, aes(x = batch, y = n)) + geom_col(fill = "darkblue") +
  theme_bw() + 
  theme(legend.position = "none") +
  xlab("Batch") +
  ylab("Frequency of genomes") +
  theme(axis.text.y = element_text(size = 35)) +
  theme(axis.title.y = element_text(size = 40, face = "bold")) +
  theme(axis.title.x = element_text(size = 40, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 35)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=25)) +
  theme(strip.text.x = element_text(size = 20)) +
  facet_wrap(~serovar)
 p4
############################################################################

# distribution of genomes per species per program

d1 <- data3 %>% select(program, species) %>% drop_na() %>% group_by(program, species) %>% count()
# Fig S3-C
p5 <- ggplot(d1, aes(x = species, y = n, fill = program)) + geom_col(position = "dodge") +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("") +
  ylab("Frequency of genomes") +
  theme(axis.text.y = element_text(size = 35, face = "italic")) +
  theme(axis.title.y = element_text(size = 40, face = "bold")) +
  theme(axis.title.x = element_text(size = 40, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 35)) +
  coord_flip() +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=25)) 
 p5
############################################################################
 
# distribution of genomes per serovars per batch

d1 <- data3 %>% select(program, serovar) %>% drop_na() %>% group_by(program, serovar) %>% count()
d2 <- d1 %>% filter(serovar != "saureus" & serovar != "listeria" & serovar != "jejuni")
d3 <- d2 %>% mutate(serovar = str_to_title(serovar))
# Fig S4-C
p6 <- ggplot(d3, aes(x = program, y = n)) + geom_col(fill = "darkblue") +
  theme_bw() + 
  theme(legend.position = "none") +
  xlab("Program") +
  ylab("Frequency of genomes") +
  theme(axis.text.y = element_text(size = 35)) +
  theme(axis.title.y = element_text(size = 40, face = "bold")) +
  theme(axis.title.x = element_text(size = 40, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 35, hjust = 1)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=25)) +
  theme(strip.text.x = element_text(size = 20)) +
  facet_wrap(~serovar)
 p6
############################################################################

data4 <- data3 
data4<- mutate(data4, Program = ifelse(program == "mlst", "MLST (standard)",
                                      ifelse(program == "stringmlst" & kmer == 10, "stringMLST (kmer_10)",
                                             ifelse(program == "stringmlst" & kmer == 20, "stringMLST (kmer_20)",
                                                    ifelse(program == "stringmlst" & kmer == 30, "stringMLST (kmer_30)",
                                                           ifelse(program == "stringmlst" & kmer == 35, "stringMLST (kmer_35)",
                                                                  ifelse(program == "stringmlst" & kmer == 45, "stringMLST (kmer_45)",
                                                                         ifelse(program == "stringmlst" & kmer == 55, "stringMLST (kmer_55)",
                                                                                ifelse(program == "stringmlst" & kmer == 65, "stringMLST (kmer_65)",
                                                                                       ifelse(program == "stringmlst" & kmer == 70, "stringMLST (kmer_70)",
                                                                                              ifelse(program == "stringmlst" & kmer == 80, "stringMLST (kmer_80)",
                                                                                                     ifelse(program == "stringmlst" & kmer == 90, "stringMLST (kmer_90)", "Others"))))))))))))
############################################################################

# distribution of genomes per species per program

d1 <- data4 %>% select(Program, species) %>% drop_na() %>% group_by(Program, species) %>% count()
# Fig S3-D
p7 <- ggplot(d1, aes(x = Program, y = n)) + geom_col() +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("") +
  ylab("Frequency of genomes") +
  theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 25, hjust = 1)) +
  coord_flip() +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=25)) +
  theme(strip.text.x = element_text(size = 20, face = "italic")) +
  facet_wrap(~species)
p7

############################################################################

# distribution of genomes per serovar per program

d1 <- data4 %>% select(Program, serovar) %>% drop_na() %>% group_by(Program, serovar) %>% count()
d2 <- d1 %>% filter(serovar != "saureus" & serovar != "listeria" & serovar != "jejuni")
d3 <- d2 %>% mutate(serovar = str_to_title(serovar))
# Fig S4-D
p8 <- ggplot(d3, aes(x = Program, y = n)) + geom_col() +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("") +
  ylab("Frequency of genomes") +
  theme(axis.text.y = element_text(size = 16)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 20, hjust = 1)) +
  coord_flip() +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=25)) +
  theme(strip.text.x = element_text(size = 20)) +
  facet_wrap(~serovar)
p8

############################################################################
# distribution of st_counts in the database per species per Program

d1 <- data4 %>% select(Program, species, st_count) %>% drop_na() %>% group_by(Program, species, st_count) %>%
  summarize(st_count_total = mean(st_count))
# Fig 3-E
p9 <- ggplot(d1, aes(x = Program, y = st_count_total, fill = species)) + geom_col() +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("") +
  ylab("Total count of unique STs (database)") +
  theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 25, hjust = 1)) +
  scale_fill_manual(values = c("gray", "orange", "darkblue", "darkgreen")) +
  coord_flip() +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=25)) +
  theme(strip.text.x = element_text(size = 30, face = "italic")) +
  ggtitle("E.") +
  theme(legend.position = "none") +
  theme(plot.title = element_text(size = 45, face = "bold")) +
  facet_wrap(~species)
p9

############################################################################

# distribution of st_counts in the database per species per Program

d1 <- data4 %>% select(Program, serovar, st_count) %>% drop_na() %>% group_by(Program, serovar, st_count) %>%
  summarize(st_count_total = mean(st_count))
d2 <- d1 %>% filter(serovar != "saureus" & serovar != "listeria" & serovar != "jejuni")
d3 <- d2 %>% mutate(serovar = str_to_title(serovar))
# Fig S4-E
p10 <- ggplot(d3, aes(x = Program, y = st_count_total)) + geom_col() +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("") +
  ylab("Total count of unique STs (database)") +
  theme(axis.text.y = element_text(size = 16)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 20, hjust = 1)) +
  coord_flip() +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=25)) +
  theme(strip.text.x = element_text(size = 20)) +
  facet_wrap(~serovar)
p10
############################################################################
# distribution of total_alleles_genes in the database per species per Program

d1 <- data4 %>% select(Program, species, total_alleles_genes) %>% drop_na() %>% group_by(Program, species, total_alleles_genes) %>%
  summarize(total_alleles_genes_total = mean(total_alleles_genes))
# Fig 3-F
p11 <- ggplot(d1, aes(x = Program, y = total_alleles_genes_total, fill = species)) + geom_col() +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("") +
  ylab("Total count of unique alleles across all loci (database)") +
  theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 25, hjust = 1)) +
  scale_fill_manual(values = c("gray", "orange", "darkblue", "darkgreen")) +
  coord_flip() +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=25)) +
  theme(strip.text.x = element_text(size = 30, face = "italic")) +
  ggtitle("F.") +
  theme(legend.position = "none") +
  theme(plot.title = element_text(size = 45, face = "bold")) +
  facet_wrap(~species)
p11

############################################################################

# distribution of st_counts in the database per species per Program

d1 <- data4 %>% select(Program, serovar, total_alleles_genes) %>% drop_na() %>% group_by(Program, serovar, total_alleles_genes) %>%
  summarize(total_alleles_genes_total = mean(total_alleles_genes))
d2 <- d1 %>% filter(serovar != "saureus" & serovar != "listeria" & serovar != "jejuni")
d3 <- d2 %>% mutate(serovar = str_to_title(serovar))
# Fig S4-F
p12 <- ggplot(d3, aes(x = Program, y = total_alleles_genes_total)) + geom_col() +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("") +
  ylab("Total count of unique alleles across all loci (database)") +
  theme(axis.text.y = element_text(size = 16)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 20, hjust = 1)) +
  coord_flip() +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=25)) +
  theme(strip.text.x = element_text(size = 20)) +
  facet_wrap(~serovar)
p12
############################################################################

# distribution of number of contigs per species

p13 <- ggplot(data4, aes(x = num_contigs)) + geom_histogram(bins = 100, fill = "darkblue") +
  theme_bw() + 
  xlab("Number of contigs per genome") +
  ylab("Frequency") +
  theme(axis.text.y = element_text(size = 30)) +
  theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 30, hjust = 1)) +
  theme(strip.text.x = element_text(size = 20, face = "italic")) +
  facet_wrap(~species, scales = "free")
p13

############################################################################

# distribution of number of nucleotides per genome per species

p14 <- ggplot(data4, aes(x = total_nucl)) + geom_histogram(bins = 100, fill = "darkblue") +
  theme_bw() + 
  xlab("Total number of nucleotides per genome") +
  ylab("Frequency") +
  theme(axis.text.y = element_text(size = 30)) +
  theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 30, hjust = 1)) +
  theme(strip.text.x = element_text(size = 20, face = "italic")) +
  facet_wrap(~species, scales = "free")
p14

############################################################################

# distribution of average %GC per genome per species

p15 <- ggplot(data4, aes(x = gc_avg)) + geom_histogram(bins = 10, fill = "darkblue") +
  theme_bw() + 
  xlab("Average %GC content per genome") +
  ylab("Frequency") +
  theme(axis.text.y = element_text(size = 30)) +
  theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 30, hjust = 1)) +
  theme(strip.text.x = element_text(size = 20, face = "italic")) +
  facet_wrap(~species, scales = "free")
p15

############################################################################

# distribution of number of contigs per species
# Fig 3-A
p16 <- ggplot(data4, aes(x = species, y = num_contigs, fill = species)) + geom_boxplot() +
  theme_bw() + 
  xlab("") +
  ylab("Number of contigs per genome") +
  theme(axis.text.y = element_text(size = 30, face = "italic")) +
  theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 30, hjust = 1)) +
  theme(strip.text.x = element_text(size = 20, face = "italic")) +
  scale_fill_manual(values = c("gray", "orange", "darkblue", "darkgreen")) +
  ggtitle("A.") +
  theme(legend.position = "none") +
  theme(plot.title = element_text(size = 45, face = "bold")) +
  coord_flip()
p16

############################################################################

# distribution of number of nucleotides per genome per species
# Fig 3-B
p17 <- ggplot(data4, aes(x = species, y = total_nucl, fill = species)) + geom_boxplot() +
  theme_bw() + 
  xlab("") +
  ylab("Total number of nucleotides per genome") +
  theme(axis.text.y = element_text(size = 30, face = "italic")) +
  theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 30, hjust = 1)) +
  theme(strip.text.x = element_text(size = 20, face = "italic")) +
  scale_fill_manual(values = c("gray", "orange", "darkblue", "darkgreen")) +
  ggtitle("B.") +
  theme(legend.position = "none") +
  theme(plot.title = element_text(size = 45, face = "bold")) +
  coord_flip()
p17

############################################################################

# distribution of average %GC per genome per species
# Fig 3-C
p18 <- ggplot(data4, aes(x = species, y = gc_avg, fill = species)) + geom_boxplot() +
  theme_bw() + 
  ylim(0, 100) +
  xlab("") +
  ylab("% GC content per genome") +
  theme(axis.text.y = element_text(size = 30, face = "italic")) +
  theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 30)) +
  theme(strip.text.x = element_text(size = 20, face = "italic")) +
  scale_fill_manual(values = c("gray", "orange", "darkblue", "darkgreen")) +
  ggtitle("C.") +
  theme(legend.position = "none") +
  theme(plot.title = element_text(size = 45, face = "bold")) +
  coord_flip()
p18

############################################################################

# distribution of quantitative variables per species
# Fig S3-G
data5 <- data4 %>% select(species, num_contigs, total_nucl, gc_avg)
p19 <- ggpairs(data5, columns = 2:4, ggplot2::aes(colour=species)) 
p19
############################################################################
############################################################################

# distribution of number of contigs per serovar

d2 <- data4 %>% filter(serovar != "saureus" & serovar != "listeria" & serovar != "jejuni")
d3 <- d2 %>% mutate(serovar = str_to_title(serovar))
# Fig S4-G
p20 <- ggplot(d3, aes(x = num_contigs)) + geom_histogram(bins = 100, fill = "darkblue") +
  theme_bw() + 
  xlab("Number of contigs per genome") +
  ylab("Frequency") +
  theme(axis.text.y = element_text(size = 16)) +
  theme(axis.title.y = element_text(size = 20, face = "bold")) +
  theme(axis.title.x = element_text(size = 20, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 16, hjust = 1)) +
  theme(strip.text.x = element_text(size = 20)) +
  facet_wrap(~serovar, scales = "free")
p20

############################################################################

# distribution of number of nucleotides per genome per serovar

d2 <- data4 %>% filter(serovar != "saureus" & serovar != "listeria" & serovar != "jejuni")
d3 <- d2 %>% mutate(serovar = str_to_title(serovar))
# Fig S4-H
p21 <- ggplot(d3, aes(x = total_nucl)) + geom_histogram(bins = 100, fill = "darkblue") +
  theme_bw() + 
  xlab("Total number of nucleotides per genome") +
  ylab("Frequency") +
  theme(axis.text.y = element_text(size = 16)) +
  theme(axis.title.y = element_text(size = 20, face = "bold")) +
  theme(axis.title.x = element_text(size = 20, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 16, hjust = 1)) +
  theme(strip.text.x = element_text(size = 20)) +
  facet_wrap(~serovar, scales = "free")
p21

############################################################################

# distribution of average %GC per genome per serovar
# Fig S4-I
d2 <- data4 %>% filter(serovar != "saureus" & serovar != "listeria" & serovar != "jejuni")
d3 <- d2 %>% mutate(serovar = str_to_title(serovar))
p22 <- ggplot(d3, aes(x = gc_avg)) + geom_histogram(bins = 10, fill = "darkblue") +
  theme_bw() + 
  xlab("Average %GC content per genome") +
  ylab("Frequency") +
  theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 20, face = "bold")) +
  theme(axis.title.x = element_text(size = 20, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 25, hjust = 1)) +
  theme(strip.text.x = element_text(size = 20)) +
  facet_wrap(~serovar, scales = "free")
p22

############################################################################

# distribution of number of contigs per serovar

d2 <- data4 %>% filter(serovar != "saureus" & serovar != "listeria" & serovar != "jejuni")
d3 <- d2 %>% mutate(serovar = str_to_title(serovar))
p23 <- ggplot(d3, aes(x = serovar, y = num_contigs)) + geom_boxplot() +
  theme_bw() + 
  xlab("") +
  ylab("Number of contigs per genome") +
  theme(axis.text.y = element_text(size = 30)) +
  theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 30, hjust = 1)) +
  theme(strip.text.x = element_text(size = 20, face = "italic")) +
  coord_flip()
p23

############################################################################

# distribution of number of nucleotides per genome per serovar

d2 <- data4 %>% filter(serovar != "saureus" & serovar != "listeria" & serovar != "jejuni")
d3 <- d2 %>% mutate(serovar = str_to_title(serovar))
p24 <- ggplot(d3, aes(x = serovar, y = total_nucl)) + geom_boxplot() +
  theme_bw() + 
  xlab("") +
  ylab("Total number of nucleotides per genome") +
  theme(axis.text.y = element_text(size = 30)) +
  theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 30, hjust = 1)) +
  theme(strip.text.x = element_text(size = 20, face = "italic")) +
  coord_flip()
p24

############################################################################

# distribution of average %GC per genome per serovar

d2 <- data4 %>% filter(serovar != "saureus" & serovar != "listeria" & serovar != "jejuni")
d3 <- d2 %>% mutate(serovar = str_to_title(serovar))
p25 <- ggplot(d3, aes(x = serovar, y = gc_avg)) + geom_boxplot() +
  theme_bw() + 
  ylim(0, 100) +
  xlab("") +
  ylab("% GC content per genome") +
  theme(axis.text.y = element_text(size = 30)) +
  theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 30, hjust = 1)) +
  theme(strip.text.x = element_text(size = 20)) +
  coord_flip()
p25

############################################################################

# distribution of quantitative variables per serovar

data5 <- data4 %>% select(serovar, num_contigs, total_nucl, gc_avg)
d2 <- data5 %>% filter(serovar != "saureus" & serovar != "listeria" & serovar != "jejuni")
d3 <- d2 %>% mutate(serovar = str_to_title(serovar))
# Fig S4-M
p26 <- ggpairs(d3, columns = 2:4, ggplot2::aes(colour=serovar),
               lower = list(continuous=wrap("points", position=position_jitter(height=0.5, width=0.5)))) 
p26
############################################################################
data4 <- data3 %>% select(species, program, kmer, st)
data4 <- rename(data4, Program = program)
data4 <- mutate(data4, program = ifelse(Program == "mlst", "MLST (standard)",
                                      ifelse(Program == "stringmlst" & kmer == 10, "stringMLST (kmer_10)",
                                             ifelse(Program == "stringmlst" & kmer == 20, "stringMLST (kmer_20)",
                                                    ifelse(Program == "stringmlst" & kmer == 30, "stringMLST (kmer_30)",
                                                           ifelse(Program == "stringmlst" & kmer == 35, "stringMLST (kmer_35)",
                                                                  ifelse(Program == "stringmlst" & kmer == 45, "stringMLST (kmer_45)",
                                                                         ifelse(Program == "stringmlst" & kmer == 55, "stringMLST (kmer_55)",
                                                                                ifelse(Program == "stringmlst" & kmer == 65, "stringMLST (kmer_65)",
                                                                                       ifelse(Program == "stringmlst" & kmer == 70, "stringMLST (kmer_70)",
                                                                                              ifelse(Program == "stringmlst" & kmer == 80, "stringMLST (kmer_80)",
                                                                                                     ifelse(Program == "stringmlst" & kmer == 90, "stringMLST (kmer_90)", "Others"))))))))))))

# plot ST distribution per species by program

st_data <- data4 %>%
              drop_na() %>%
              select(species, program, st) %>%
              group_by(species, program, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              filter(species == "Campylobacter jejuni" & prop >= 2) %>%
              mutate(ST = as.factor(st))
  
# Fig S5-A  
p27a <- ggplot(st_data, aes(x = prop, y = program, fill = ST)) + xlim(0, 50) +
  theme_bw() + 
  theme() +
  xlab("Proportion (>= 2%)") +
  ylab("") +
  theme(axis.text.y = element_text(size = 22)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 25)) +
  theme(legend.title=element_text(size=30, face = "bold"))+
  theme(legend.text=element_text(size=24)) +
  ggtitle("") +
  scale_fill_manual(values=c("darkgreen", "darkblue", "gray", "red", "black", "orange", "purple", 
                             "steelblue", "coral1", "cyan", "darkgoldenrod", "darkred", 
                             "deeppink1", "bisque4", "cadetblue1", "chartreuse", "darksalmon", "darkseagreen", "deepskyblue1")) +
  theme(plot.title = element_text(size = 30, face = "bold")) + 
  theme(strip.text.x = element_text(size = 30, face = "italic")) +
  geom_col(position = "dodge") +
  facet_wrap(~ species)
p27a

st_data <- data4 %>%
              drop_na() %>%
              select(species, program, st) %>%
              group_by(species, program, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              filter(species == "Listeria monocytogenes" & prop >= 2) %>%
              mutate(ST = as.factor(st))
  
# Fig S5-B
p27b <- ggplot(st_data, aes(x = prop, y = program, fill = ST)) + xlim(0, 50) +
  theme_bw() + 
  theme() +
  xlab("Proportion (>= 2%)") +
  ylab("") +
  theme(axis.text.y = element_text(size = 22)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 25)) +
  theme(legend.title=element_text(size=30, face = "bold"))+
  theme(legend.text=element_text(size=24)) +
  ggtitle("") +
  scale_fill_manual(values=c("darkgreen", "darkblue", "gray", "red", "black", "orange", "purple", 
                             "steelblue", "coral1", "cyan", "darkgoldenrod", "darkred", 
                             "deeppink1", "bisque4", "cadetblue1", "chartreuse", "darksalmon", "darkseagreen", "deepskyblue1")) +
  theme(plot.title = element_text(size = 30, face = "bold")) + 
  theme(strip.text.x = element_text(size = 30, face = "italic")) +
  geom_col(position = "dodge") +
  facet_wrap(~ species)
p27b

st_data <- data4 %>%
              drop_na() %>%
              select(species, program, st) %>%
              group_by(species, program, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              filter(species == "Salmonella enterica" & prop >= 2) %>%
              mutate(ST = as.factor(st))
  
# Fig S5-C
p27c <- ggplot(st_data, aes(x = prop, y = program, fill = ST)) + xlim(0, 50) +
  theme_bw() + 
  theme() +
  xlab("Proportion (>= 2%)") +
  ylab("") +
  theme(axis.text.y = element_text(size = 22)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 25)) +
  theme(legend.title=element_text(size=30, face = "bold"))+
  theme(legend.text=element_text(size=24)) +
  ggtitle("") +
  scale_fill_manual(values=c("darkgreen", "darkblue", "gray", "red", "black", "orange", "purple", 
                             "steelblue", "coral1", "cyan", "darkgoldenrod", "darkred", 
                             "deeppink1", "bisque4", "cadetblue1", "chartreuse", "darksalmon", "darkseagreen", "deepskyblue1", "darkmagenta", "darkslategrey", "aquamarine3")) +
  theme(plot.title = element_text(size = 30, face = "bold")) + 
  theme(strip.text.x = element_text(size = 30, face = "italic")) +
  geom_col(position = "dodge") +
  facet_wrap(~ species)
p27c

st_data <- data4 %>%
              drop_na() %>%
              select(species, program, st) %>%
              group_by(species, program, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              filter(species == "Staphylococcus aureus" & prop >= 2) %>%
              mutate(ST = as.factor(st))
  
# Fig S5-D
p27d <- ggplot(st_data, aes(x = prop, y = program, fill = ST)) + xlim(0, 50) +
  theme_bw() + 
  theme() +
  xlab("Proportion (>= 2%)") +
  ylab("") +
  theme(axis.text.y = element_text(size = 22)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 25)) +
  theme(legend.title=element_text(size=30, face = "bold"))+
  theme(legend.text=element_text(size=24)) +
  ggtitle("") +
  scale_fill_manual(values=c("darkgreen", "darkblue", "gray", "red", "black", "orange", "purple", 
                             "steelblue", "coral1", "cyan", "darkgoldenrod", "darkred", 
                             "deeppink1", "bisque4", "cadetblue1", "chartreuse", "darksalmon", "darkseagreen", "deepskyblue1")) +
  theme(plot.title = element_text(size = 30, face = "bold")) + 
  theme(strip.text.x = element_text(size = 30, face = "italic")) +
  geom_col(position = "dodge") +
  facet_wrap(~ species)
p27d

# ST relative frequency

st_data_species <- data4 %>%
              drop_na() %>%
              select(species, program, st) %>%
              group_by(species, program, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(ST = as.factor(st))

write.csv(st_data_species, 'st_data_species.csv')
####################################################################################
data4 <- data3 %>% select(serovar, program, kmer, st)
data4 <- data4 %>% filter(serovar != "saureus" & serovar != "listeria" & serovar != "jejuni")
data4 <- rename(data4, Program = program)
data4 <- mutate(data4, program = ifelse(Program == "mlst", "MLST (standard)",
                                      ifelse(Program == "stringmlst" & kmer == 10, "stringMLST (kmer_10)",
                                             ifelse(Program == "stringmlst" & kmer == 20, "stringMLST (kmer_20)",
                                                    ifelse(Program == "stringmlst" & kmer == 30, "stringMLST (kmer_30)",
                                                           ifelse(Program == "stringmlst" & kmer == 35, "stringMLST (kmer_35)",
                                                                  ifelse(Program == "stringmlst" & kmer == 45, "stringMLST (kmer_45)",
                                                                         ifelse(Program == "stringmlst" & kmer == 55, "stringMLST (kmer_55)",
                                                                                ifelse(Program == "stringmlst" & kmer == 65, "stringMLST (kmer_65)",
                                                                                       ifelse(Program == "stringmlst" & kmer == 70, "stringMLST (kmer_70)",
                                                                                              ifelse(Program == "stringmlst" & kmer == 80, "stringMLST (kmer_80)",
                                                                                                     ifelse(Program == "stringmlst" & kmer == 90, "stringMLST (kmer_90)", "Others"))))))))))))


# plot the distribution of ST per serovar 

data4 <- data4 %>% mutate(serovar = str_to_title(serovar))
st_data <- data4 %>%
              drop_na() %>%
              select(serovar, program, st) %>%
              group_by(serovar, program, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              filter(prop >= 15) %>%
              mutate(ST = as.factor(st))
# Fig S4-N
p28 <- ggplot(st_data, aes(x = prop, y = program, fill = ST)) + xlim(0, 100) +
  theme_bw() + 
  theme() +
  xlab("Proportion (>= 15%)") +
  ylab("") +
  theme(axis.text.y = element_text(size = 14)) +
  theme(axis.title.y = element_text(size = 20, face = "bold")) +
  theme(axis.title.x = element_text(size = 20, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 16)) +
  theme(legend.title=element_text(size=25, face = "bold"))+
  theme(legend.text=element_text(size=20)) +
  ggtitle("") +
  scale_fill_manual(values=c("darkgreen", "darkblue", "gray", "red", "black", "orange", "purple", 
                             "steelblue", "coral1", "cyan", "darkgoldenrod", "darkred", 
                             "deeppink1", "bisque4", "cadetblue1", "chartreuse", "darksalmon", "darkseagreen", "deepskyblue1", "coral3", "burlywood", "cornflowerblue", "darkkhaki", "darkolivegreen", "darkturquoise")) +
  theme(plot.title = element_text(size = 30, face = "bold")) + 
  theme(strip.text.x = element_text(size = 16)) +
  geom_col(position = "dodge") +
  facet_wrap(~ serovar)
p28  

# export st data serovar

st_data_salmonella_serovar <- data4 %>%
              drop_na() %>%
              select(serovar, program, st) %>%
              group_by(serovar, program, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(ST = as.factor(st))

write.csv(st_data_salmonella_serovar, 'st_data_salmonella_serovar.csv')
```

Analysis of simpson index of diversity by species 

```{r, include = FALSE}
# removed NA and ST classified as zero

#######################################################################

# MLST program calculations for ST relative frequencies

d1 <- data3 %>%
              filter(program == "mlst") %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "MLST (standard)") 

#######################################################################

# stringmslt program kmer 10 calculations for ST relative frequencies

d2 <- data3 %>%
              filter(program == "stringmlst" & kmer == 10) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "stringMLST (kmer_10)")

#######################################################################

# stringmslt program kmer 20 calculations for ST relative frequencies

d3 <- data3 %>%
              filter(program == "stringmlst" & kmer == 20) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "stringMLST (kmer_20)")

#######################################################################

# stringmslt program kmer 30 calculations for ST relative frequencies

d4 <- data3 %>%
              filter(program == "stringmlst" & kmer == 30) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "stringMLST (kmer_30)")

#######################################################################

# stringmslt program kmer 35 calculations for ST relative frequencies

d5 <- data3 %>%
              filter(program == "stringmlst" & kmer == 35) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "stringMLST (kmer_35)")

#######################################################################

# stringmslt program kmer 45 calculations for ST relative frequencies

d6 <- data3 %>%
              filter(program == "stringmlst" & kmer == 45) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "stringMLST (kmer_45)") 

#######################################################################

# stringmslt program kmer 55 calculations for ST relative frequencies

d7 <- data3 %>%
              filter(program == "stringmlst" & kmer == 55) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "stringMLST (kmer_55)")

#######################################################################

# stringmslt program kmer 65 calculations for ST relative frequencies

d8 <- data3 %>%
              filter(program == "stringmlst" & kmer == 65) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "stringMLST (kmer_65)")

#######################################################################

# stringmslt program kmer 70 calculations for ST relative frequencies

d9 <- data3 %>%
              filter(program == "stringmlst" & kmer == 70) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "stringMLST (kmer_70)")

#######################################################################

# stringmslt program kmer 80 calculations for ST relative frequencies

d10 <- data3 %>%
              filter(program == "stringmlst" & kmer == 80) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "stringMLST (kmer_80)")

#######################################################################

# stringmslt program kmer 90 calculations for ST relative frequencies

d11 <- data3 %>%
              filter(program == "stringmlst" & kmer == 90) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "stringMLST (kmer_90)")

#######################################################################

# concatenate datasets

d12b <- rbind(d1, d2, d3, d4, d5, d6, d7, d8, d9, d10, d11)

# check for NAs

skim(d12b)

########################################################################
# Plot the number of unique STs 
# Fig 4-B
p1 <- ggplot(d12b, aes(x = program, y = simpson, fill = species)) + ylim(0, 1) +
  theme_bw() + 
  theme(legend.position = "none") +
  xlab("") +
  ylab("Index value") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("B. Simpson's D index of diversity using ST counts across species") +
  scale_fill_manual(name = "Bacterial species", values = c("gray", "orange", "darkblue", "darkgreen"),
                    labels = c(expression(italic("Campylobacter jejuni")), 
                               expression(italic("Listeria monocytogenes")),
                               expression(italic("Salmonella enterica")),
                               expression(italic("Staphylococcus aureus")))) +
  theme(plot.title = element_text(size = 26, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24)) +
  geom_boxplot() + coord_flip() 
p1

###############################################################
# run permanova for species and program only

sum(is.na(d12b))
vis_miss(d12b)
assert_all_are_not_na(d12b)
set.seed(1233)
mod1 <- adonis(simpson ~ species*program, data=d12b, permutations=1000)
mod1 

###############################################################

# merge results with original data3 to run the statistical model

#data5 <- data3 %>% select(sra_id, species, batch, num_contigs, total_nucl, gc_avg, st_count, total_alleles_genes)

#data6 <- left_join(data5, d12b, by = c("species" = "species", "batch" = "batch"))

#data7 <- distinct(data6)

# richness_species dataset

#simpson_species <- data7 %>% drop_na()

# check the distribution of specific variables 

#sim1 <- simpson_species %>% select(!c(sra_id, batch))
#dis <- sim1 %>%
#                group_by(species) %>%
#                summarize(across(
#    .cols = is.numeric, 
#    .fns = list(Mean = mean, Median = median, SD = sd), na.rm = TRUE, 
#    .names = "{col}_{fn}"
#    ))

# write data out

#write.csv(simpson_species, '~/Documents/stringMLST_and_MLST_output_files/simpson_species.csv')

###############################################################
# run permanova model using mean for covariates

# get the mean and sd for each metric and species 

data5 <- data3 %>% select(species, batch, program, kmer, num_contigs, total_nucl, gc_avg, st_count, total_alleles_genes)
data5 <- rename(data5, Program = program)
data5<- mutate(data5, program = ifelse(Program == "mlst", "MLST (standard)",
                                      ifelse(Program == "stringmlst" & kmer == 10, "stringMLST (kmer_10)",
                                             ifelse(Program == "stringmlst" & kmer == 20, "stringMLST (kmer_20)",
                                                    ifelse(Program == "stringmlst" & kmer == 30, "stringMLST (kmer_30)",
                                                           ifelse(Program == "stringmlst" & kmer == 35, "stringMLST (kmer_35)",
                                                                  ifelse(Program == "stringmlst" & kmer == 45, "stringMLST (kmer_45)",
                                                                         ifelse(Program == "stringmlst" & kmer == 55, "stringMLST (kmer_55)",
                                                                                ifelse(Program == "stringmlst" & kmer == 65, "stringMLST (kmer_65)",
                                                                                       ifelse(Program == "stringmlst" & kmer == 70, "stringMLST (kmer_70)",
                                                                                              ifelse(Program == "stringmlst" & kmer == 80, "stringMLST (kmer_80)",
                                                                                                     ifelse(Program == "stringmlst" & kmer == 90, "stringMLST (kmer_90)", "Others"))))))))))))

# get the mean for num_contigs, total_nucl, gc_avg
par1 <- data5 %>% 
               group_by(species, batch) %>%
               summarize(num_contigs_median = median(num_contigs), 
                         total_nucl_mean = mean(total_nucl),
                         gc_avg_mean = mean(gc_avg))

# get the mean for st_count, total_alleles_genes
par2 <- data5 %>% 
               group_by(species, batch, program) %>%
               summarize(st_count_mean = mean(st_count), 
                         total_alleles_genes_mean = mean(total_alleles_genes))


# merge par2 and par1

par3 <- left_join(par2, par1, on = c("species" = "species", "batch" = "batch", "program"= "program"))

# merge with simpson index data

d13 <- left_join(d12b, par3, on = c("species" = "species", "batch" = "batch", "program"= "program"))

############################################################################################
# Running all models 

# Model with species and program
set.seed(1233)
# Fig S7-A
mod1 <- adonis(simpson ~ species * program, data=d13, permutations=1000)
mod1 
species_program_Simpson <- data.frame(mod1$aov.tab)
write.csv(species_program_Simpson, 'species_program_Simpson.csv')

# model for species only

set.seed(1233)
# Fig S7-B
mod2 <- adonis(simpson ~ species, data=d13, permutations=1000)
mod2 
speciesSimpson<- data.frame(mod2$aov.tab)
write.csv(speciesSimpson, 'speciesSimpson.csv')

# model for program only

set.seed(1233)
# Fig S7-C
mod3 <- adonis(simpson ~ program, data=d13, permutations=1000)
mod3 
programSimpson<- data.frame(mod3$aov.tab)
write.csv(programSimpson, 'programSimpson.csv')

# model for st_count_mean only

set.seed(1233)
# Fig S7-G
mod4 <- adonis(simpson ~ st_count_mean, data=d13, permutations=1000)
mod4 
stcountmeanSimpson <- data.frame(mod4$aov.tab)
write.csv(stcountmeanSimpson, 'stcountmeanSimpson.csv')

# model for total_alleles_genes_mean only

set.seed(1233)
# Fig S7-H
mod5 <- adonis(simpson ~ total_alleles_genes_mean, data=d13, permutations=1000)
mod5 
totalallelesgenesmeanSimpson <- data.frame(mod5$aov.tab)
write.csv(totalallelesgenesmeanSimpson, 'totalallelesgenesmeanSimpson.csv')

# model for num_contigs_median only

set.seed(1233)
# Fig S7-D
mod6 <- adonis(simpson ~ num_contigs_median, data=d13, permutations=1000)
mod6 
numcontigsSimpson <- data.frame(mod6$aov.tab)
write.csv(numcontigsSimpson, 'numcontigsSimpson.csv')

# model for total_nucl_mean only

set.seed(1233)
# Fig S7-E
mod7 <- adonis(simpson ~ total_nucl_mean, data=d13, permutations=1000)
mod7 
total_nucl_meanSimpson <- data.frame(mod7$aov.tab)
write.csv(total_nucl_meanSimpson, 'total_nucl_meanSimpson.csv')

# model for gc_avg_mean only

set.seed(1233)
# Fig S7-F
mod8 <- adonis(simpson ~ gc_avg_mean, data=d13, permutations=1000)
mod8 
gcavgSimpson <- data.frame(mod8$aov.tab)
write.csv(gcavgSimpson, 'gcavgSimpson.csv')

###############################################################
###############################################################
###############################################################

# now let's run the same model using the sd 

# get the sd for num_contigs, total_nucl, gc_avg
par4 <- data5 %>% 
               group_by(species) %>%
               summarize(num_contigs_sd = sd(num_contigs), 
                         total_nucl_sd = sd(total_nucl),
                         gc_avg_sd = sd(gc_avg))

# merge with simpson index data 

d14 <- left_join(d12b, par4, on = c("species" = "species"))

# model using number of contigs only per species

set.seed(1233)
# Fig S7-I
m1 <- adonis(simpson ~ num_contigs_sd, data=d14, permutations=1000)
m1
numcontigsSDsimp <- data.frame(m1$aov.tab)
write.csv(numcontigsSDsimp, 'numcontigsSDsimp.csv')

# model using total number of nucleotides per genome only per species

set.seed(1233)
# Fig S7-J
m2 <- adonis(simpson ~ total_nucl_sd, data=d14, permutations=1000)
m2
totalnuclSDsimp <- data.frame(m2$aov.tab)
write.csv(totalnuclSDsimp, 'totalnuclSDsimp.csv')

# model using the average GC content per genome per species

set.seed(1233)
# Fig S7-K
m3 <- adonis(simpson ~ gc_avg_sd, data=d14, permutations=1000)
m3
gcavgSDsimp <- data.frame(m3$aov.tab)
write.csv(gcavgSDsimp, 'gcavgSDsimp.csv')
```

Total number of unique STs across species

```{r, include = FALSE}

# before starting this analysis I have removed all rows with missing values for STs and also ST classified as zero

#######################################################################

# MLST program calculations for ST relative frequencies

d1 <- data3 %>%
              filter(program == "mlst") %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "MLST (standard)")

#######################################################################

# stringmslt program kmer 10 calculations for ST relative frequencies

d2 <- data3 %>%
              filter(program == "stringmlst" & kmer == 10) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_10)")

#######################################################################

# stringmslt program kmer 20 calculations for ST relative frequencies

d3 <- data3 %>%
              filter(program == "stringmlst" & kmer == 20) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_20)")

#######################################################################

# stringmslt program kmer 30 calculations for ST relative frequencies

d4 <- data3 %>%
              filter(program == "stringmlst" & kmer == 30) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_30)")

#######################################################################

# stringmslt program kmer 35 calculations for ST relative frequencies

d5 <- data3 %>%
              filter(program == "stringmlst" & kmer == 35) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_35)")

#######################################################################

# stringmslt program kmer 45 calculations for ST relative frequencies

d6 <- data3 %>%
              filter(program == "stringmlst" & kmer == 45) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_45)")

#######################################################################

# stringmslt program kmer 55 calculations for ST relative frequencies

d7 <- data3 %>%
              filter(program == "stringmlst" & kmer == 55) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_55)")

#######################################################################

# stringmslt program kmer 65 calculations for ST relative frequencies

d8 <- data3 %>%
              filter(program == "stringmlst" & kmer == 65) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_65)")

#######################################################################

# stringmslt program kmer 70 calculations for ST relative frequencies

d9 <- data3 %>%
              filter(program == "stringmlst" & kmer == 70) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_70)")

#######################################################################

# stringmslt program kmer 80 calculations for ST relative frequencies

d10 <- data3 %>%
              filter(program == "stringmlst" & kmer == 80) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_80)")

#######################################################################

# stringmslt program kmer 90 calculations for ST relative frequencies

d11 <- data3 %>%
              filter(program == "stringmlst" & kmer == 90) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_90)")

#######################################################################

# concatenate datasets

d12 <- rbind(d1, d2, d3, d4, d5, d6, d7, d8, d9, d10, d11)

# check for NAs

skim(d12)

########################################################################
# Plot the number of unique STs 
# Fig 4-A
p1 <- ggplot(d12, aes(x = program, y = n, fill = species)) + ylim(0, 150) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("") +
  ylab("Richness") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("A. Total number of unique STs across species") +
  scale_fill_manual(name = "Bacterial species", values = c("gray", "orange", "darkblue", "darkgreen"),
                    labels = c(expression(italic("Campylobacter jejuni")), 
                               expression(italic("Listeria monocytogenes")),
                               expression(italic("Salmonella enterica")),
                               expression(italic("Staphylococcus aureus")))) +
  theme(plot.title = element_text(size = 30, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24)) +
  geom_boxplot() + coord_flip() 
p1

# merge d12 and d12b to get simpson index as well
data4 <- left_join(d12, d12b, by = c("species" = "species", "program" = "program", "batch" = "batch"))

# merge results with original data3 to run the statistical model

data5 <- data3 %>% select(sra_id, species, batch, num_contigs, total_nucl, gc_avg, st_count, total_alleles_genes)

data6 <- left_join(data5, data4, by = c("species" = "species", "batch" = "batch"))

data7 <- distinct(data6)

# richness_species dataset

rich_species <- data7 %>% drop_na()

# write data out

write.csv(rich_species, 'rich_species.csv')

###############################################################
# run permanova model using mean for covariates

# get the mean and sd for each metric and species 

data5 <- data3 %>% select(species, batch, program, kmer, num_contigs, total_nucl, gc_avg, st_count, total_alleles_genes)
data5 <- rename(data5, Program = program)
data5<- mutate(data5, program = ifelse(Program == "mlst", "MLST (standard)",
                                      ifelse(Program == "stringmlst" & kmer == 10, "stringMLST (kmer_10)",
                                             ifelse(Program == "stringmlst" & kmer == 20, "stringMLST (kmer_20)",
                                                    ifelse(Program == "stringmlst" & kmer == 30, "stringMLST (kmer_30)",
                                                           ifelse(Program == "stringmlst" & kmer == 35, "stringMLST (kmer_35)",
                                                                  ifelse(Program == "stringmlst" & kmer == 45, "stringMLST (kmer_45)",
                                                                         ifelse(Program == "stringmlst" & kmer == 55, "stringMLST (kmer_55)",
                                                                                ifelse(Program == "stringmlst" & kmer == 65, "stringMLST (kmer_65)",
                                                                                       ifelse(Program == "stringmlst" & kmer == 70, "stringMLST (kmer_70)",
                                                                                              ifelse(Program == "stringmlst" & kmer == 80, "stringMLST (kmer_80)",
                                                                                                     ifelse(Program == "stringmlst" & kmer == 90, "stringMLST (kmer_90)", "Others"))))))))))))

# get the mean for num_contigs, total_nucl, gc_avg
par1 <- data5 %>% 
               group_by(species, batch) %>%
               summarize(num_contigs_median = median(num_contigs), 
                         total_nucl_mean = mean(total_nucl),
                         gc_avg_mean = mean(gc_avg))

# get the mean for st_count, total_alleles_genes
par2 <- data5 %>% 
               group_by(species, batch, program) %>%
               summarize(st_count_mean = mean(st_count), 
                         total_alleles_genes_mean = mean(total_alleles_genes))


# merge par2 and par1

par3 <- left_join(par2, par1, on = c("species" = "species", "batch" = "batch", "program"= "program"))

# merge with simpson index data

d13 <- left_join(d12, par3, on = c("species" = "species", "batch" = "batch", "program"= "program"))
d14 <- left_join(d13, d12b, on = c("species" = "species", "batch" = "batch", "program"= "program"))

############################################################################################
# Running all models 

# Model with species and program
set.seed(1233)
# Fig S6-A
mod1 <- adonis(n ~ species * program, data=d14, permutations=1000)
mod1 
species_program_STRichness <- data.frame(mod1$aov.tab)
write.csv(species_program_STRichness, 'species_program_STRichness.csv')

# model for species only

set.seed(1233)
# Fig S6-B
mod2 <- adonis(n ~ species, data=d14, permutations=1000)
mod2 
speciesSTRichness <- data.frame(mod2$aov.tab)
write.csv(speciesSTRichness, 'speciesSTRichness.csv')

# model for program only

set.seed(1233)
# Fig S6-C
mod3 <- adonis(n ~ program, data=d14, permutations=1000)
mod3 
programSTRichness <- data.frame(mod3$aov.tab)
write.csv(programSTRichness, 'programSTRichness.csv')

# model for st_count_mean only

set.seed(1233)
# Fig S6-G
mod4 <- adonis(n ~ st_count_mean, data=d14, permutations=1000)
mod4 
stcountmeanSTRichness <- data.frame(mod4$aov.tab)
write.csv(stcountmeanSTRichness, 'stcountmeanSTRichness.csv')

# model for total_alleles_genes_mean only

set.seed(1233)
# Fig S6-H
mod5 <- adonis(n ~ total_alleles_genes_mean, data=d14, permutations=1000)
mod5 
totalallelesgenesmeanSTRichness <- data.frame(mod5$aov.tab)
write.csv(totalallelesgenesmeanSTRichness, 'totalallelesgenesmeanSTRichness.csv')

# model for num_contigs_median only

set.seed(1233)
# Fig S6-D
mod6 <- adonis(n ~ num_contigs_median, data=d14, permutations=1000)
mod6 
numcontigsSTRichness <- data.frame(mod6$aov.tab)
write.csv(numcontigsSTRichness, 'numcontigsSTRichness.csv')

# model for total_nucl_mean only

set.seed(1233)
# Fig S6-E
mod7 <- adonis(n ~ total_nucl_mean, data=d14, permutations=1000)
mod7 
total_nucl_meanSTRichness <- data.frame(mod7$aov.tab)
write.csv(total_nucl_meanSTRichness, 'total_nucl_meanSTRichness.csv')

# model for gc_avg_mean only

set.seed(1233)
# Fig S6-F
mod8 <- adonis(n ~ gc_avg_mean, data=d14, permutations=1000)
mod8 
gcavgSTRichness <- data.frame(mod8$aov.tab)
write.csv(gcavgSTRichness, 'gcavgSTRichness.csv')

# model for the simpson index only

set.seed(1233)
# Fig S6-I
mod8 <- adonis(n ~ simpson, data=d14, permutations=1000)
mod8 
SimpsonSTRichness <- data.frame(mod8$aov.tab)
write.csv(SimpsonSTRichness, 'SimpsonSTRichness.csv')


###############################################################
###############################################################
###############################################################
###############################################################

# now let's run the same model using the sd 

# get the sd for num_contigs, total_nucl, gc_avg
par4 <- data5 %>% 
               group_by(species) %>%
               summarize(num_contigs_sd = sd(num_contigs), 
                         total_nucl_sd = sd(total_nucl),
                         gc_avg_sd = sd(gc_avg))

# merge with simpson index data

d15 <- left_join(d12, par4, on = c("species" = "species"))
d16 <- left_join(d15, d12b, on = c("species" = "species", "program" = "program", "batch" = "batch"))

# model using number of contigs only per species

set.seed(1233)
# Fig S10-J
# Fig S6-J
m1 <- adonis(n ~ num_contigs_sd, data=d16, permutations=1000)
m1
numcontigsSDstrichness <- data.frame(m1$aov.tab)
write.csv(numcontigsSDstrichness, 'numcontigsSDstrichness.csv')

# model using total number of nucleotides per genome only per species

set.seed(1233)
# Fig S10-K
# Fig S6-K
m2 <- adonis(n ~ total_nucl_sd, data=d16, permutations=1000)
m2
totalnuclSDstrichness <- data.frame(m2$aov.tab)
write.csv(totalnuclSDstrichness, 'totalnuclSDstrichness.csv')

# model using the average GC content per genome per species

set.seed(1233)
# Fig S10-L
# Fig S6-L
m3 <- adonis(n ~ gc_avg_sd, data=d16, permutations=1000)
m3
gcavgSDstrichness <- data.frame(m3$aov.tab)
write.csv(gcavgSDstrichness, 'gcavgSDstrichness.csv')
```

Analysis of blank calls (not classified) across species 

```{r, include = FALSE}
data2b <- data2 %>% filter(serovar != "saintpaul")
# now here we care about using missing values 
#######################################################################

# MLST program calculations for ST relative frequencies

d1 <- data2b %>%
              filter(program == "mlst") %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "MLST (standard)")

#######################################################################

# stringmslt program kmer 10 calculations for ST relative frequencies

d2 <- data2b %>%
              filter(program == "stringmlst" & kmer == 10) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_10)")

#######################################################################

# stringmslt program kmer 20 calculations for ST relative frequencies

d3 <- data2b %>%
              filter(program == "stringmlst" & kmer == 20) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_20)")

#######################################################################

# stringmslt program kmer 30 calculations for ST relative frequencies

d4 <- data2b %>%
              filter(program == "stringmlst" & kmer == 30) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_30)")

#######################################################################

# stringmslt program kmer 35 calculations for ST relative frequencies

d5 <- data2b %>%
              filter(program == "stringmlst" & kmer == 35) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_35)")

#######################################################################

# stringmslt program kmer 45 calculations for ST relative frequencies

d6 <- data2b %>%
              filter(program == "stringmlst" & kmer == 45) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_45)")

#######################################################################

# stringmslt program kmer 55 calculations for ST relative frequencies

d7 <- data2b %>%
              filter(program == "stringmlst" & kmer == 55) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_55)")

#######################################################################

# stringmslt program kmer 65 calculations for ST relative frequencies

d8 <- data2b %>%
              filter(program == "stringmlst" & kmer == 65) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_65)")

#######################################################################

# stringmslt program kmer 70 calculations for ST relative frequencies

d9 <- data2b %>%
              filter(program == "stringmlst" & kmer == 70) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_70)")

#######################################################################

# stringmslt program kmer 80 calculations for ST relative frequencies

d10 <- data2b %>%
              filter(program == "stringmlst" & kmer == 80) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_80)")

#######################################################################

# stringmslt program kmer 90 calculations for ST relative frequencies

d11 <- data2b %>%
              filter(program == "stringmlst" & kmer == 90) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_90)")

#######################################################################

# concatenate datasets

d12c <- rbind(d1, d2, d3, d4, d5, d6, d7, d8, d9, d10, d11)

# check for NAs

skim(d12c)

########################################################################
# Plot the proportion of non-classified STs

# change from NA to non-classified

d13 <- d12c

# Set ST equal zero to not classified 

d13 <- mutate(d13, ST = ifelse(st %in% 0, "Not classified", "correct"))

# filter not classified

d14 <- d13 %>% filter(ST == "Not classified") 
d15 <- d14 %>% select(species, batch, prop, program)

# export data for analysis


data5 <- data3 %>% select(sra_id, species, batch, num_contigs, total_nucl, gc_avg, st_count, total_alleles_genes)

d17<- left_join(data5, d15, by = c("species" = "species", "batch" = "batch"))

d18 <- distinct(d17)

# merge with simpson index data

d19 <- left_join(d18, d12b, by = c("species" = "species", "batch" = "batch", "program" = "program"))

d20 <- distinct(d19)

# richness_species dataset

prop_species <- d20 %>% drop_na()

# write data out

write.csv(prop_species, 'prop_species.csv')


# Plot the proportion of not classified genomes across programs 
# Fig 4-C
p1 <- ggplot(d15, aes(x = program, y = prop, fill = species)) + ylim(0, 100) +
  theme_bw() + 
  theme(legend.position = "none") +
  xlab("") +
  ylab("Proportion") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("C. Proportion of non-classified STs across species") +
  scale_fill_manual(name = "Bacterial species", values = c("gray", "orange", "darkblue", "darkgreen"),
                    labels = c(expression(italic("Campylobacter jejuni")), 
                               expression(italic("Listeria monocytogenes")),
                               expression(italic("Salmonella enterica")),
                               expression(italic("Staphylococcus aureus")))) +
  theme(plot.title = element_text(size = 25, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24)) +
  geom_boxplot() + coord_flip() 
p1

###############################################################
# run permanova model using mean for covariates

# get the mean and sd for each metric and species 

data5 <- data3 %>% select(species, batch, program, kmer, num_contigs, total_nucl, gc_avg, st_count, total_alleles_genes)
data5 <- rename(data5, Program = program)
data5<- mutate(data5, program = ifelse(Program == "mlst", "MLST (standard)",
                                      ifelse(Program == "stringmlst" & kmer == 10, "stringMLST (kmer_10)",
                                             ifelse(Program == "stringmlst" & kmer == 20, "stringMLST (kmer_20)",
                                                    ifelse(Program == "stringmlst" & kmer == 30, "stringMLST (kmer_30)",
                                                           ifelse(Program == "stringmlst" & kmer == 35, "stringMLST (kmer_35)",
                                                                  ifelse(Program == "stringmlst" & kmer == 45, "stringMLST (kmer_45)",
                                                                         ifelse(Program == "stringmlst" & kmer == 55, "stringMLST (kmer_55)",
                                                                                ifelse(Program == "stringmlst" & kmer == 65, "stringMLST (kmer_65)",
                                                                                       ifelse(Program == "stringmlst" & kmer == 70, "stringMLST (kmer_70)",
                                                                                              ifelse(Program == "stringmlst" & kmer == 80, "stringMLST (kmer_80)",
                                                                                                     ifelse(Program == "stringmlst" & kmer == 90, "stringMLST (kmer_90)", "Others"))))))))))))

# get the mean for num_contigs, total_nucl, gc_avg
par1 <- data5 %>% 
               group_by(species, batch) %>%
               summarize(num_contigs_median = median(num_contigs), 
                         total_nucl_mean = mean(total_nucl),
                         gc_avg_mean = mean(gc_avg))

# get the mean for st_count, total_alleles_genes
par2 <- data5 %>% 
               group_by(species, batch, program) %>%
               summarize(st_count_mean = mean(st_count), 
                         total_alleles_genes_mean = mean(total_alleles_genes))


# merge par2 and par1

par3 <- left_join(par2, par1, on = c("species" = "species", "batch" = "batch", "program"= "program"))

# merge with simpson index data

d16 <- left_join(d15, d12b, on = c("species" = "species", "batch" = "batch", "program"= "program"))
d17 <- left_join(d16, par3, on = c("species" = "species", "batch" = "batch", "program"= "program"))
d18 <- d17 %>% drop_na()

############################################################################################
# Running all models 

# Model with species and program
set.seed(1233)
# Fig S8-A
mod1 <- adonis(prop ~ species * program, data=d18, permutations=1000)
mod1 
species_program_prop <- data.frame(mod1$aov.tab)
write.csv(species_program_prop, 'species_program_prop.csv')

# model for species only

set.seed(1233)
# Fig S8-B
mod2 <- adonis(prop ~ species, data=d18, permutations=1000)
mod2 
speciesprop <- data.frame(mod2$aov.tab)
write.csv(speciesprop, 'speciesprop.csv')

# model for program only

set.seed(1233)
# Fig S8-C
mod3 <- adonis(prop ~ program, data=d18, permutations=1000)
mod3 
programprop <- data.frame(mod3$aov.tab)
write.csv(programprop, 'programprop.csv')

# model for st_count_mean only

set.seed(1233)
# Fig S8-G
mod4 <- adonis(prop ~ st_count_mean, data=d18, permutations=1000)
mod4 
stcountmeanprop <- data.frame(mod4$aov.tab)
write.csv(stcountmeanprop, 'stcountmeanprop.csv')

# model for total_alleles_genes_mean only

set.seed(1233)
# Fig S8-H
mod5 <- adonis(prop ~ total_alleles_genes_mean, data=d18, permutations=1000)
mod5 
totalallelesgenesmeanprop <- data.frame(mod5$aov.tab)
write.csv(totalallelesgenesmeanprop, 'totalallelesgenesmeanprop.csv')

# model for num_contigs_median only

set.seed(1233)
# Fig S8-D
mod6 <- adonis(prop ~ num_contigs_median, data=d18, permutations=1000)
mod6 
numcontigsprop <- data.frame(mod6$aov.tab)
write.csv(numcontigsprop, 'numcontigsprop.csv')

# model for total_nucl_mean only

set.seed(1233)
# Fig S8-E
mod7 <- adonis(prop ~ total_nucl_mean, data=d18, permutations=1000)
mod7 
total_nucl_meanprop <- data.frame(mod7$aov.tab)
write.csv(total_nucl_meanprop, 'total_nucl_meanprop.csv')

# model for gc_avg_mean only

set.seed(1233)
# Fig S8-F
mod8 <- adonis(prop ~ gc_avg_mean, data=d18, permutations=1000)
mod8 
gcavgprop <- data.frame(mod8$aov.tab)
write.csv(gcavgprop, 'gcavgprop.csv')

# model for the simpson index only

set.seed(1233)
# Fig S8-I
mod8 <- adonis(prop ~ simpson, data=d18, permutations=1000)
mod8 
Simpsonprop <- data.frame(mod8$aov.tab)
write.csv(Simpsonprop, 'Simpsonprop.csv')


###############################################################
###############################################################
###############################################################
###############################################################

###############################################################

# now let's run the same model using the sd 

# get the sd for num_contigs, total_nucl, gc_avg
par4 <- data5 %>% 
               group_by(species) %>%
               summarize(num_contigs_sd = sd(num_contigs), 
                         total_nucl_sd = sd(total_nucl),
                         gc_avg_sd = sd(gc_avg))

# merge with simpson index data

d16 <- left_join(d15, d12b, on = c("species" = "species", "program"= "program", "batch" = "batch"))
d17 <- left_join(d16, par4, on = c("species" = "species"))
d18b <- d17 %>% drop_na()

# model using number of contigs only per species

set.seed(1233)
# Fig S8-J
m1 <- adonis(prop ~ num_contigs_sd, data=d18b, permutations=1000)
m1
numcontigsSDprop <- data.frame(m1$aov.tab)
write.csv(numcontigsSDprop, 'numcontigsSDprop.csv')

# model using total number of nucleotides per genome only per species

set.seed(1233)
# Fig S8-K
m2 <- adonis(prop ~ total_nucl_sd, data=d18b, permutations=1000)
m2
totalnuclSDprop <- data.frame(m2$aov.tab)
write.csv(totalnuclSDprop, 'totalnuclSDprop.csv')

# model using the average GC content per genome per species

set.seed(1233)
# Fig S8-L
m3 <- adonis(prop ~ gc_avg_sd, data=d18b, permutations=1000)
m3
gcavgSDprop <- data.frame(m3$aov.tab)
write.csv(gcavgSDprop, 'gcavgSDprop.csv')

######################################################################

# calculate the standard deviation for non-classification

d16 <- d15 %>%
         group_by(species, program) %>%
         summarise(std = sd(prop))
# Fig 4-D
p2 <- ggplot(d16, aes(x = program, y = std, fill = species)) + ylim(0, 5) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("") +
  ylab("SD") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("D. Standard deviation for the proportion of non-classified STs across species") +
  scale_fill_manual(name = "Bacterial species", values = c("gray", "orange", "darkblue", "darkgreen"),
                    labels = c(expression(italic("Campylobacter jejuni")), 
                               expression(italic("Listeria monocytogenes")),
                               expression(italic("Salmonella enterica")),
                               expression(italic("Staphylococcus aureus")))) +
  theme(plot.title = element_text(size = 25, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24)) +
  geom_bar(stat="identity", position=position_dodge()) + coord_flip() 
p2
```

3D plot unique STs, simpson index, by program and species

```{r, include = FALSE}
# to calculate the unique CLASSIFIED STs, we have to remove the missing values because we are counting
# genomes that were successfully classified only. Pay attention to everything I am writing here because it 
# will help you when writing the paper

# if you always want to check if there are NAs run either of the following functions

# sum(is.na(data))
# skim(data)

#######################################################################

# MLST program calculations for ST relative frequencies

d1 <- data3 %>%
              filter(program == "mlst") %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "MLST (standard)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 10 calculations for ST relative frequencies

d2 <- data3 %>%
              filter(program == "stringmlst" & kmer == 10) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_10)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 20 calculations for ST relative frequencies

d3 <- data3 %>%
              filter(program == "stringmlst" & kmer == 20) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_20)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 30 calculations for ST relative frequencies

d4 <- data3 %>%
              filter(program == "stringmlst" & kmer == 30) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_30)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 35 calculations for ST relative frequencies

d5 <- data3 %>%
              filter(program == "stringmlst" & kmer == 35) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_35)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 45 calculations for ST relative frequencies

d6 <- data3 %>%
              filter(program == "stringmlst" & kmer == 45) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_45)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 55 calculations for ST relative frequencies

d7 <- data3 %>%
              filter(program == "stringmlst" & kmer == 55) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_55)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 65 calculations for ST relative frequencies

d8 <- data3 %>%
              filter(program == "stringmlst" & kmer == 65) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_65)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 70 calculations for ST relative frequencies

d9 <- data3 %>%
              filter(program == "stringmlst" & kmer == 70) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_70)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 80 calculations for ST relative frequencies

d10 <- data3 %>%
              filter(program == "stringmlst" & kmer == 80) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_80)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 90 calculations for ST relative frequencies

d11 <- data3 %>%
              filter(program == "stringmlst" & kmer == 90) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_90)") %>%
              mutate(metric = "Richness")

#######################################################################

# concatenate datasets

d12 <- rbind(d1, d2, d3, d4, d5, d6, d7, d8, d9, d10, d11)

# check for NAs

skim(d12)

########################################################################
#######################################################################

# MLST program calculations for ST relative frequencies

d1b <- data3 %>%
              filter(program == "mlst") %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "MLST (standard)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 10 calculations for ST relative frequencies

d2b <- data3 %>%
              filter(program == "stringmlst" & kmer == 10) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_10)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 20 calculations for ST relative frequencies

d3b <- data3 %>%
              filter(program == "stringmlst" & kmer == 20) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_20)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 30 calculations for ST relative frequencies

d4b <- data3 %>%
              filter(program == "stringmlst" & kmer == 30) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_30)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 35 calculations for ST relative frequencies

d5b <- data3 %>%
              filter(program == "stringmlst" & kmer == 35) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_35)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 45 calculations for ST relative frequencies

d6b <- data3 %>%
              filter(program == "stringmlst" & kmer == 45) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_45)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 55 calculations for ST relative frequencies

d7b <- data3 %>%
              filter(program == "stringmlst" & kmer == 55) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_55)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 65 calculations for ST relative frequencies

d8b <- data3 %>%
              filter(program == "stringmlst" & kmer == 65) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_65)") %>%
              mutate(metric = "Simpson")
              
#######################################################################

# stringmslt program kmer 70 calculations for ST relative frequencies

d9b <- data3 %>%
              filter(program == "stringmlst" & kmer == 70) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_70)") %>%
              mutate(metric = "Simpson")
              
#######################################################################

# stringmslt program kmer 80 calculations for ST relative frequencies

d10b <- data3 %>%
              filter(program == "stringmlst" & kmer == 80) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_80)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 90 calculations for ST relative frequencies

d11b <- data3 %>%
              filter(program == "stringmlst" & kmer == 90) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_90)") %>%
              mutate(metric = "Simpson")

#######################################################################

# concatenate datasets

d12b <- rbind(d1b, d2b, d3b, d4b, d5b, d6b, d7b, d8b, d9b, d10b, d11b)

# check for NAs

skim(d12b)

#######################################################################
#######################################################################
#######################################################################

data2b <- data2 %>% filter(serovar != "saintpaul")
# now here we care about using missing values 
#######################################################################

# MLST program calculations for ST relative frequencies

d1 <- data2b %>%
              filter(program == "mlst") %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "MLST (standard)")

#######################################################################

# stringmslt program kmer 10 calculations for ST relative frequencies

d2 <- data2b %>%
              filter(program == "stringmlst" & kmer == 10) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_10)")

#######################################################################

# stringmslt program kmer 20 calculations for ST relative frequencies

d3 <- data2b %>%
              filter(program == "stringmlst" & kmer == 20) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_20)")

#######################################################################

# stringmslt program kmer 30 calculations for ST relative frequencies

d4 <- data2b %>%
              filter(program == "stringmlst" & kmer == 30) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_30)")

#######################################################################

# stringmslt program kmer 35 calculations for ST relative frequencies

d5 <- data2b %>%
              filter(program == "stringmlst" & kmer == 35) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_35)")

#######################################################################

# stringmslt program kmer 45 calculations for ST relative frequencies

d6 <- data2b %>%
              filter(program == "stringmlst" & kmer == 45) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_45)")

#######################################################################

# stringmslt program kmer 55 calculations for ST relative frequencies

d7 <- data2b %>%
              filter(program == "stringmlst" & kmer == 55) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_55)")

#######################################################################

# stringmslt program kmer 65 calculations for ST relative frequencies

d8 <- data2b %>%
              filter(program == "stringmlst" & kmer == 65) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_65)")

#######################################################################

# stringmslt program kmer 70 calculations for ST relative frequencies

d9 <- data2b %>%
              filter(program == "stringmlst" & kmer == 70) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_70)")

#######################################################################

# stringmslt program kmer 80 calculations for ST relative frequencies

d10 <- data2b %>%
              filter(program == "stringmlst" & kmer == 80) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_80)")

#######################################################################

# stringmslt program kmer 90 calculations for ST relative frequencies

d11 <- data2b %>%
              filter(program == "stringmlst" & kmer == 90) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_90)")

#######################################################################

# concatenate datasets

d12c <- rbind(d1, d2, d3, d4, d5, d6, d7, d8, d9, d10, d11)

# check for NAs

skim(d12c)

# change from NA to non-classified

d13 <- d12c

# Set ST equal zero to not classified 

d13 <- mutate(d13, ST = ifelse(st %in% 0, "Not classified", "correct"))

# filter not classified

d14 <- d13 %>% filter(ST == "Not classified") 
d15 <- d14 %>% select(species, batch, prop, program)
d15 <- rename(d15, index = prop)
d15 <- d15 %>% mutate(metric = "Miscalls")
#######################################################################
#######################################################################
#######################################################################
# concatenate datasets 

d16 <- bind_rows(d12, d12b, d15)

# get the mean for each species

#d17 <- d16 %>% group_by(species, program, metric) %>% summarise(index = mean(index))

# spread the data

d18 <- d16 %>% spread(metric, index) %>% drop_na()

# plot
# Fig 5-B
p2 <- ggplot(d18, aes(x = Richness, y = Simpson, color = program, shape = species)) + ylim(0, 1.05) + xlim(0, 150) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Richness") +
  ylab("Simpson's index of diversity") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("B.") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 45, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24, face = "italic")) +
  geom_jitter(size = 5, width = 0.05, height = 0.05) +
  labs(shape = "Bacterial species") 
p2

# Fig 5-C
p3 <- ggplot(d18, aes(x = Miscalls, y = Simpson, color = program, shape = species)) + ylim(0, 1.05) + xlim(0, 20) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Proportion of non-classidied STs") +
  ylab("Simpson's index of diversity") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("C.") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 45, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24, face = "italic")) +
  geom_jitter(size = 5, width = 0.05, height = 0.05) +
  labs(shape = "Bacterial species") 
p3
# Fig 5-D
p4 <- ggplot(d18, aes(x = Miscalls, y = Richness, color = program, shape = species)) + ylim(0, 150) + xlim(0, 20) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Proportion of non-classidied STs") +
  ylab("Richness") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("D.") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 45, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24, face = "italic")) +
  geom_jitter(size = 5, width = 0.05, height = 0.05) +
  labs(shape = "Bacterial species") 
p4

# plot A
# Fig 5-A
p1 <- plot_ly(d18, x = ~Miscalls, y = ~Richness, z = ~Simpson, color = ~species, colors = "Set1")
p1 <- p1 %>% add_markers()
p1 <- p1 %>% layout(scene = list(xaxis = list(title = 'Proportion of non-classified STs'),
                     yaxis = list(title = 'ST richness'),
                     zaxis = list(title = 'Simpson index')))

p1
```

3D biplot across species accounting for total_nucl_mean, gc_avg_mean, num_contigs_median

```{r, include = FALSE}
#######################################################################

# MLST program calculations for ST relative frequencies

d1 <- data3 %>%
              filter(program == "mlst") %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "MLST (standard)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 10 calculations for ST relative frequencies

d2 <- data3 %>%
              filter(program == "stringmlst" & kmer == 10) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_10)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 20 calculations for ST relative frequencies

d3 <- data3 %>%
              filter(program == "stringmlst" & kmer == 20) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_20)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 30 calculations for ST relative frequencies

d4 <- data3 %>%
              filter(program == "stringmlst" & kmer == 30) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_30)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 35 calculations for ST relative frequencies

d5 <- data3 %>%
              filter(program == "stringmlst" & kmer == 35) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_35)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 45 calculations for ST relative frequencies

d6 <- data3 %>%
              filter(program == "stringmlst" & kmer == 45) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_45)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 55 calculations for ST relative frequencies

d7 <- data3 %>%
              filter(program == "stringmlst" & kmer == 55) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_55)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 65 calculations for ST relative frequencies

d8 <- data3 %>%
              filter(program == "stringmlst" & kmer == 65) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_65)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 70 calculations for ST relative frequencies

d9 <- data3 %>%
              filter(program == "stringmlst" & kmer == 70) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_70)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 80 calculations for ST relative frequencies

d10 <- data3 %>%
              filter(program == "stringmlst" & kmer == 80) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_80)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 90 calculations for ST relative frequencies

d11 <- data3 %>%
              filter(program == "stringmlst" & kmer == 90) %>%
              drop_na() %>%
              group_by(species, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_90)") %>%
              mutate(metric = "Richness")

#######################################################################

# concatenate datasets

d12 <- rbind(d1, d2, d3, d4, d5, d6, d7, d8, d9, d10, d11)

# check for NAs

skim(d12)

########################################################################
#######################################################################

# MLST program calculations for ST relative frequencies

d1b <- data3 %>%
              filter(program == "mlst") %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "MLST (standard)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 10 calculations for ST relative frequencies

d2b <- data3 %>%
              filter(program == "stringmlst" & kmer == 10) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_10)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 20 calculations for ST relative frequencies

d3b <- data3 %>%
              filter(program == "stringmlst" & kmer == 20) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_20)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 30 calculations for ST relative frequencies

d4b <- data3 %>%
              filter(program == "stringmlst" & kmer == 30) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_30)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 35 calculations for ST relative frequencies

d5b <- data3 %>%
              filter(program == "stringmlst" & kmer == 35) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_35)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 45 calculations for ST relative frequencies

d6b <- data3 %>%
              filter(program == "stringmlst" & kmer == 45) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_45)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 55 calculations for ST relative frequencies

d7b <- data3 %>%
              filter(program == "stringmlst" & kmer == 55) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_55)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 65 calculations for ST relative frequencies

d8b <- data3 %>%
              filter(program == "stringmlst" & kmer == 65) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_65)") %>%
              mutate(metric = "Simpson")
              
#######################################################################

# stringmslt program kmer 70 calculations for ST relative frequencies

d9b <- data3 %>%
              filter(program == "stringmlst" & kmer == 70) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_70)") %>%
              mutate(metric = "Simpson")
              
#######################################################################

# stringmslt program kmer 80 calculations for ST relative frequencies

d10b <- data3 %>%
              filter(program == "stringmlst" & kmer == 80) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_80)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 90 calculations for ST relative frequencies

d11b <- data3 %>%
              filter(program == "stringmlst" & kmer == 90) %>%
              drop_na() %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(species, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_90)") %>%
              mutate(metric = "Simpson")

#######################################################################

# concatenate datasets

d12b <- rbind(d1b, d2b, d3b, d4b, d5b, d6b, d7b, d8b, d9b, d10b, d11b)

# check for NAs

skim(d12b)

#######################################################################
#######################################################################
#######################################################################

data2b <- data2 %>% filter(serovar != "saintpaul")
# now here we care about using missing values 
#######################################################################

# MLST program calculations for ST relative frequencies

d1 <- data2b %>%
              filter(program == "mlst") %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "MLST (standard)")

#######################################################################

# stringmslt program kmer 10 calculations for ST relative frequencies

d2 <- data2b %>%
              filter(program == "stringmlst" & kmer == 10) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_10)")

#######################################################################

# stringmslt program kmer 20 calculations for ST relative frequencies

d3 <- data2b %>%
              filter(program == "stringmlst" & kmer == 20) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_20)")

#######################################################################

# stringmslt program kmer 30 calculations for ST relative frequencies

d4 <- data2b %>%
              filter(program == "stringmlst" & kmer == 30) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_30)")

#######################################################################

# stringmslt program kmer 35 calculations for ST relative frequencies

d5 <- data2b %>%
              filter(program == "stringmlst" & kmer == 35) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_35)")

#######################################################################

# stringmslt program kmer 45 calculations for ST relative frequencies

d6 <- data2b %>%
              filter(program == "stringmlst" & kmer == 45) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_45)")

#######################################################################

# stringmslt program kmer 55 calculations for ST relative frequencies

d7 <- data2b %>%
              filter(program == "stringmlst" & kmer == 55) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_55)")

#######################################################################

# stringmslt program kmer 65 calculations for ST relative frequencies

d8 <- data2b %>%
              filter(program == "stringmlst" & kmer == 65) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_65)")

#######################################################################

# stringmslt program kmer 70 calculations for ST relative frequencies

d9 <- data2b %>%
              filter(program == "stringmlst" & kmer == 70) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_70)")

#######################################################################

# stringmslt program kmer 80 calculations for ST relative frequencies

d10 <- data2b %>%
              filter(program == "stringmlst" & kmer == 80) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_80)")

#######################################################################

# stringmslt program kmer 90 calculations for ST relative frequencies

d11 <- data2b %>%
              filter(program == "stringmlst" & kmer == 90) %>%
              group_by(species, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_90)")

#######################################################################

# concatenate datasets

d12c <- rbind(d1, d2, d3, d4, d5, d6, d7, d8, d9, d10, d11)

# check for NAs

skim(d12c)

# change from NA to non-classified

d13 <- d12c

# Set ST equal zero to not classified 

d13 <- mutate(d13, ST = ifelse(st %in% 0, "Not classified", "correct"))

# filter not classified

d14 <- d13 %>% filter(ST == "Not classified") 
d15 <- d14 %>% select(species, batch, prop, program)
d15 <- rename(d15, index = prop)
d15 <- d15 %>% mutate(metric = "Miscalls")
#######################################################################
#######################################################################
# concatenate datasets 

d16 <- bind_rows(d12, d12b, d15)
###############################################################
# run permanova model using mean for covariates

# get the mean and sd for each metric and species 

data5 <- data3 %>% select(species, batch, program, kmer, num_contigs, total_nucl, gc_avg, st_count, total_alleles_genes)
data5 <- rename(data5, Program = program)
data5<- mutate(data5, program = ifelse(Program == "mlst", "MLST (standard)",
                                      ifelse(Program == "stringmlst" & kmer == 10, "stringMLST (kmer_10)",
                                             ifelse(Program == "stringmlst" & kmer == 20, "stringMLST (kmer_20)",
                                                    ifelse(Program == "stringmlst" & kmer == 30, "stringMLST (kmer_30)",
                                                           ifelse(Program == "stringmlst" & kmer == 35, "stringMLST (kmer_35)",
                                                                  ifelse(Program == "stringmlst" & kmer == 45, "stringMLST (kmer_45)",
                                                                         ifelse(Program == "stringmlst" & kmer == 55, "stringMLST (kmer_55)",
                                                                                ifelse(Program == "stringmlst" & kmer == 65, "stringMLST (kmer_65)",
                                                                                       ifelse(Program == "stringmlst" & kmer == 70, "stringMLST (kmer_70)",
                                                                                              ifelse(Program == "stringmlst" & kmer == 80, "stringMLST (kmer_80)",
                                                                                                     ifelse(Program == "stringmlst" & kmer == 90, "stringMLST (kmer_90)", "Others"))))))))))))

# get the mean for num_contigs, total_nucl, gc_avg
par1 <- data5 %>% 
               group_by(species, batch) %>%
               summarize(num_contigs_median = median(num_contigs), 
                         total_nucl_mean = mean(total_nucl),
                         gc_avg_mean = mean(gc_avg))

# get the mean for st_count, total_alleles_genes
par2 <- data5 %>% 
               group_by(species, batch, program) %>%
               summarize(st_count_mean = mean(st_count), 
                         total_alleles_genes_mean = mean(total_alleles_genes))


# merge par2 and par1

par3 <- left_join(par2, par1, c("species" = "species", "batch" = "batch"))

# merge with simpson index data

d17 <- left_join(d16, par3, on = c("species" = "species", "batch" = "batch", "program"= "program"))
############################################################################################
#######################################################################

# spread the data

d18 <- d17 %>% spread(metric, index) %>% drop_na()

############################################################################################
#######################################################################

# plots Richness

# p1 
# Fig S9-A
p1 <- ggplot(d18, aes(x = num_contigs_median, y = Richness, color = program, shape = species)) + ylim(0, 150) + xlim(0, 300) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Median number of contigs per species") +
  ylab("Richness") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("A.") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 45, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24, face = "italic")) +
  geom_jitter(size = 5, width = 0.05, height = 0.05) +
  labs(shape = "Bacterial species") 
p1

# p2
# Fig S9-B
p2 <- ggplot(d18, aes(x = total_nucl_mean, y = Richness, color = program, shape = species)) + ylim(0, 150) + xlim(0, 5000000) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Average of total number of nucleotides per species") +
  ylab("Richness") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("B.") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 45, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24, face = "italic")) +
  geom_jitter(size = 5, width = 0.05, height = 0.05) +
  labs(shape = "Bacterial species") 
p2


# p3
# Fig S9-C
p3 <- ggplot(d18, aes(x = gc_avg_mean, y = Richness, color = program, shape = species)) + ylim(0, 150) + xlim(0, 60) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Average %GC content per genome per species") +
  ylab("Richness") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("C.") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 45, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24, face = "italic")) +
  geom_jitter(size = 5, width = 0.05, height = 0.05) +
  labs(shape = "Bacterial species") 
p3

#############################################################################

# plots Simposn

# p1 
# Fig S9-D
p1 <- ggplot(d18, aes(x = num_contigs_median, y = Simpson, color = program, shape = species)) + ylim(0, 1.05) + xlim(0, 300) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Median number of contigs per species") +
  ylab("Simpson's D index of diversity") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("A.") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 45, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24, face = "italic")) +
  geom_jitter(size = 5, width = 0.05, height = 0.05) +
  labs(shape = "Bacterial species") 
p1

# p2
# Fig S9-E
p2 <- ggplot(d18, aes(x = total_nucl_mean, y = Simpson, color = program, shape = species)) + ylim(0, 1.05) + xlim(0, 5000000) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Average of total number of nucleotides per species") +
  ylab("Simpson's D index of diversity") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("B.") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 45, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24, face = "italic")) +
  geom_jitter(size = 5, width = 0.05, height = 0.05) +
  labs(shape = "Bacterial species") 
p2


# p3
# Fig S9-F
p3 <- ggplot(d18, aes(x = gc_avg_mean, y = Simpson, color = program, shape = species)) + ylim(0, 1.05) + xlim(0, 60) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Average %GC content per genome per species") +
  ylab("Simpson's D index of diversity") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("C.") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 45, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24, face = "italic")) +
  geom_jitter(size = 5, width = 0.05, height = 0.05) +
  labs(shape = "Bacterial species") 
p3

#############################################################################

# plots Miscalls

# p1 
# Fig S9-G
p1 <- ggplot(d18, aes(x = num_contigs_median, y = Miscalls, color = program, shape = species)) + ylim(0, 100) + xlim(0, 300) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Median number of contigs per species") +
  ylab("Proportion of non-classified STs") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("A.") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 45, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24, face = "italic")) +
  geom_jitter(size = 5, width = 0.05, height = 0.05) +
  labs(shape = "Bacterial species") 
p1

# p2
# Fig S9-H
p2 <- ggplot(d18, aes(x = total_nucl_mean, y = Miscalls, color = program, shape = species)) + ylim(0, 100) + xlim(0, 5000000) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Average of total number of nucleotides per species") +
  ylab("Proportion of non-classified STs") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("B.") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 45, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24, face = "italic")) +
  geom_jitter(size = 5, width = 0.05, height = 0.05) +
  labs(shape = "Bacterial species") 
p2


# p3
# Fig S9-I
p3 <- ggplot(d18, aes(x = gc_avg_mean, y = Miscalls, color = program, shape = species)) + ylim(0, 100) + xlim(0, 60) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Average %GC content per genome per species") +
  ylab("Proportion of non-classified STs") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("C.") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 45, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24, face = "italic")) +
  geom_jitter(size = 5, width = 0.05, height = 0.05) +
  labs(shape = "Bacterial species") 
p3

#############################################################################
#############################################################################
#############################################################################

# plots St_count

# p1 
# Fig S9-J
p1 <- ggplot(d18, aes(x = st_count_mean, y = Richness, color = program, shape = species)) + ylim(0, 150) + xlim(0, 15000) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Total number of unique STs per species \n & database generated across programs") +
  ylab("ST richness") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("A.") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 45, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24, face = "italic")) +
  geom_jitter(size = 5, width = 0.05, height = 0.05) +
  labs(shape = "Bacterial species") 
p1

# p2
# Fig S9-K
p2 <- ggplot(d18, aes(x = st_count_mean, y = Simpson, color = program, shape = species)) + ylim(0, 1.05) + xlim(0, 15000) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Total number of unique STs per species \n & database generated across programs") +
  ylab("Simpson's D index of diveristy") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("B.") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 45, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24, face = "italic")) +
  geom_jitter(size = 5, width = 0.05, height = 0.05) +
  labs(shape = "Bacterial species") 
p2


# p3
# Fig S9-L
p3 <- ggplot(d18, aes(x = st_count_mean, y = Miscalls, color = program, shape = species)) + ylim(0, 20) + xlim(0, 15000) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Total number of unique STs per species \n & database generated across programs") +
  ylab("Proportion of non-classified STs") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("C.") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 45, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24, face = "italic")) +
  geom_jitter(size = 5, width = 0.05, height = 0.05) +
  labs(shape = "Bacterial species") 
p3

#############################################################################

# plots total number of allles per databaser generated

# p1 
# Fig S9-M
p1b <- ggplot(d18, aes(x = total_alleles_genes_mean, y = Richness, color = program, shape = species)) + ylim(0, 150) + xlim(0, 10000) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Total number of unique alleles per species \n & database generated across programs") +
  ylab("ST richness") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("A.") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 45, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24, face = "italic")) +
  geom_jitter(size = 5, width = 0.05, height = 0.05) +
  labs(shape = "Bacterial species") 
p1b

# p2
# Fig S9-N
p2b <- ggplot(d18, aes(x = total_alleles_genes_mean, y = Simpson, color = program, shape = species)) + ylim(0, 1.05) + xlim(0, 10000) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Total number of unique alleles per species \n & database generated across programs") +
  ylab("Simpson's D index of diveristy") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("B.") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 45, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24, face = "italic")) +
  geom_jitter(size = 5, width = 0.05, height = 0.05) +
  labs(shape = "Bacterial species") 
p2b


# p3
# Fig S9-O
p3b <- ggplot(d18, aes(x = total_alleles_genes_mean, y = Miscalls, color = program, shape = species)) + ylim(0, 20) + xlim(0, 10000) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Total number of unique alleles per species \n & database generated across programs") +
  ylab("Proportion of non-classified STs") +
theme(axis.text.y = element_text(size = 25)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 25)) +
  ggtitle("C.") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 45, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24, face = "italic")) +
  geom_jitter(size = 5, width = 0.05, height = 0.05) +
  labs(shape = "Bacterial species") 
p3b
```

Analysis of simpson index of diversity by serovar within S. enterica

```{r, include = FALSE}
# remove st with missing counts or classified as zero

#######################################################################

# MLST program calculations for ST relative frequencies

d1 <- data3 %>%
              filter(program == "mlst" & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "MLST (standard)") 

#######################################################################

# stringmslt program kmer 10 calculations for ST relative frequencies

d2 <- data3 %>%
              filter(program == "stringmlst" & kmer == 10 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "stringMLST (kmer_10)")

#######################################################################

# stringmslt program kmer 20 calculations for ST relative frequencies

d3 <- data3 %>%
              filter(program == "stringmlst" & kmer == 20 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "stringMLST (kmer_20)")

#######################################################################

# stringmslt program kmer 30 calculations for ST relative frequencies

d4 <- data3 %>%
              filter(program == "stringmlst" & kmer == 30 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "stringMLST (kmer_30)")

#######################################################################

# stringmslt program kmer 35 calculations for ST relative frequencies

d5 <- data3 %>%
              filter(program == "stringmlst" & kmer == 35 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "stringMLST (kmer_35)")

#######################################################################

# stringmslt program kmer 45 calculations for ST relative frequencies

d6 <- data3 %>%
              filter(program == "stringmlst" & kmer == 45 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "stringMLST (kmer_45)") 

#######################################################################

# stringmslt program kmer 55 calculations for ST relative frequencies

d7 <- data3 %>%
              filter(program == "stringmlst" & kmer == 55 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "stringMLST (kmer_55)")

#######################################################################

# stringmslt program kmer 65 calculations for ST relative frequencies

d8 <- data3 %>%
              filter(program == "stringmlst" & kmer == 65 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "stringMLST (kmer_65)")

#######################################################################

# stringmslt program kmer 70 calculations for ST relative frequencies

d9 <- data3 %>%
              filter(program == "stringmlst" & kmer == 70 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "stringMLST (kmer_70)")

#######################################################################

# stringmslt program kmer 80 calculations for ST relative frequencies

d10 <- data3 %>%
              filter(program == "stringmlst" & kmer == 80 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "stringMLST (kmer_80)")

#######################################################################

# stringmslt program kmer 90 calculations for ST relative frequencies

d11 <- data3 %>%
              filter(program == "stringmlst" & kmer == 90 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(simpson = mean(simpson)) %>%
              mutate(program = "stringMLST (kmer_90)")

#######################################################################

# concatenate datasets

d12 <- rbind(d1, d2, d3, d4, d5, d6, d7, d8, d9, d10, d11)

# check for NAs

skim(d12)

# transform serovar names

d13 <- d12 %>% mutate(Serovars = str_to_title(serovar))

########################################################################
# Plot the number of unique STs 
# Fig S4-P
p1 <- ggplot(d13, aes(x = program, y = simpson, fill = Serovars)) + ylim(0, 1) +
  theme_bw() + 
  theme(legend.position = "none") +
  xlab("") +
  ylab("Index value") +
theme(axis.text.y = element_text(size = 13)) +
  theme(axis.title.y = element_text(size = 20, face = "bold")) +
  theme(axis.title.x = element_text(size = 20, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 12)) +
  ggtitle("Simpson's D index of diversity using ST counts across serovars") +
  scale_fill_manual(values = c("gray", "red", "orange", "darkgreen", "darkblue", 
                               "yellow", "purple", "brown", "darkolivegreen2", "cadetblue1",
                               "cyan4", "darksalmon", "pink", "coral1", "azure4",
                               "bisque2", "cyan", "darkolivegreen3", "darkorchid2", "darkorange",
                               "cyan3")) +
  theme(plot.title = element_text(size = 20, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=20, face = "bold")) +
  theme(legend.text=element_text(size=15)) +
  geom_boxplot() + coord_flip() + facet_wrap(~Serovars)
p1

# merge with data3 

data5 <- data3 %>% select(sra_id, serovar, batch, num_contigs, total_nucl, gc_avg, st_count, total_alleles_genes)

data6 <- data5 %>% filter(serovar != "jejuni" & serovar != "saureus" & serovar != "listeria")

d15<- left_join(data6, d12, by = c("serovar" = "serovar", "batch" = "batch"))

d16 <- distinct(d15)

# richness_species dataset

simpson_serovar <- d16 %>% drop_na()

simpson_serovar <- simpson_serovar %>% filter(serovar != "infantis")

# write data out

write.csv(simpson_serovar, 'simpson_serovar.csv')

###############################################################
# run permanova model using mean for covariates

# get the mean and sd for each metric and species 

data5 <- data3 %>% select(serovar, batch, program, kmer, num_contigs, total_nucl, gc_avg, st_count, total_alleles_genes)
data5 <- rename(data5, Program = program)
data5<- mutate(data5, program = ifelse(Program == "mlst", "MLST (standard)",
                                      ifelse(Program == "stringmlst" & kmer == 10, "stringMLST (kmer_10)",
                                             ifelse(Program == "stringmlst" & kmer == 20, "stringMLST (kmer_20)",
                                                    ifelse(Program == "stringmlst" & kmer == 30, "stringMLST (kmer_30)",
                                                           ifelse(Program == "stringmlst" & kmer == 35, "stringMLST (kmer_35)",
                                                                  ifelse(Program == "stringmlst" & kmer == 45, "stringMLST (kmer_45)",
                                                                         ifelse(Program == "stringmlst" & kmer == 55, "stringMLST (kmer_55)",
                                                                                ifelse(Program == "stringmlst" & kmer == 65, "stringMLST (kmer_65)",
                                                                                       ifelse(Program == "stringmlst" & kmer == 70, "stringMLST (kmer_70)",
                                                                                              ifelse(Program == "stringmlst" & kmer == 80, "stringMLST (kmer_80)",
                                                                                                     ifelse(Program == "stringmlst" & kmer == 90, "stringMLST (kmer_90)", "Others"))))))))))))


data6 <- data5 %>% filter(serovar != "jejuni" & serovar != "saureus" & serovar != "listeria")

# get the mean for num_contigs, total_nucl, gc_avg
par1 <- data6 %>% 
               group_by(serovar, batch) %>%
               summarize(num_contigs_median = median(num_contigs), 
                         total_nucl_mean = mean(total_nucl),
                         gc_avg_mean = mean(gc_avg))

# get the mean for st_count, total_alleles_genes
par2 <- data6 %>% 
               group_by(serovar, batch, program) %>%
               summarize(st_count_mean = mean(st_count), 
                         total_alleles_genes_mean = mean(total_alleles_genes))


# merge par2 and par1

par3 <- left_join(par2, par1, on = c("serovar" = "serovar", "batch" = "batch"))

# merge with simpson index data

d13 <- left_join(d12, par3, on = c("serovar" = "serovar", "batch" = "batch", "program"= "program"))

# filter out infantis
d14 <- d13 %>% filter(serovar != "infantis" & serovar != "schwarzengrund")

############################################################################################
# Running all models 

# Model with species and program
set.seed(1233)
# Fig S11-A
mod1 <- adonis(simpson ~ serovar * program, data=d14, permutations=1000)
mod1 
serovar_program_Simpson <- data.frame(mod1$aov.tab)
write.csv(serovar_program_Simpson, 'serovar_program_Simpson.csv')

# model for species only

set.seed(1233)
# Fig S11-B
mod2 <- adonis(simpson ~ serovar, data=d14, permutations=1000)
mod2 
serovarSimpson<- data.frame(mod2$aov.tab)
write.csv(serovarSimpson, 'serovarSimpson.csv')

# model for program only

set.seed(1233)
# Fig S11-C
mod3 <- adonis(simpson ~ program, data=d14, permutations=1000)
mod3 
serovarprogramSimpson<- data.frame(mod3$aov.tab)
write.csv(serovarprogramSimpson, 'serovarprogramSimpson.csv')

# model for st_count_mean only

set.seed(1233)
# Fig S11-G
mod4 <- adonis(simpson ~ st_count_mean, data=d14, permutations=1000)
mod4 
stcountmeanSimpsonSerovar <- data.frame(mod4$aov.tab)
write.csv(stcountmeanSimpsonSerovar, 'stcountmeanSimpsonSerovar.csv')

# model for total_alleles_genes_mean only

set.seed(1233)
# Fig S11-H
mod5 <- adonis(simpson ~ total_alleles_genes_mean, data=d14, permutations=1000)
mod5 
totalallelesgenesmeanSimpsonSerovar <- data.frame(mod5$aov.tab)
write.csv(totalallelesgenesmeanSimpsonSerovar, 'totalallelesgenesmeanSimpsonSerovar.csv')

# model for num_contigs_median only

set.seed(1233)
# Fig S11-D
mod6 <- adonis(simpson ~ num_contigs_median, data=d14, permutations=1000)
mod6 
numcontigsSimpsonSerovar <- data.frame(mod6$aov.tab)
write.csv(numcontigsSimpsonSerovar, 'numcontigsSimpsonSerovar.csv')

# model for total_nucl_mean only

set.seed(1233)
# Fig S11-E
mod7 <- adonis(simpson ~ total_nucl_mean, data=d14, permutations=1000)
mod7 
total_nucl_meanSimpsonSerovar <- data.frame(mod7$aov.tab)
write.csv(total_nucl_meanSimpsonSerovar, 'total_nucl_meanSimpsonSerovar.csv')

# model for gc_avg_mean only

set.seed(1233)
# Fig S11-F
mod8 <- adonis(simpson ~ gc_avg_mean, data=d14, permutations=1000)
mod8 
gcavgSimpsonSerovar <- data.frame(mod8$aov.tab)
write.csv(gcavgSimpsonSerovar, 'gcavgSimpsonSerovar.csv')

###############################################################
###############################################################
###############################################################

# now let's run the same model using the sd 

# get the sd for num_contigs, total_nucl, gc_avg
par4 <- data6 %>% 
               group_by(serovar) %>%
               summarize(num_contigs_sd = sd(num_contigs), 
                         total_nucl_sd = sd(total_nucl),
                         gc_avg_sd = sd(gc_avg))

# merge with simpson index data

d15 <- left_join(d12, par4, on = c("serovar" = "serovar"))

# filter out infantis
d16 <- d15 %>% filter(serovar != "infantis" & serovar != "schwarzengrund")

# model using number of contigs only per serovar

set.seed(1233)
# Fig S11-I
m1 <- adonis(simpson ~ num_contigs_sd, data=d16, permutations=1000)
m1
numcontigsSDsimpSerovar <- data.frame(m1$aov.tab)
write.csv(numcontigsSDsimpSerovar, 'numcontigsSDsimpSerovar.csv')

# model using total number of nucleotides per genome only per serovar

set.seed(1233)
# Fig S11-J
m2 <- adonis(simpson ~ total_nucl_sd, data=d16, permutations=1000)
m2
totalnuclSDsimpSerovar <- data.frame(m2$aov.tab)
write.csv(totalnuclSDsimpSerovar, 'totalnuclSDsimpSerovar.csv')

# model using the average GC content per genome per serovar

set.seed(1233)
# Fig S11-K
m3 <- adonis(simpson ~ gc_avg_sd, data=d16, permutations=1000)
m3
gcavgSDsimpSerovar <- data.frame(m3$aov.tab)
write.csv(gcavgSDsimpSerovar, 'gcavgSDsimpSerovar.csv')
```

Now we repeat the analysis for serovars of Salmonella enterica - counts of unique STs

```{r, include = FALSE}
# eliminate STs with missing values or zero values

#######################################################################

# MLST program calculations for ST relative frequencies

d1 <- data3 %>%
              filter(program == "mlst" & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "MLST (standard)")

#######################################################################

# stringmslt program kmer 10 calculations for ST relative frequencies

d2 <- data3 %>%
              filter(program == "stringmlst" & kmer == 10 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_10)")

#######################################################################

# stringmslt program kmer 20 calculations for ST relative frequencies

d3 <- data3 %>%
              filter(program == "stringmlst" & kmer == 20 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_20)")

#######################################################################

# stringmslt program kmer 30 calculations for ST relative frequencies

d4 <- data3 %>%
              filter(program == "stringmlst" & kmer == 30 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_30)")

#######################################################################

# stringmslt program kmer 35 calculations for ST relative frequencies

d5 <- data3 %>%
              filter(program == "stringmlst" & kmer == 35 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_35)")

#######################################################################

# stringmslt program kmer 45 calculations for ST relative frequencies

d6 <- data3 %>%
              filter(program == "stringmlst" & kmer == 45 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_45)")

#######################################################################

# stringmslt program kmer 55 calculations for ST relative frequencies

d7 <- data3 %>%
              filter(program == "stringmlst" & kmer == 55 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_55)")

#######################################################################

# stringmslt program kmer 65 calculations for ST relative frequencies

d8 <- data3 %>%
              filter(program == "stringmlst" & kmer == 65 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_65)")

#######################################################################

# stringmslt program kmer 70 calculations for ST relative frequencies

d9 <- data3 %>%
              filter(program == "stringmlst" & kmer == 70 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_70)")

#######################################################################

# stringmslt program kmer 80 calculations for ST relative frequencies

d10 <- data3 %>%
              filter(program == "stringmlst" & kmer == 80 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_80)")

#######################################################################

# stringmslt program kmer 90 calculations for ST relative frequencies

d11 <- data3 %>%
              filter(program == "stringmlst" & kmer == 90 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(n = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_90)")

#######################################################################

# concatenate datasets

d12b <- rbind(d1, d2, d3, d4, d5, d6, d7, d8, d9, d10, d11)

# check for NAs

skim(d12b)

# transform serovar names

d13 <- d12b %>% mutate(Serovars = str_to_title(serovar))

########################################################################
# Plot the number of unique STs 
# Fig S4-O
p1 <- ggplot(d13, aes(x = program, y = n, fill = Serovars)) + ylim(0, 20) +
  theme_bw() + 
  theme(legend.position = "none") +
  xlab("") +
  ylab("Richness") +
theme(axis.text.y = element_text(size = 12)) +
  theme(axis.title.y = element_text(size = 30, face = "bold")) +
  theme(axis.title.x = element_text(size = 30, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 20)) +
  ggtitle("Total number of unique STs across serovars") +
  scale_fill_manual(values = c("gray", "red", "orange", "darkgreen", "darkblue", 
                               "yellow", "purple", "brown", "darkolivegreen2", "cadetblue1",
                               "cyan4", "darksalmon", "pink", "coral1", "azure4",
                               "bisque2", "cyan", "darkolivegreen3", "darkorchid2", "darkorange",
                               "cyan3")) +
  theme(plot.title = element_text(size = 30, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24)) +
  geom_boxplot() + coord_flip() +facet_wrap(~Serovars)
p1

#####################################################################

# merge d12 and d12b to get simpson index as well
data4 <- left_join(d12, d12b, by = c("serovar" = "serovar", "program" = "program", "batch" = "batch"))

# merge results with original data3 to run the statistical model

data5 <- data3 %>% select(sra_id, serovar, batch, num_contigs, total_nucl, gc_avg, st_count, total_alleles_genes)
data5 <- data5 %>% filter(serovar != "jejuni" & serovar != "saureus" & serovar != "listeria")

data6 <- left_join(data5, data4, by = c("serovar" = "serovar", "batch" = "batch"))

data7 <- distinct(data6)

# richness_species dataset

rich_serovar <- data7 %>% drop_na()

# write data out

write.csv(rich_serovar, 'rich_serovar.csv')

###############################################################
# run permanova model using mean for covariates

# get the mean and sd for each metric and species 

data5 <- data3 %>% select(serovar, batch, program, kmer, num_contigs, total_nucl, gc_avg, st_count, total_alleles_genes)
data5 <- rename(data5, Program = program)
data5<- mutate(data5, program = ifelse(Program == "mlst", "MLST (standard)",
                                      ifelse(Program == "stringmlst" & kmer == 10, "stringMLST (kmer_10)",
                                             ifelse(Program == "stringmlst" & kmer == 20, "stringMLST (kmer_20)",
                                                    ifelse(Program == "stringmlst" & kmer == 30, "stringMLST (kmer_30)",
                                                           ifelse(Program == "stringmlst" & kmer == 35, "stringMLST (kmer_35)",
                                                                  ifelse(Program == "stringmlst" & kmer == 45, "stringMLST (kmer_45)",
                                                                         ifelse(Program == "stringmlst" & kmer == 55, "stringMLST (kmer_55)",
                                                                                ifelse(Program == "stringmlst" & kmer == 65, "stringMLST (kmer_65)",
                                                                                       ifelse(Program == "stringmlst" & kmer == 70, "stringMLST (kmer_70)",
                                                                                              ifelse(Program == "stringmlst" & kmer == 80, "stringMLST (kmer_80)",
                                                                                                     ifelse(Program == "stringmlst" & kmer == 90, "stringMLST (kmer_90)", "Others"))))))))))))

data6 <- data5 %>% filter(serovar != "jejuni" & serovar != "saureus" & serovar != "listeria")

# get the mean for num_contigs, total_nucl, gc_avg
par1 <- data6 %>% 
               group_by(serovar, batch) %>%
               summarize(num_contigs_median = median(num_contigs), 
                         total_nucl_mean = mean(total_nucl),
                         gc_avg_mean = mean(gc_avg))

# get the mean for st_count, total_alleles_genes
par2 <- data6 %>% 
               group_by(serovar, batch, program) %>%
               summarize(st_count_mean = mean(st_count), 
                         total_alleles_genes_mean = mean(total_alleles_genes))


# merge par2 and par1

par3 <- left_join(par2, par1, on = c("serovar" = "serovar", "batch" = "batch"))

# merge with simpson index data

d13 <- left_join(d12, d12b, on = c("serovar" = "serovar", "batch" = "batch", "program"= "program"))
d14 <- left_join(d13, par3, on = c("serovar" = "serovar", "batch" = "batch", "program"= "program"))

############################################################################################
# Running all models 

# Model with species and program
set.seed(1233)
# Fig S10-A
mod1 <- adonis(n ~ serovar * program, data=d14, permutations=1000)
mod1 
serovar_program_STRichness <- data.frame(mod1$aov.tab)
write.csv(serovar_program_STRichness, 'serovar_program_STRichness.csv')

# model for species only

set.seed(1233)
# Fig S10-B
mod2 <- adonis(n ~ serovar, data=d14, permutations=1000)
mod2 
serovarSTRichness <- data.frame(mod2$aov.tab)
write.csv(serovarSTRichness, 'serovarSTRichness.csv')

# model for program only

set.seed(1233)
# Fig S10-C
mod3 <- adonis(n ~ program, data=d14, permutations=1000)
mod3 
programSTRichnessSerovar <- data.frame(mod3$aov.tab)
write.csv(programSTRichnessSerovar, 'programSTRichnessSerovar.csv')

# model for st_count_mean only

set.seed(1233)
# Fig S10-G
mod4 <- adonis(n ~ st_count_mean, data=d14, permutations=1000)
mod4 
stcountmeanSTRichnessSerovar <- data.frame(mod4$aov.tab)
write.csv(stcountmeanSTRichnessSerovar, 'stcountmeanSTRichnessSerovar.csv')

# model for total_alleles_genes_mean only

set.seed(1233)
# Fig S10-H
mod5 <- adonis(n ~ total_alleles_genes_mean, data=d14, permutations=1000)
mod5 
totalallelesgenesmeanSTRichnessSerovar <- data.frame(mod5$aov.tab)
write.csv(totalallelesgenesmeanSTRichnessSerovar, 'totalallelesgenesmeanSTRichnessSerovar.csv')

# model for num_contigs_median only

set.seed(1233)
# Fig S10-D
mod6 <- adonis(n ~ num_contigs_median, data=d14, permutations=1000)
mod6 
numcontigsSTRichnessSerovar <- data.frame(mod6$aov.tab)
write.csv(numcontigsSTRichnessSerovar, 'numcontigsSTRichnessSerovar.csv')

# model for total_nucl_mean only

set.seed(1233)
# Fig S10-E
mod7 <- adonis(n ~ total_nucl_mean, data=d14, permutations=1000)
mod7 
total_nucl_meanSTRichnessSerovar <- data.frame(mod7$aov.tab)
write.csv(total_nucl_meanSTRichnessSerovar, 'total_nucl_meanSTRichnessSerovar.csv')

# model for gc_avg_mean only

set.seed(1233)
# Fig S10-F
mod8 <- adonis(n ~ gc_avg_mean, data=d14, permutations=1000)
mod8 
gcavgSTRichnessSerovar <- data.frame(mod8$aov.tab)
write.csv(gcavgSTRichnessSerovar, 'gcavgSTRichnessSerovar.csv')

# model for the simpson index only

set.seed(1233)
# Fig S10-I
mod8 <- adonis(n ~ simpson, data=d14, permutations=1000)
mod8 
SimpsonSTRichnessSerovar <- data.frame(mod8$aov.tab)
write.csv(SimpsonSTRichnessSerovar, 'SimpsonSTRichnessSerovar.csv')


###############################################################
###############################################################
###############################################################

# now let's run the same model using the sd 

# get the sd for num_contigs, total_nucl, gc_avg
par4 <- data6 %>% 
               group_by(serovar) %>%
               summarize(num_contigs_sd = sd(num_contigs), 
                         total_nucl_sd = sd(total_nucl),
                         gc_avg_sd = sd(gc_avg))

# merge with simpson index data

d15 <- left_join(d12, d12b, on = c("serovar" = "serovar", "program"= "program", "batch" = "batch"))
d16 <- left_join(d15, par4, on = c("serovar" = "serovar"))

# model using number of contigs only per species

set.seed(1233)
# Fig S10-J
m1 <- adonis(n ~ num_contigs_sd, data=d16, permutations=1000)
m1
numcontigsSDstrichnessSerovar <- data.frame(m1$aov.tab)
write.csv(numcontigsSDstrichnessSerovar, 'numcontigsSDstrichnessSerovar.csv')

# model using total number of nucleotides per genome only per species

set.seed(1233)
# Fig S10-K
m2 <- adonis(n ~ total_nucl_sd, data=d16, permutations=1000)
m2
totalnuclSDstrichnessSerovar <- data.frame(m2$aov.tab)
write.csv(totalnuclSDstrichnessSerovar, 'totalnuclSDstrichnessSerovar.csv')

# model using the average GC content per genome per species

set.seed(1233)
# Fig S10-L
m3 <- adonis(n ~ gc_avg_sd, data=d16, permutations=1000)
m3
gcavgSDstrichnessSerovar <- data.frame(m3$aov.tab)
write.csv(gcavgSDstrichnessSerovar, 'gcavgSDstrichnessSerovar.csv')
```

Analysis of blank calls (not classified) across serovars for Salmonella enterica

```{r, include = FALSE}
data2b <- data2 %>% filter(serovar != "saintpaul")
# now here we care about using missing values 
#######################################################################

# MLST program calculations for ST relative frequencies

d1 <- data2b %>%
              filter(program == "mlst" & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "MLST (standard)")

#######################################################################

# stringmslt program kmer 10 calculations for ST relative frequencies

d2 <- data2b %>%
              filter(program == "stringmlst" & kmer == 10 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_10)")

#######################################################################

# stringmslt program kmer 20 calculations for ST relative frequencies

d3 <- data2b %>%
              filter(program == "stringmlst" & kmer == 20 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_20)")

#######################################################################

# stringmslt program kmer 30 calculations for ST relative frequencies

d4 <- data2b %>%
              filter(program == "stringmlst" & kmer == 30 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_30)")

#######################################################################

# stringmslt program kmer 35 calculations for ST relative frequencies

d5 <- data2b %>%
              filter(program == "stringmlst" & kmer == 35 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_35)")

#######################################################################

# stringmslt program kmer 45 calculations for ST relative frequencies

d6 <- data2b %>%
              filter(program == "stringmlst" & kmer == 45 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_45)")

#######################################################################

# stringmslt program kmer 55 calculations for ST relative frequencies

d7 <- data2b %>%
              filter(program == "stringmlst" & kmer == 55 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_55)")

#######################################################################

# stringmslt program kmer 65 calculations for ST relative frequencies

d8 <- data2b %>%
              filter(program == "stringmlst" & kmer == 65 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_65)")

#######################################################################

# stringmslt program kmer 70 calculations for ST relative frequencies

d9 <- data2b %>%
              filter(program == "stringmlst" & kmer == 70 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_70)")

#######################################################################

# stringmslt program kmer 80 calculations for ST relative frequencies

d10 <- data2b %>%
              filter(program == "stringmlst" & kmer == 80 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_80)")

#######################################################################

# stringmslt program kmer 90 calculations for ST relative frequencies

d11 <- data2b %>%
              filter(program == "stringmlst" & kmer == 90 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_90)")

#######################################################################

# concatenate datasets

d12d <- rbind(d1, d2, d3, d4, d5, d6, d7, d8, d9, d10, d11)

# check for NAs

skim(d12d)

########################################################################
# Plot the proportion of non-classified STs

# change from NA to non-classified

d13 <- d12d

# Set ST equal zero to not classified 

d13 <- mutate(d13, ST = ifelse(st %in% 0, "Not classified", "correct"))

# filter not classified

d14 <- d13 %>% filter(ST == "Not classified") 
d15 <- d14 %>% select(serovar, batch, prop, program)

d16 <- d15 %>% mutate(Serovars = str_to_title(serovar))

################################################################
# export data for analysis


data5 <- data3 %>% select(sra_id, serovar, batch, num_contigs, total_nucl, gc_avg, st_count, total_alleles_genes)
data6 <- data5 %>% filter(serovar != "jejuni" & serovar != "listeria" & serovar != "saureus")

d17<- left_join(data6, d15, by = c("serovar" = "serovar", "batch" = "batch"))

d18 <- distinct(d17)

# merge with simpson index data

d19 <- left_join(d18, d12, by = c("serovar" = "serovar", "batch" = "batch", "program" = "program"))

d20 <- distinct(d19)

# richness_species dataset

prop_serovar <- d20 %>% drop_na()

# write data out

write.csv(prop_serovar, 'prop_serovar.csv')

# Plot the proportion of not classified genomes across programs 
# Fig S4-Q
p1 <- ggplot(d16, aes(x = program, y = prop, fill = Serovars)) + ylim(0, 100) +
  theme_bw() + 
  theme(legend.position = "none") +
  xlab("") +
  ylab("Proportion") +
theme(axis.text.y = element_text(size = 10)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 15)) +
  ggtitle("C. Proportion of non-classified STs across serovars") +
  scale_fill_manual(values = c("gray", "red", "orange", "darkgreen", "darkblue", 
                               "yellow", "purple", "brown", "darkolivegreen2", "cadetblue1",
                               "cyan4", "darksalmon", "pink", "coral1", "azure4",
                               "bisque2", "cyan", "darkolivegreen3", "darkorchid2", "darkorange",
                               "cyan3")) +
  theme(plot.title = element_text(size = 25, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24)) +
  geom_boxplot() + coord_flip() + facet_wrap(~Serovars)
p1
######################################################################

# calculate the standard deviation for non-classification

d17 <- d16 %>%
         group_by(Serovars, program) %>%
         summarise(std = sd(prop)) %>%
         drop_na()
# Fig S4-R
p2 <- ggplot(d17, aes(x = program, y = std, fill = Serovars)) + ylim(0, 10) +
  theme_bw() + 
  theme(legend.position = "none") +
  xlab("") +
  ylab("SD") +
theme(axis.text.y = element_text(size = 12)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 16, hjust=1)) +
  ggtitle("D. Standard deviation for the proportion of non-classified STs across serovars") +
  scale_fill_manual(values = c("gray", "red", "orange", "darkgreen", "darkblue", 
                               "yellow", "purple", "brown", "darkolivegreen2", "cadetblue1",
                               "cyan4", "darksalmon", "pink", "coral1", "azure4",
                               "bisque2", "cyan", "darkolivegreen3", "darkorchid2", "darkorange",
                               "cyan3")) +
  theme(plot.title = element_text(size = 23, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=30, face = "bold")) +
  theme(legend.text=element_text(size=24)) +
  geom_bar(stat="identity", position=position_dodge()) + coord_flip() + facet_wrap(~Serovars)
p2

###############################################################
# run permanova model using mean for covariates

# get the mean and sd for each metric and species 

data5 <- data3 %>% select(serovar, batch, program, kmer, num_contigs, total_nucl, gc_avg, st_count, total_alleles_genes)
data5 <- rename(data5, Program = program)
data5<- mutate(data5, program = ifelse(Program == "mlst", "MLST (standard)",
                                      ifelse(Program == "stringmlst" & kmer == 10, "stringMLST (kmer_10)",
                                             ifelse(Program == "stringmlst" & kmer == 20, "stringMLST (kmer_20)",
                                                    ifelse(Program == "stringmlst" & kmer == 30, "stringMLST (kmer_30)",
                                                           ifelse(Program == "stringmlst" & kmer == 35, "stringMLST (kmer_35)",
                                                                  ifelse(Program == "stringmlst" & kmer == 45, "stringMLST (kmer_45)",
                                                                         ifelse(Program == "stringmlst" & kmer == 55, "stringMLST (kmer_55)",
                                                                                ifelse(Program == "stringmlst" & kmer == 65, "stringMLST (kmer_65)",
                                                                                       ifelse(Program == "stringmlst" & kmer == 70, "stringMLST (kmer_70)",
                                                                                              ifelse(Program == "stringmlst" & kmer == 80, "stringMLST (kmer_80)",
                                                                                                     ifelse(Program == "stringmlst" & kmer == 90, "stringMLST (kmer_90)", "Others"))))))))))))

data6 <- data5 %>% filter(serovar != "jejuni" & serovar != "saureus" & serovar != "listeria")

# get the mean for num_contigs, total_nucl, gc_avg
par1 <- data6 %>% 
               group_by(serovar, batch) %>%
               summarize(num_contigs_median = median(num_contigs), 
                         total_nucl_mean = mean(total_nucl),
                         gc_avg_mean = mean(gc_avg))

# get the mean for st_count, total_alleles_genes
par2 <- data6 %>% 
               group_by(serovar, batch, program) %>%
               summarize(st_count_mean = mean(st_count), 
                         total_alleles_genes_mean = mean(total_alleles_genes))


# merge par2 and par1

par3 <- left_join(par2, par1, on = c("serovar" = "serovar", "batch" = "batch"))

# merge with simpson index data

d16b <- left_join(d15, d12, on = c("serovar" = "serovar", "batch" = "batch", "program"= "program"))
d17b <- left_join(d16b, par3, on = c("serovar" = "serovar", "batch" = "batch", "program"= "program"))
d18b <- d17b %>% drop_na()

############################################################################################
# Running all models 

# Model with species and program
set.seed(1233)
# Fig S12-A
mod1 <- adonis(prop ~ serovar * program, data=d18b, permutations=1000)
mod1 
serovar_program_prop <- data.frame(mod1$aov.tab)
write.csv(serovar_program_prop, 'serovar_program_prop.csv')

# model for species only

set.seed(1233)
# Fig S12-B
mod2 <- adonis(prop ~ serovar, data=d18b, permutations=1000)
mod2 
serovarprop <- data.frame(mod2$aov.tab)
write.csv(serovarprop, 'serovarprop.csv')

# model for program only

set.seed(1233)
# Fig S12-C
mod3 <- adonis(prop ~ program, data=d18b, permutations=1000)
mod3 
programpropserovar <- data.frame(mod3$aov.tab)
write.csv(programpropserovar, 'programpropserovar.csv')

# model for st_count_mean only

set.seed(1233)
# Fig S12-G
mod4 <- adonis(prop ~ st_count_mean, data=d18b, permutations=1000)
mod4 
stcountmeanpropserovar <- data.frame(mod4$aov.tab)
write.csv(stcountmeanpropserovar, 'stcountmeanpropserovar.csv')

# model for total_alleles_genes_mean only

set.seed(1233)
# Fig S12-H
mod5 <- adonis(prop ~ total_alleles_genes_mean, data=d18b, permutations=1000)
mod5 
totalallelesgenesmeanpropserovar <- data.frame(mod5$aov.tab)
write.csv(totalallelesgenesmeanpropserovar, 'totalallelesgenesmeanpropserovar.csv')

# model for num_contigs_median only

set.seed(1233)
# Fig S12-D
mod6 <- adonis(prop ~ num_contigs_median, data=d18b, permutations=1000)
mod6 
numcontigspropserovar <- data.frame(mod6$aov.tab)
write.csv(numcontigspropserovar, 'numcontigspropserovar.csv')

# model for total_nucl_mean only

set.seed(1233)
# Fig S12-E
mod7 <- adonis(prop ~ total_nucl_mean, data=d18b, permutations=1000)
mod7 
total_nucl_meanpropserovar <- data.frame(mod7$aov.tab)
write.csv(total_nucl_meanpropserovar, 'total_nucl_meanpropserovar.csv')

# model for gc_avg_mean only

set.seed(1233)
# Fig S12-F
mod8 <- adonis(prop ~ gc_avg_mean, data=d18b, permutations=1000)
mod8 
gcavgpropserovar <- data.frame(mod8$aov.tab)
write.csv(gcavgpropserovar, 'gcavgpropserovar.csv')

# model for the simpson index only

set.seed(1233)
# Fig S12-I
mod8 <- adonis(prop ~ simpson, data=d18b, permutations=1000)
mod8 
Simpsonpropserovar <- data.frame(mod8$aov.tab)
write.csv(Simpsonpropserovar, 'Simpsonpropserovar.csv')


###############################################################
###############################################################

###############################################################

# now let's run the same model using the sd 

# get the sd for num_contigs, total_nucl, gc_avg
par4 <- data6 %>% 
               group_by(serovar) %>%
               summarize(num_contigs_sd = sd(num_contigs), 
                         total_nucl_sd = sd(total_nucl),
                         gc_avg_sd = sd(gc_avg))

# merge with simpson index data

d16c <- left_join(d15, d12, on = c("serovar" = "serovar", "program"= "program", "batch" = "batch"))
d17c <- left_join(d16c, par4, on = c("serovar" = "serovar", "program"= "program"))
d18c <- d17c %>% drop_na()

# model using number of contigs only per species

set.seed(1233)
# Fig S12-J
m1 <- adonis(prop ~ num_contigs_sd, data=d18c, permutations=1000)
m1
numcontigsSDpropserovar <- data.frame(m1$aov.tab)
write.csv(numcontigsSDpropserovar, 'numcontigsSDpropserovar.csv')

# model using total number of nucleotides per genome only per species

set.seed(1233)
# Fig S12-K
m2 <- adonis(prop ~ total_nucl_sd, data=d18c, permutations=1000)
m2
totalnuclSDpropserovar <- data.frame(m2$aov.tab)
write.csv(totalnuclSDpropserovar, 'totalnuclSDpropserovar.csv')

# model using the average GC content per genome per species

set.seed(1233)
# Fig S12-L
m3 <- adonis(prop ~ gc_avg_sd, data=d18c, permutations=1000)
m3
gcavgSDpropserovar <- data.frame(m3$aov.tab)
write.csv(gcavgSDpropserovar, 'gcavgSDpropserovar.csv')
```

biplot richness and simpson serovars

```{r, include = FALSE}
# to calculate the unique CLASSIFIED STs, we have to remove the missing values because we are counting
# genomes that were successfully classified only. Pay attention to everything I am writing here because it 
# will help you when writing the paper

# if you always want to check if there are NAs run either of the following functions

# sum(is.na(data))
# skim(data)

#######################################################################

# MLST program calculations for ST relative frequencies

d1 <- data3 %>%
              filter(program == "mlst" & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "MLST (standard)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 10 calculations for ST relative frequencies

d2 <- data3 %>%
              filter(program == "stringmlst" & kmer == 10 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_10)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 20 calculations for ST relative frequencies

d3 <- data3 %>%
              filter(program == "stringmlst" & kmer == 20 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_20)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 30 calculations for ST relative frequencies

d4 <- data3 %>%
              filter(program == "stringmlst" & kmer == 30 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_30)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 35 calculations for ST relative frequencies

d5 <- data3 %>%
              filter(program == "stringmlst" & kmer == 35 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_35)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 45 calculations for ST relative frequencies

d6 <- data3 %>%
              filter(program == "stringmlst" & kmer == 45 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_45)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 55 calculations for ST relative frequencies

d7 <- data3 %>%
              filter(program == "stringmlst" & kmer == 55 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_55)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 65 calculations for ST relative frequencies

d8 <- data3 %>%
              filter(program == "stringmlst" & kmer == 65 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_65)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 70 calculations for ST relative frequencies

d9 <- data3 %>%
              filter(program == "stringmlst" & kmer == 70 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_70)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 80 calculations for ST relative frequencies

d10 <- data3 %>%
              filter(program == "stringmlst" & kmer == 80 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_80)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 90 calculations for ST relative frequencies

d11 <- data3 %>%
              filter(program == "stringmlst" & kmer == 90 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_90)") %>%
              mutate(metric = "Richness")

#######################################################################

# concatenate datasets

d12 <- rbind(d1, d2, d3, d4, d5, d6, d7, d8, d9, d10, d11)

# check for NAs

skim(d12)
#########################################################################
#######################################################################

# MLST program calculations for ST relative frequencies

d1b <- data3 %>%
              filter(program == "mlst" & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "MLST (standard)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 10 calculations for ST relative frequencies

d2b <- data3 %>%
              filter(program == "stringmlst" & kmer == 10 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_10)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 20 calculations for ST relative frequencies

d3b <- data3 %>%
              filter(program == "stringmlst" & kmer == 20 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_20)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 30 calculations for ST relative frequencies

d4b <- data3 %>%
              filter(program == "stringmlst" & kmer == 30 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_30)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 35 calculations for ST relative frequencies

d5b <- data3 %>%
              filter(program == "stringmlst" & kmer == 35 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_35)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 45 calculations for ST relative frequencies

d6b <- data3 %>%
              filter(program == "stringmlst" & kmer == 45 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_45)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 55 calculations for ST relative frequencies

d7b <- data3 %>%
              filter(program == "stringmlst" & kmer == 55 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index= mean(index)) %>%
              mutate(program = "stringMLST (kmer_55)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 65 calculations for ST relative frequencies

d8b <- data3 %>%
              filter(program == "stringmlst" & kmer == 65 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_65)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 70 calculations for ST relative frequencies

d9b <- data3 %>%
              filter(program == "stringmlst" & kmer == 70 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_70)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 80 calculations for ST relative frequencies

d10b <- data3 %>%
              filter(program == "stringmlst" & kmer == 80 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_80)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 90 calculations for ST relative frequencies

d11b <- data3 %>%
              filter(program == "stringmlst" & kmer == 90 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_90)") %>%
              mutate(metric = "Simpson")

#######################################################################

# concatenate datasets

d12b <- rbind(d1b, d2b, d3b, d4b, d5b, d6b, d7b, d8b, d9b, d10b, d11b)

# check for NAs

skim(d12b)

########################################################################

data2b <- data2 %>% filter(serovar != "saintpaul")
# now here we care about using missing values 
#######################################################################

# MLST program calculations for ST relative frequencies

d1 <- data2b %>%
              filter(program == "mlst" & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "MLST (standard)")

#######################################################################

# stringmslt program kmer 10 calculations for ST relative frequencies

d2 <- data2b %>%
              filter(program == "stringmlst" & kmer == 10 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_10)")

#######################################################################

# stringmslt program kmer 20 calculations for ST relative frequencies

d3 <- data2b %>%
              filter(program == "stringmlst" & kmer == 20 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_20)")

#######################################################################

# stringmslt program kmer 30 calculations for ST relative frequencies

d4 <- data2b %>%
              filter(program == "stringmlst" & kmer == 30 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_30)")

#######################################################################

# stringmslt program kmer 35 calculations for ST relative frequencies

d5 <- data2b %>%
              filter(program == "stringmlst" & kmer == 35 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_35)")

#######################################################################

# stringmslt program kmer 45 calculations for ST relative frequencies

d6 <- data2b %>%
              filter(program == "stringmlst" & kmer == 45 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_45)")

#######################################################################

# stringmslt program kmer 55 calculations for ST relative frequencies

d7 <- data2b %>%
              filter(program == "stringmlst" & kmer == 55 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_55)")

#######################################################################

# stringmslt program kmer 65 calculations for ST relative frequencies

d8 <- data2b %>%
              filter(program == "stringmlst" & kmer == 65 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_65)")

#######################################################################

# stringmslt program kmer 70 calculations for ST relative frequencies

d9 <- data2b %>%
              filter(program == "stringmlst" & kmer == 70 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_70)")

#######################################################################

# stringmslt program kmer 80 calculations for ST relative frequencies

d10 <- data2b %>%
              filter(program == "stringmlst" & kmer == 80 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_80)")

#######################################################################

# stringmslt program kmer 90 calculations for ST relative frequencies

d11 <- data2b %>%
              filter(program == "stringmlst" & kmer == 90 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_90)")

#######################################################################

# concatenate datasets

d12d <- rbind(d1, d2, d3, d4, d5, d6, d7, d8, d9, d10, d11)

# check for NAs

skim(d12d)

########################################################################
# Plot the proportion of non-classified STs

# change from NA to non-classified

d13 <- d12d

# Set ST equal zero to not classified 

d13 <- mutate(d13, ST = ifelse(st %in% 0, "Not classified", "correct"))

# filter not classified

d14 <- d13 %>% filter(ST == "Not classified") 
d15 <- d14 %>% select(serovar, batch, prop, program)
d15 <- rename(d15, index = prop)
d15 <- d15 %>% mutate(metric = "Miscalls")
##########################################################################
# concatenate datasets 

d17 <- bind_rows(d12, d12b, d15)

# get the mean for each species

#d18 <- d17 %>% group_by(serovar, program, metric) %>% summarise(index = mean(index))

# spread the data

d18 <- d17 %>% spread(metric, index) %>% drop_na()

# transform serovar names

d19 <- d18 %>% mutate(Serovars = str_to_title(serovar)) %>% drop_na()

# plot 1
# Fig S13-A
p1 <- ggplot(d19, aes(x = Richness, y = Simpson, color = program)) + ylim(-0.5, 1) + xlim(0, 20) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Richness") +
  ylab("Simpson's index of diversity") +
theme(axis.text.y = element_text(size = 20)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 20)) +
  ggtitle("Biplot using mean of indexes across serovars") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 35, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=25, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  geom_jitter(size = 2, width = 0.5, height = 0.1) +
  facet_wrap(~Serovars)
p1

# plot 2
# Fig S13-B
p2 <- ggplot(d19, aes(x = Miscalls, y = Simpson, color = program)) + ylim(-0.5, 1) + xlim(0, 50) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Proportion of non-classified STs") +
  ylab("Simpson's index of diversity") +
theme(axis.text.y = element_text(size = 20)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 20)) +
  ggtitle("Biplot using mean of indexes across serovars") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 35, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=25, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  geom_jitter(size = 2, width = 0.5, height = 0.1) +
  facet_wrap(~Serovars)
p2

# plot 3
# Fig S13-C
p3 <- ggplot(d19, aes(x = Miscalls, y = Richness, color = program)) + ylim(0, 20) + xlim(0, 50) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Proportion of non-classified STs") +
  ylab("Richness") +
theme(axis.text.y = element_text(size = 20)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 20)) +
  ggtitle("Biplot using mean of indexes across serovars") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 35, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=25, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  geom_jitter(size = 2, width = 0.5, height = 0.1) +
  facet_wrap(~Serovars)
p3
```

biplot for serovar accounting for genome intrisic features

```{r, include = FALSE}
#######################################################################

# MLST program calculations for ST relative frequencies

d1 <- data3 %>%
              filter(program == "mlst" & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "MLST (standard)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 10 calculations for ST relative frequencies

d2 <- data3 %>%
              filter(program == "stringmlst" & kmer == 10 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_10)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 20 calculations for ST relative frequencies

d3 <- data3 %>%
              filter(program == "stringmlst" & kmer == 20 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_20)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 30 calculations for ST relative frequencies

d4 <- data3 %>%
              filter(program == "stringmlst" & kmer == 30 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_30)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 35 calculations for ST relative frequencies

d5 <- data3 %>%
              filter(program == "stringmlst" & kmer == 35 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_35)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 45 calculations for ST relative frequencies

d6 <- data3 %>%
              filter(program == "stringmlst" & kmer == 45 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_45)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 55 calculations for ST relative frequencies

d7 <- data3 %>%
              filter(program == "stringmlst" & kmer == 55 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_55)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 65 calculations for ST relative frequencies

d8 <- data3 %>%
              filter(program == "stringmlst" & kmer == 65 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_65)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 70 calculations for ST relative frequencies

d9 <- data3 %>%
              filter(program == "stringmlst" & kmer == 70 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_70)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 80 calculations for ST relative frequencies

d10 <- data3 %>%
              filter(program == "stringmlst" & kmer == 80 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_80)") %>%
              mutate(metric = "Richness")

#######################################################################

# stringmslt program kmer 90 calculations for ST relative frequencies

d11 <- data3 %>%
              filter(program == "stringmlst" & kmer == 90 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch) %>%
              summarise(index = n_distinct(st)) %>%
              mutate(program = "stringMLST (kmer_90)") %>%
              mutate(metric = "Richness")

#######################################################################

# concatenate datasets

d12 <- rbind(d1, d2, d3, d4, d5, d6, d7, d8, d9, d10, d11)

# check for NAs

skim(d12)
#########################################################################
#######################################################################

# MLST program calculations for ST relative frequencies

d1b <- data3 %>%
              filter(program == "mlst" & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "MLST (standard)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 10 calculations for ST relative frequencies

d2b <- data3 %>%
              filter(program == "stringmlst" & kmer == 10 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_10)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 20 calculations for ST relative frequencies

d3b <- data3 %>%
              filter(program == "stringmlst" & kmer == 20 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_20)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 30 calculations for ST relative frequencies

d4b <- data3 %>%
              filter(program == "stringmlst" & kmer == 30 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_30)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 35 calculations for ST relative frequencies

d5b <- data3 %>%
              filter(program == "stringmlst" & kmer == 35 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_35)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 45 calculations for ST relative frequencies

d6b <- data3 %>%
              filter(program == "stringmlst" & kmer == 45 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_45)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 55 calculations for ST relative frequencies

d7b <- data3 %>%
              filter(program == "stringmlst" & kmer == 55 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index= mean(index)) %>%
              mutate(program = "stringMLST (kmer_55)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 65 calculations for ST relative frequencies

d8b <- data3 %>%
              filter(program == "stringmlst" & kmer == 65 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_65)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 70 calculations for ST relative frequencies

d9b <- data3 %>%
              filter(program == "stringmlst" & kmer == 70 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_70)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 80 calculations for ST relative frequencies

d10b <- data3 %>%
              filter(program == "stringmlst" & kmer == 80 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_80)") %>%
              mutate(metric = "Simpson")

#######################################################################

# stringmslt program kmer 90 calculations for ST relative frequencies

d11b <- data3 %>%
              filter(program == "stringmlst" & kmer == 90 & species == "Salmonella enterica") %>%
              drop_na() %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(index = diversity(n, "simpson")) %>%
              group_by(serovar, batch) %>%
              summarise(index = mean(index)) %>%
              mutate(program = "stringMLST (kmer_90)") %>%
              mutate(metric = "Simpson")

#######################################################################

# concatenate datasets

d12b <- rbind(d1b, d2b, d3b, d4b, d5b, d6b, d7b, d8b, d9b, d10b, d11b)

# check for NAs

skim(d12b)

########################################################################

data2b <- data2 %>% filter(serovar != "saintpaul")
# now here we care about using missing values 
#######################################################################

# MLST program calculations for ST relative frequencies

d1 <- data2b %>%
              filter(program == "mlst" & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "MLST (standard)")

#######################################################################

# stringmslt program kmer 10 calculations for ST relative frequencies

d2 <- data2b %>%
              filter(program == "stringmlst" & kmer == 10 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_10)")

#######################################################################

# stringmslt program kmer 20 calculations for ST relative frequencies

d3 <- data2b %>%
              filter(program == "stringmlst" & kmer == 20 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_20)")

#######################################################################

# stringmslt program kmer 30 calculations for ST relative frequencies

d4 <- data2b %>%
              filter(program == "stringmlst" & kmer == 30 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_30)")

#######################################################################

# stringmslt program kmer 35 calculations for ST relative frequencies

d5 <- data2b %>%
              filter(program == "stringmlst" & kmer == 35 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_35)")

#######################################################################

# stringmslt program kmer 45 calculations for ST relative frequencies

d6 <- data2b %>%
              filter(program == "stringmlst" & kmer == 45 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_45)")

#######################################################################

# stringmslt program kmer 55 calculations for ST relative frequencies

d7 <- data2b %>%
              filter(program == "stringmlst" & kmer == 55 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_55)")

#######################################################################

# stringmslt program kmer 65 calculations for ST relative frequencies

d8 <- data2b %>%
              filter(program == "stringmlst" & kmer == 65 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_65)")

#######################################################################

# stringmslt program kmer 70 calculations for ST relative frequencies

d9 <- data2b %>%
              filter(program == "stringmlst" & kmer == 70 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_70)")

#######################################################################

# stringmslt program kmer 80 calculations for ST relative frequencies

d10 <- data2b %>%
              filter(program == "stringmlst" & kmer == 80 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_80)")

#######################################################################

# stringmslt program kmer 90 calculations for ST relative frequencies

d11 <- data2b %>%
              filter(program == "stringmlst" & kmer == 90 & species == "Salmonella enterica") %>%
              group_by(serovar, batch, st) %>%
              summarise(n = n()) %>%
              mutate(prop = n/sum(n)*100) %>%
              mutate(program = "stringMLST (kmer_90)")

#######################################################################

# concatenate datasets

d12d <- rbind(d1, d2, d3, d4, d5, d6, d7, d8, d9, d10, d11)

# check for NAs

skim(d12d)

########################################################################
# Plot the proportion of non-classified STs

# change from NA to non-classified

d13 <- d12d

# Set ST equal zero to not classified 

d13 <- mutate(d13, ST = ifelse(st %in% 0, "Not classified", "correct"))

# filter not classified

d14 <- d13 %>% filter(ST == "Not classified") 
d15 <- d14 %>% select(serovar, batch, prop, program)
d15 <- rename(d15, index = prop)
d15 <- d15 %>% mutate(metric = "Miscalls")
##########################################################################
# concatenate datasets 

d17 <- bind_rows(d12, d12b, d15)

# get the mean for each species

#d18 <- d17 %>% group_by(serovar, program, metric) %>% summarise(index = mean(index))

# spread the data

d18 <- d17 %>% spread(metric, index) %>% drop_na()

# transform serovar names

#d19 <- d18 %>% mutate(Serovars = str_to_title(serovar)) %>% drop_na()


############################################################################################
#######################################################################
# get the mean and sd for each metric and species 

data5 <- data3 %>% select(serovar, batch, program, kmer, num_contigs, total_nucl, gc_avg, st_count, total_alleles_genes)
data5 <- rename(data5, Program = program)
data5<- mutate(data5, program = ifelse(Program == "mlst", "MLST (standard)",
                                      ifelse(Program == "stringmlst" & kmer == 10, "stringMLST (kmer_10)",
                                             ifelse(Program == "stringmlst" & kmer == 20, "stringMLST (kmer_20)",
                                                    ifelse(Program == "stringmlst" & kmer == 30, "stringMLST (kmer_30)",
                                                           ifelse(Program == "stringmlst" & kmer == 35, "stringMLST (kmer_35)",
                                                                  ifelse(Program == "stringmlst" & kmer == 45, "stringMLST (kmer_45)",
                                                                         ifelse(Program == "stringmlst" & kmer == 55, "stringMLST (kmer_55)",
                                                                                ifelse(Program == "stringmlst" & kmer == 65, "stringMLST (kmer_65)",
                                                                                       ifelse(Program == "stringmlst" & kmer == 70, "stringMLST (kmer_70)",
                                                                                              ifelse(Program == "stringmlst" & kmer == 80, "stringMLST (kmer_80)",
                                                                                                     ifelse(Program == "stringmlst" & kmer == 90, "stringMLST (kmer_90)", "Others"))))))))))))

data6 <- data5 %>% filter(serovar != "jejuni" & serovar != "saureus" & serovar != "listeria")

# get the mean for num_contigs, total_nucl, gc_avg
par1 <- data6 %>% 
               group_by(serovar, batch) %>%
               summarize(num_contigs_median = median(num_contigs), 
                         total_nucl_mean = mean(total_nucl),
                         gc_avg_mean = mean(gc_avg))

# get the mean for st_count, total_alleles_genes
par2 <- data6 %>% 
               group_by(serovar, batch, program) %>%
               summarize(st_count_mean = mean(st_count), 
                         total_alleles_genes_mean = mean(total_alleles_genes))


# merge par2 and par1

par3 <- left_join(par2, par1, on = c("serovar" = "serovar", "batch" = "batch"))

# merge with simpson index data

d20 <- left_join(d18, par3, on = c("serovar" = "serovar", "batch" = "batch", "program" = "program"))
d21 <- d20 %>% mutate(Serovars = str_to_title(serovar)) %>% drop_na()

############################################################################################################

# plot 1
# Fig S13-D
p1 <- ggplot(d21, aes(x = num_contigs_median, y = Richness, color = program)) + xlim(0, 300) + ylim(0, 20) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Median number of contigs per genome per serovar") +
  ylab("Richness") +
theme(axis.text.y = element_text(size = 20)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 20, hjust = 1)) +
  ggtitle("Biplot using mean of indexes across serovars") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 35, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=25, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  geom_jitter(size = 2, width = 0.5, height = 0.5) +
  facet_wrap(~Serovars)
p1

# plot 2
# Fig S13-E
p2 <- ggplot(d21, aes(x = total_nucl_mean, y = Richness, color = program)) + xlim(0, 5000000) + ylim(0, 20) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Average total number of nucleotides per genome per serovar") +
  ylab("Richness") +
theme(axis.text.y = element_text(size = 20)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 16, hjust = 1)) +
  ggtitle("Biplot using mean of indexes across serovars") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 35, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=25, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  geom_jitter(size = 2, width = 0.5, height = 0.5) +
  facet_wrap(~Serovars)
p2

# plot 3
# Fig S13-F
p3 <- ggplot(d21, aes(x = gc_avg_mean, y = Richness, color = program)) + xlim(0, 60) + ylim(0, 20) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Average %GC content per genome per serovar") +
  ylab("Richness") +
theme(axis.text.y = element_text(size = 20)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 20)) +
  ggtitle("Biplot using mean of indexes across serovars") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 35, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=25, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  geom_jitter(size = 2, width = 0.5, height = 0.5) +
  facet_wrap(~Serovars)
p3

# plot 4
# Fig S13-G
p4 <- ggplot(d21, aes(x = st_count_mean, y = Richness, color = program)) + xlim(0, 10000) + ylim(0, 20) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Total number of unique STs per serovar \n & database generated across programs") +
  ylab("Richness") +
theme(axis.text.y = element_text(size = 20)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 20, hjust = 1)) +
  ggtitle("Biplot using mean of indexes across serovars") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 35, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=25, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  geom_jitter(size = 2, width = 0.5, height = 0.5) +
  facet_wrap(~Serovars)
p4

# plot 5
# Fig S13-H
p5 <- ggplot(d21, aes(x = total_alleles_genes_mean, y = Richness, color = program)) + xlim(0, 10000) + ylim(0, 20) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Total number of unique alleles per serovar \n & database generated across programs") +
  ylab("Richness") +
theme(axis.text.y = element_text(size = 20)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 20, hjust = 1)) +
  ggtitle("Biplot using mean of indexes across serovars") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 35, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=25, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  geom_jitter(size = 2, width = 0.5, height = 0.5) +
  facet_wrap(~Serovars)
p5
####################################################################################
############################################################################################################

# plot 1
# Fig S13-I
p1b <- ggplot(d21, aes(x = num_contigs_median, y = Simpson, color = program)) + xlim(0, 300) + ylim(-0.5, 1) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Median number of contigs per genome per serovar") +
  ylab("Simpson's D index of diversity") +
theme(axis.text.y = element_text(size = 20)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 20, hjust = 1)) +
  ggtitle("Biplot using mean of indexes across serovars") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 35, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=25, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  geom_jitter(size = 2, width = 0.1, height = 0.1) +
  facet_wrap(~Serovars)
p1b

# plot 2
# Fig S13-J
p2b <- ggplot(d21, aes(x = total_nucl_mean, y = Simpson, color = program)) + xlim(0, 5000000) + ylim(-0.5, 1) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Average total number of nucleotides per genome per serovar") +
  ylab("Simpson's D index of diversity") +
  theme(axis.text.y = element_text(size = 20)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 16, hjust = 1)) +
  ggtitle("Biplot using mean of indexes across serovars") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 35, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=25, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  geom_jitter(size = 2, width = 0.1, height = 0.1) +
  facet_wrap(~Serovars)
p2b

# plot 3
# Fig S13-K
p3b <- ggplot(d21, aes(x = gc_avg_mean, y = Simpson, color = program)) + xlim(0, 60) + ylim(-0.5, 1) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Average %GC content per genome per serovar") +
  ylab("Simpson's D index of diversity") +
  theme(axis.text.y = element_text(size = 20)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 20)) +
  ggtitle("Biplot using mean of indexes across serovars") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 35, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=25, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  geom_jitter(size = 2, width = 0.1, height = 0.1) +
  facet_wrap(~Serovars)
p3b

# plot 4
# Fig S13-L
p4b <- ggplot(d21, aes(x = st_count_mean, y = Simpson, color = program)) + xlim(0, 10000) + ylim(-0.5, 1) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Total number of unique STs per serovar \n & database generated across programs") +
  ylab("Simpson's D index of diversity") +
theme(axis.text.y = element_text(size = 20)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 20, hjust = 1)) +
  ggtitle("Biplot using mean of indexes across serovars") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 35, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=25, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  geom_jitter(size = 2, width = 0.1, height = 0.1) +
  facet_wrap(~Serovars)
p4b

# plot 5
# Fig S13-M
p5b <- ggplot(d21, aes(x = total_alleles_genes_mean, y = Simpson, color = program)) + xlim(0, 10000) + ylim(-0.5, 1) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Total number of unique alleles per serovar \n & database generated across programs") +
  ylab("Simpson's D index of diversity") +
theme(axis.text.y = element_text(size = 20)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 20, hjust = 1)) +
  ggtitle("Biplot using mean of indexes across serovars") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 35, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=25, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  geom_jitter(size = 2, width = 0.1, height = 0.1) +
  facet_wrap(~Serovars)
p5b
####################################################################################
####################################################################################
############################################################################################################

# plot 1
# Fig S13-N
p1c <- ggplot(d21, aes(x = num_contigs_median, y = Miscalls, color = program)) + xlim(0, 300) + ylim(0, 100) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Median number of contigs per genome per serovar") +
  ylab("Proportion of non-classified STs") +
theme(axis.text.y = element_text(size = 20)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 20, hjust = 1)) +
  ggtitle("Biplot using mean of indexes across serovars") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 35, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=25, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  geom_jitter(size = 2, width = 0.1, height = 0.1) +
  facet_wrap(~Serovars)
p1c

# plot 2
# Fig S13-O
p2c <- ggplot(d21, aes(x = total_nucl_mean, y = Miscalls, color = program)) + xlim(0, 5000000) + ylim(0, 100) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Average total number of nucleotides per genome per serovar") +
  ylab("Proportion of non-classified STs") +
  theme(axis.text.y = element_text(size = 20)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 16, hjust = 1)) +
  ggtitle("Biplot using mean of indexes across serovars") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 35, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=25, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  geom_jitter(size = 2, width = 0.1, height = 0.1) +
  facet_wrap(~Serovars)
p2c

# plot 3
# Fig S13-P
p3c <- ggplot(d21, aes(x = gc_avg_mean, y = Miscalls, color = program)) + xlim(0, 60) + ylim(0, 100) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Average %GC content per genome per serovar") +
  ylab("Proportion of non-classified STs") +
  theme(axis.text.y = element_text(size = 20)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 20)) +
  ggtitle("Biplot using mean of indexes across serovars") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 35, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=25, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  geom_jitter(size = 2, width = 0.1, height = 0.1) +
  facet_wrap(~Serovars)
p3c

# plot 4
# Fig S13-Q
p4c <- ggplot(d21, aes(x = st_count_mean, y = Miscalls, color = program)) + xlim(0, 10000) + ylim(0, 100) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Total number of unique STs per serovar \n & database generated across programs") +
  ylab("Proportion of non-classified STs") +
theme(axis.text.y = element_text(size = 20)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 20, hjust = 1)) +
  ggtitle("Biplot using mean of indexes across serovars") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 35, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=25, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  geom_jitter(size = 2, width = 0.1, height = 0.1) +
  facet_wrap(~Serovars)
p4c

# plot 5
# Fig S13-R
p5c <- ggplot(d21, aes(x = total_alleles_genes_mean, y = Miscalls, color = program)) + xlim(0, 10000) + ylim(0, 100) +
  theme_bw() + 
  #theme(legend.position = "none") +
  xlab("Total number of unique alleles per serovar \n & database generated across programs") +
  ylab("Proportion of non-classified STs") +
theme(axis.text.y = element_text(size = 20)) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 20, hjust = 1)) +
  ggtitle("Biplot using mean of indexes across serovars") +
  scale_color_manual(name = "Programs (Treatments)", 
                    values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                               "black", "purple", "cyan", "steelblue")) +
  theme(plot.title = element_text(size = 35, face = "bold")) + 
  theme(strip.text.x = element_text(size = 20)) +
  theme(legend.title=element_text(size=25, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  geom_jitter(size = 2, width = 0.1, height = 0.1) +
  facet_wrap(~Serovars)
p5c
####################################################################################
```

Dinucleotide data by species

```{r, include = FALSE}

# enter the data

data1 <- read_csv('stringmlst_mlst_stats.csv')

# skim the data

skim(data1)

# replace NA with zeros

data2 <- data1 %>% replace(is.na(.), 0)

# transform potential extraneous characters into NA - hyphen and ?

data2 <- data2 %>%
                    mutate_all(na_if, "-") %>%
                        mutate_all(na_if, "?") 

# drop all ST equals zero
data2 <- data2 %>% replace(is.na(.), 0)
data3 <- data2 %>% filter(st != 0)

# after filter out NAs and ST numbered zero we have left 

exc <- dim(data2)[1] - dim(data3)[1]
exc

# percentage of excluded rows 

(exc/dim(data2)[1])*100

# 10.1% 

# check for NA

sum(is.na(data3))

# filter out saintpaul

data3 <- data3 %>% filter(serovar != "saintpaul")

####################################################################################
# enter dinucleotide information 

data4 <- read_csv('dinucleotide_stats.csv')

# merge that with data4

data5 <- left_join(data3, data4, on = "serovar")

# select species data for plotting 

data6 <- data5 %>% select(sra_id, species, dinucl, freq) %>% distinct() %>% drop_na()
# Fig S3-E
p1 <- ggplot(data6, aes(x = dinucl, y = freq)) + geom_boxplot() + 
  theme_bw() + 
  xlab("Dinucleotides (including 3 outliers)") +
  ylab("Proportion") +
  theme(axis.text.y = element_text(size = 30, face = "italic")) +
  theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 30, hjust = 1)) +
  theme(strip.text.x = element_text(size = 20, face = "italic")) +
  facet_wrap(~species)
p1

data7 <- data6 %>% filter(freq < 0.4) %>% drop_na()
# Fig S3-F
p2 <- ggplot(data7, aes(x = dinucl, y = freq)) + geom_boxplot() + 
  theme_bw() + 
  xlab("Dinucleotides (excluding 3 outliers)") +
  ylab("Proportion") +
  theme(axis.text.y = element_text(size = 30, face = "italic")) +
  theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 30, hjust = 1)) +
  theme(strip.text.x = element_text(size = 20, face = "italic")) +
  facet_wrap(~species)
p2

# transpose data to wide format to do a pca analysis

#new_data <- data8 %>% filter_all(any_vars(is.na(.))) # checking for NA anywhere
data8 <- data7 %>% spread(dinucl, freq) %>% drop_na()

# computer PCA
data9 <- column_to_rownames(data8, var = "sra_id")
training_data <- data9 %>% select(-species)

# Compute the mean and the sd of all numeric columns
a <- data9 %>%
  group_by(species) %>%
  summarise(across(
    .cols = is.numeric, 
    .fns = list(SD = sd), na.rm = TRUE, 
    .names = "{col}_{fn}"
    ))
a

res.pca <- prcomp(training_data, scale = TRUE)
plot1 <- fviz_eig(res.pca) # screeplot

eig.val <- get_eigenvalue(res.pca)
eig.val

# subset first 2 pca 

d1 <- data.frame(res.pca$x[, 1:2])
d2 <- rownames_to_column(d1, var = "sra_id")
d3 <- left_join(data8, d2, on = "sra_id")

# plot pca by species
# Fig 3-D
p3 <- ggplot(d3, aes(x = PC1, y = PC2, color = species)) + geom_jitter(size = 3) +
  theme_bw() + 
  xlab("PC1 (73.8%)") +
  ylab("PC2 (13.5%)") +
  scale_color_manual(name = "Species", values = c("gray", "orange", "darkblue", "darkgreen"),
                     guide = guide_legend(override.aes = list(size = 7))) +
  theme(axis.text.y = element_text(size = 30)) +
  theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, size = 30)) +
  theme(strip.text.x = element_text(size = 20, face = "italic")) +
  theme(legend.title=element_text(size=30, face = "bold"))+
  theme(legend.text=element_text(size=24, face = "italic"))  +
  ggtitle("D.") +
  #theme(legend.position = "none") +
  theme(plot.title = element_text(size = 45, face = "bold")) 
p3
```

Dinucleotide data by Salmonella enterica serovar

```{r, include = FALSE}

# enter the data

data1 <- read_csv('stringmlst_mlst_stats.csv')

# skim the data

skim(data1)

# replace NA with zeros

data2 <- data1 %>% replace(is.na(.), 0)

# transform potential extraneous characters into NA - hyphen and ?

data2 <- data2 %>%
                    mutate_all(na_if, "-") %>%
                        mutate_all(na_if, "?") 

# drop all ST equals zero
data2 <- data2 %>% replace(is.na(.), 0)
data3 <- data2 %>% filter(st != 0)

# after filter out NAs and ST numbered zero we have left 

exc <- dim(data2)[1] - dim(data3)[1]
exc

# percentage of excluded rows 

(exc/dim(data2)[1])*100

# 10.1% 

# check for NA

sum(is.na(data3))

# filter out saintpaul

data3 <- data3 %>% filter(serovar != "saintpaul")

####################################################################################
# enter dinucleotide information 

data4 <- read_csv('dinucleotide_stats.csv')

# merge that with data4

data5 <- left_join(data3, data4, on = "serovar")

data6 <- data5 %>% filter(serovar != "jejuni" & serovar != "saureus" & serovar != "listeria")

data6 <- data6 %>% mutate(Serovars = str_to_title(serovar))

# select species data for plotting 

data7 <- data6 %>% select(sra_id, Serovars, dinucl, freq) %>% distinct() %>% drop_na()
# Fig S4-J
p1 <- ggplot(data7, aes(x = dinucl, y = freq)) + geom_boxplot() +
  theme_bw() + 
  xlab("Dinucleotides (including 1 outlier)") +
  ylab("Proportion") +
  theme(axis.text.y = element_text(size = 20, face = "italic")) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 13, hjust = 1)) +
  theme(strip.text.x = element_text(size = 20, face = "bold")) +
  facet_wrap(~Serovars)
p1

data8 <- data7 %>% filter(freq < 0.5) %>% drop_na()
# Fig S4-K
p2 <- ggplot(data8, aes(x = dinucl, y = freq)) + geom_boxplot() + 
  theme_bw() + 
  xlab("Dinucleotides (excluding 1 outlier)") +
  ylab("Proportion") +
  theme(axis.text.y = element_text(size = 20, face = "italic")) +
  theme(axis.title.y = element_text(size = 25, face = "bold")) +
  theme(axis.title.x = element_text(size = 25, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 13, hjust = 1)) +
  theme(strip.text.x = element_text(size = 20, face = "bold")) +
  facet_wrap(~Serovars)
p2

# transpose data to wide format to do a pca analysis

#new_data <- data8 %>% filter_all(any_vars(is.na(.))) # checking for NA anywhere
data9 <- data8 %>% spread(dinucl, freq) %>% drop_na()

# computer PCA
data10 <- column_to_rownames(data9, var = "sra_id")
training_data <- data10 %>% select(-Serovars)

# Compute the mean and the sd of all numeric columns
a <- data10 %>%
  group_by(Serovars) %>%
  summarise(across(
    .cols = is.numeric, 
    .fns = list(SD = sd), na.rm = TRUE, 
    .names = "{col}_{fn}"
    ))
a

res.pca <- prcomp(training_data, scale = TRUE)
plot1 <- fviz_eig(res.pca) # screeplot

eig.val <- get_eigenvalue(res.pca)
eig.val

# subset first 2 pca 

d1 <- data.frame(res.pca$x[, 1:2])
d2 <- rownames_to_column(d1, var = "sra_id")
d3 <- left_join(data9, d2, on = "sra_id")

# plot pca by species
# Fig S4-L
p3 <- ggplot(d3, aes(x = PC1, y = PC2, color = Serovars)) + geom_jitter(size = 3) + 
  theme_bw() + 
  xlab("PC1 (44.3%)") +
  ylab("PC2 (25.3%)") +
  #scale_color_manual(name = "Serovars",
                  #   guide = guide_legend(override.aes = list(size = 5))) +
  theme(axis.text.y = element_text(size = 30, face = "italic")) +
  theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, size = 30, hjust = 1)) +
  theme(strip.text.x = element_text(size = 20, face = "italic")) +
  theme(legend.title=element_text(size=30, face = "bold"))+
  theme(legend.text=element_text(size=24)) 
p3


input <- read.csv("kmers_mapping.csv", stringsAsFactors = FALSE)
input$occurence_reads_log <- log(input$occurence_reads)
input$organism <- revalue(x = input$organism, 
                           c("jejuni" = "Campylobacter jejuni", "listeria" = "Listeria monocytogenes", 
                             "saureus" = "Staphylococcus aureus", "typhimurium" = "Salmonella Typhimurium"))
input$kmer <- c(1,2,3,4,5,6,7,8,9,10)
input2 <- read.csv("kmers_freqs.csv", stringsAsFactors = FALSE)
input2$freq <- (input2$common_kmers/input2$uniq_obs)*100
input2$organism <- revalue(x = input2$organism, 
                          c("jejuni" = "Campylobacter jejuni", "listeria" = "Listeria monocytogenes", 
                            "saureus" = "Staphylococcus aureus", "typhimurium" = "Salmonella Typhimurium"))
input2$kmer <- c(1,2,3,4,5,6,7,8,9,10)
input2

# Plot common kmers between stringMLST and raw reads (DSK)
# Fig S14-A
p_freqs <- ggplot(input2, aes(x=as.factor(kmer), y=freq, fill=as.factor(kmer))) + ylim(0, 10) + 
  facet_wrap(~organism, ncol = 5, strip.position = "bottom") + theme_bw() +
  geom_bar(stat="identity", position = position_dodge()) + facet_wrap(~organism) +
  scale_fill_manual(
    values = c("brown", "orange", "blue", "darkgreen", "red", "pink", 
               "black", "purple", "cyan", "steelblue")) +
  theme(legend.position = "none") +
  theme(strip.text.x = element_text(size = 30, face = "italic")) +
  theme(axis.text.y = element_text(size = 20)) +
  theme(axis.title.x = element_blank()) +
  theme(axis.title.y = element_text(size = 40, face = "bold")) +
  theme(axis.text.x = element_text(angle = 75, hjust=1, size = 20)) +
  theme(legend.title=element_text(size=35, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  labs(y = "Relative frequency of common kmers") + 
  scale_x_discrete(breaks = c("1","2","3","4","5","6","7","8","9","10"), 
                   labels=c("kmer_10","kmer_20","kmer_30","kmer_35","kmer_45","kmer_55","kmer_65","kmer_70","kmer_80","kmer_90"))
p_freqs

# skip kmer 10
input3 <- input2[input2$kmer != 1, ]     
# Fig S14-B
p_freqs <- ggplot(input3, aes(x=as.factor(kmer), y=freq, fill=as.factor(kmer))) + ylim(0, 0.1) + 
  facet_wrap(~organism, ncol = 5, strip.position = "bottom") + theme_bw() +
  geom_bar(stat="identity", position = position_dodge()) + facet_wrap(~organism) +
  scale_fill_manual(
    values = c("brown", "orange", "blue", "darkgreen", "red", "pink", 
               "black", "purple", "cyan", "steelblue")) +
  theme(legend.position = "none") +
  theme(strip.text.x = element_text(size = 30, face = "italic")) +
  theme(axis.text.y = element_text(size = 20)) +
  theme(axis.title.x = element_blank()) +
  theme(axis.title.y = element_text(size = 40, face = "bold")) +
  theme(axis.text.x = element_text(angle = 75, hjust=1, size = 20)) +
  theme(legend.title=element_text(size=35, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  labs(y = "Relative frequency of common kmers") + 
  scale_x_discrete(breaks = c("2","3","4","5","6","7","8","9","10"), 
                   labels=c("kmer_20","kmer_30","kmer_35","kmer_45","kmer_55","kmer_65","kmer_70","kmer_80","kmer_90"))
p_freqs





input <- read.csv("1000_fin.csv", stringsAsFactors = FALSE)
# - and 0 ST are NA
data1 <- input %>%
  mutate_all(na_if, "-") %>%
  mutate_all(na_if, "0") 
# merge columns program_batch into one  
data2 <- transform(data1, program_batch=paste(program, batch, sep="_"))
# convert program_batch row to column
data3 <- data2 %>%
  spread(program_batch, st)
data4 <- aggregate(. ~ sra_id + batch, data=data3,
                   FUN=function(x) x[!is.na(x)][1], na.action=na.pass)
drops <- c("program","batch")
data5 <- data4[ , !(names(data4) %in% drops)]
colnames(data5)[1] <- "sra_id"
data5$stringmlst_1000 <- as.character(data5$stringmlst_1000)
data5$mlst_1000 <- as.character(data5$mlst_1000)
data5[is.na(data5)] <- "-3"
# same 1, different 0
df <- transform(data5, MLST_vs_stringMLST_1000 = ifelse(data5$mlst_1000==data5$stringmlst_1000, 1, 0))
dts <- df %>%
  group_by(species) %>%
  do(data.frame(batch_1000=c(mean(.$MLST_vs_stringMLST_1000)*100)
  )
  )
# count NAs, differences 
newdata <- subset(data5, data5$species=="Salmonella enterica")
tab1 <- table(newdata$mlst_1000)
tab2 <- table(newdata$stringmlst_1000)
df_1 <- data.frame(species="Salmonella enterica", group=rep(c("mlst", "stringMLST"), times=c(length(tab1), length(tab2))), values=c(tab1, tab2), xval=as.numeric(c(names(tab1), names(tab2))))
df_1[is.na(df_1)] <- 0
newdata <- subset(data5, data5$species=="Campylobacter jejuni")
tab1 <- table(newdata$mlst_1000)
tab2 <- table(newdata$stringmlst_1000)
df_2 <- data.frame(species="Campylobacter jejuni", group=rep(c("mlst", "stringMLST"), times=c(length(tab1), length(tab2))), values=c(tab1, tab2), xval=as.numeric(c(names(tab1), names(tab2))))
df_2[is.na(df_2)] <- 0
newdata <- subset(data5, data5$species=="Listeria monocytogenes")
tab1 <- table(newdata$mlst_1000)
tab2 <- table(newdata$stringmlst_1000)
df_3 <- data.frame(species="Listeria monocytogenes", group=rep(c("mlst", "stringMLST"), times=c(length(tab1), length(tab2))), values=c(tab1, tab2), xval=as.numeric(c(names(tab1), names(tab2))))
df_3[is.na(df_3)] <- 0
newdata <- subset(data5, data5$species=="Staphylococcus aureus")
tab1 <- table(newdata$mlst_1000)
tab2 <- table(newdata$stringmlst_1000)
df_4 <- data.frame(species="Staphylococcus aureus", group=rep(c("mlst", "stringMLST"), times=c(length(tab1), length(tab2))), values=c(tab1, tab2), xval=as.numeric(c(names(tab1), names(tab2))))
df_4[is.na(df_4)] <- 0
a <- rbind(df_1, df_2)
b <- rbind(a, df_3)
c <- rbind(b, df_4)
# Fig S15-1000
p <- ggplot(c, aes(as.factor(xval), values, label = xval, fill=group)) + 
  geom_bar(position="dodge", stat="identity") + theme_bw() +
  facet_wrap(~species, scales="free") +
  labs(y = "Count", x = "ST") +
  theme(strip.text.x = element_text(size = 80, face = "italic")) +
  theme(axis.text.y = element_text(size = 80)) +
  theme(axis.title.x=element_blank()) +
  theme(axis.text.x = element_blank()) + 
  theme(axis.ticks.x = element_blank()) +
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +
  theme(axis.title.y = element_text(size = 90, face = "bold"))
  #theme(axis.text.x = element_text(angle = 0, hjust=0.5, size = 15)) +
  #theme(legend.title=element_text(size=85, face = "bold")) +
  #theme(legend.text=element_text(size=80))
p
input <- read.csv("100_fin.csv", stringsAsFactors = FALSE)
# - and 0 ST are NA
data1 <- input %>%
  mutate_all(na_if, "-") %>%
  mutate_all(na_if, "0") 
# merge columns program_batch into one  
data2 <- transform(data1, program_batch=paste(program, batch, sep="_"))
# convert program_batch row to column
data3 <- data2 %>%
  spread(program_batch, st)
data4 <- aggregate(. ~ sra_id + batch, data=data3,
                   FUN=function(x) x[!is.na(x)][1], na.action=na.pass)
drops <- c("program","batch")
data5 <- data4[ , !(names(data4) %in% drops)]
colnames(data5)[1] <- "sra_id"
data5$stringmlst_100 <- as.character(data5$stringmlst_100)
data5$mlst_100 <- as.character(data5$mlst_100)
data5[is.na(data5)] <- "-3"
# same 1, different 0
df <- transform(data5, MLST_vs_stringMLST_100 = ifelse(data5$mlst_100==data5$stringmlst_100, 1, 0))
dts <- df %>%
  group_by(species) %>%
  do(data.frame(batch_100=c(mean(.$MLST_vs_stringMLST_100)*100)
  )
  )
# count NAs, differences 
newdata <- subset(data5, data5$species=="Salmonella enterica")
tab1 <- table(newdata$mlst_100)
tab2 <- table(newdata$stringmlst_100)
df_1 <- data.frame(species="Salmonella enterica", group=rep(c("mlst", "stringMLST"), times=c(length(tab1), length(tab2))), values=c(tab1, tab2), xval=as.numeric(c(names(tab1), names(tab2))))
df_1[is.na(df_1)] <- 0
newdata <- subset(data5, data5$species=="Campylobacter jejuni")
tab1 <- table(newdata$mlst_100)
tab2 <- table(newdata$stringmlst_100)
df_2 <- data.frame(species="Campylobacter jejuni", group=rep(c("mlst", "stringMLST"), times=c(length(tab1), length(tab2))), values=c(tab1, tab2), xval=as.numeric(c(names(tab1), names(tab2))))
df_2[is.na(df_2)] <- 0
newdata <- subset(data5, data5$species=="Listeria monocytogenes")
tab1 <- table(newdata$mlst_100)
tab2 <- table(newdata$stringmlst_100)
df_3 <- data.frame(species="Listeria monocytogenes", group=rep(c("mlst", "stringMLST"), times=c(length(tab1), length(tab2))), values=c(tab1, tab2), xval=as.numeric(c(names(tab1), names(tab2))))
df_3[is.na(df_3)] <- 0
newdata <- subset(data5, data5$species=="Staphylococcus aureus")
tab1 <- table(newdata$mlst_100)
tab2 <- table(newdata$stringmlst_100)
df_4 <- data.frame(species="Staphylococcus aureus", group=rep(c("mlst", "stringMLST"), times=c(length(tab1), length(tab2))), values=c(tab1, tab2), xval=as.numeric(c(names(tab1), names(tab2))))
df_4[is.na(df_4)] <- 0
a <- rbind(df_1, df_2)
b <- rbind(a, df_3)
c <- rbind(b, df_4)
# Fig S15-100
p <- ggplot(c, aes(as.factor(xval), values, label = xval, fill=group)) + 
  geom_bar(position="dodge", stat="identity") + theme_bw() +
  facet_wrap(~species, scales="free") +
  labs(y = "Count", x = "ST") +
  theme(strip.text.x = element_text(size = 80, face = "italic")) +
  theme(axis.text.y = element_text(size = 80)) +
  theme(axis.title.x=element_blank()) +
  theme(axis.text.x = element_blank()) + 
  theme(axis.ticks.x = element_blank()) +
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +
  theme(axis.title.y = element_text(size = 90, face = "bold"))
  #theme(axis.text.x = element_text(angle = 0, hjust=0.5, size = 15)) +
  #theme(legend.title=element_text(size=85, face = "bold")) +
  #theme(legend.text=element_text(size=80))
p



input <- read.csv("kmers_mapping.csv", stringsAsFactors = FALSE)
input$occurence_reads_log <- log(input$occurence_reads)
input$organism <- revalue(x = input$organism, 
                           c("jejuni" = "Campylobacter jejuni", "listeria" = "Listeria monocytogenes", 
                             "saureus" = "Staphylococcus aureus", "typhimurium" = "Salmonella Typhimurium"))
input$kmer <- c(1,2,3,4,5,6,7,8,9,10)
# Plot mean of kmer occurrences in DSK output from raw reads
# Fig S1
p_occurence_mean <- ggplot(input, aes(x=as.factor(kmer), y=log(mean), fill=as.factor(kmer))) + ylim(0, 20) + 
  facet_wrap(~organism, ncol = 5, strip.position = "bottom") + theme_bw() +
  geom_bar(stat="identity", position = position_dodge()) + facet_wrap(~organism) +
  scale_fill_manual(
    values = c("brown", "orange", "blue", "darkgreen", "red", "pink", 
               "black", "purple", "cyan", "steelblue")) +
  theme(legend.position = "none") +
  theme(strip.text.x = element_text(size = 30, face = "italic")) +
  theme(axis.text.y = element_text(size = 20)) +
  theme(axis.title.x = element_blank()) +
  theme(axis.title.y = element_text(size = 40, face = "bold")) +
  theme(axis.text.x = element_text(angle = 75, hjust=1, size = 20)) +
  theme(legend.title=element_text(size=35, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  labs(y = "Log of mean of kmer occurence in raw reads") + 
  scale_x_discrete(breaks = c("1","2","3","4","5","6","7","8","9","10"), 
                   labels=c("kmer_10","kmer_20","kmer_30","kmer_35","kmer_45","kmer_55","kmer_65","kmer_70","kmer_80","kmer_90"))
p_occurence_mean





input <- read.csv("resource_stringmlst_mlst.csv", stringsAsFactors = FALSE)
# merge columns program_kmer into one
data2 <- transform(input, program_kmer=paste(program, kmer, sep="_"))
# convert program_kmer row to column
data3 <- data2 %>%
  gather(key, value, memory_max_bytes) %>%
  spread(program_kmer, value)
# data4 <- aggregate(data3[-1], list(data3$sra_id), FUN = max, na.rm = TRUE)
data4 <- aggregate(data3[7:ncol(data3)],data3[c("organism", "batch")], FUN=max, na.rm = TRUE)
drops <- c("program","kmer")
data5 <- data4[ , !(names(data4) %in% drops)]
dts <- data5 %>%
  group_by(organism, batch) %>%
  do(data.frame(mlst=c(mean(.$mlst_100)/1000000000),
                kmer_10=c(mean(.$stringmlst_10)/1000000000),
                kmer_20=c(mean(.$stringmlst_20)/1000000000),
                kmer_30=c(mean(.$stringmlst_30)/1000000000),
                kmer_35=c(mean(.$stringmlst_35)/1000000000),
                kmer_45=c(mean(.$stringmlst_45)/1000000000),
                kmer_55=c(mean(.$stringmlst_55)/1000000000),
                kmer_65=c(mean(.$stringmlst_65)/1000000000),
                kmer_70=c(mean(.$stringmlst_70)/1000000000),
                kmer_80=c(mean(.$stringmlst_80)/1000000000),
                kmer_90=c(mean(.$stringmlst_90)/1000000000)
  )
  )
dts <- dts[-1,]
DFtall <- dts %>% gather(key = kmer, value = Value, mlst,kmer_10:kmer_90)
DFtall$batch <- as.factor(DFtall$batch)
DFtall$organism <- revalue(x = DFtall$organism, 
                         c("typhimurium" = "Salmonella Typhimurium", "jejuni" = "Campylobacter jejuni", 
                           "listeria" = "Listeria monocytogenes", "saureus" = "Staphylococcus aureus"))
DFtall$kmer <- revalue(x = DFtall$kmer, 
                           c("mlst" = "MLST (standard)", "kmer_10" = "stringMLST (kmer_10)", "kmer_20" = "stringMLST (kmer_20)",
                             "kmer_30" = "stringMLST (kmer_30)", "kmer_35" = "stringMLST (kmer_35)", "kmer_45" = "stringMLST (kmer_45)",
                             "kmer_55" = "stringMLST (kmer_55)", "kmer_65" = "stringMLST (kmer_65)", "kmer_70" = "stringMLST (kmer_70)",
                             "kmer_80" = "stringMLST (kmer_80)", "kmer_90" = "stringMLST (kmer_90)"))

# Fig S2
p <- ggplot(data = DFtall, aes(x = kmer, y = Value, fill = kmer)) + geom_boxplot(position = "dodge") + ylim(0,35) +
  facet_wrap(~organism, ncol = 5) + theme_bw() +
  theme(legend.position = "none") +
  scale_fill_manual(
    values = c("gray", "orange", "blue", "darkgreen", "red", "pink", 
               "black", "purple", "cyan", "steelblue", "brown")) +
  theme(axis.text.y = element_text(size = 80),
        axis.text=element_text(size=80),
        axis.title.y=element_blank(),
        axis.title.x = element_text(size = 90, face = "bold"),
        strip.text.x = element_text(size = 80, face = "italic"),
        strip.text.y = element_text(size = 80)) +
  labs(y = "Memory usage in GBs") + coord_flip()
p




input <- read.csv("resource_stringmlst_mlst.csv", stringsAsFactors = FALSE)
# merge columns program_kmer into one
data2 <- transform(input, program_kmer=paste(program, kmer, sep="_"))
data2$time_avg_sec <- data2$time_sum_sec/200
# convert program_kmer row to column
data3 <- data2 %>%
  gather(key, value, time_avg_sec) %>%
  spread(program_kmer, value)
# data4 <- aggregate(data3[-1], list(data3$sra_id), FUN = max, na.rm = TRUE)
data4 <- aggregate(data3[8:ncol(data3)],data3[c("organism", "batch")], FUN=max, na.rm = TRUE)
drops <- c("program","kmer")
data5 <- data4[ , !(names(data4) %in% drops)]
dts <- data5 %>%
  group_by(organism, batch) %>%
  do(data.frame(mlst=c(mean(.$mlst_100)/60),
                kmer_10=c(mean(.$stringmlst_10)/60),
                kmer_20=c(mean(.$stringmlst_20)/60),
                kmer_30=c(mean(.$stringmlst_30)/60),
                kmer_35=c(mean(.$stringmlst_35)/60),
                kmer_45=c(mean(.$stringmlst_45)/60),
                kmer_55=c(mean(.$stringmlst_55)/60),
                kmer_65=c(mean(.$stringmlst_65)/60),
                kmer_70=c(mean(.$stringmlst_70)/60),
                kmer_80=c(mean(.$stringmlst_80)/60),
                kmer_90=c(mean(.$stringmlst_90)/60)
  )
  )
dts <- dts[-1,]
DFtall <- dts %>% gather(key = kmer, value = Value, mlst,kmer_10:kmer_90)
DFtall$batch <- as.factor(DFtall$batch)
DFtall$organism <- revalue(x = DFtall$organism, 
                           c("typhimurium" = "Salmonella Typhimurium", "jejuni" = "Campylobacter jejuni", 
                             "listeria" = "Listeria monocytogenes", "saureus" = "Staphylococcus aureus"))
DFtall$kmer <- revalue(x = DFtall$kmer, 
                       c("mlst" = "MLST (standard)", "kmer_10" = "stringMLST (kmer_10)", "kmer_20" = "stringMLST (kmer_20)",
                         "kmer_30" = "stringMLST (kmer_30)", "kmer_35" = "stringMLST (kmer_35)", "kmer_45" = "stringMLST (kmer_45)",
                         "kmer_55" = "stringMLST (kmer_55)", "kmer_65" = "stringMLST (kmer_65)", "kmer_70" = "stringMLST (kmer_70)",
                         "kmer_80" = "stringMLST (kmer_80)", "kmer_90" = "stringMLST (kmer_90)"))
# Fig 2
p <- ggplot(data = DFtall, aes(x = kmer, y = Value, fill = kmer)) + geom_boxplot(position = "dodge") + ylim(0,80) +
  facet_wrap(~organism, ncol = 5) + theme_bw() +
  theme(legend.position = "none") +
  scale_fill_manual(
    values = c("gray", "orange", "blue", "darkgreen", "red", "pink", 
               "black", "purple", "cyan", "steelblue", "brown")) +
  theme(axis.text.y = element_text(size = 80),
        axis.text=element_text(size=80),
        axis.title.y=element_blank(),
        axis.title.x = element_text(size = 90, face = "bold"),
        strip.text.x = element_text(size = 80, face = "italic"),
        strip.text.y = element_text(size = 80)) +
  labs(y = "Average runtime in minutes per genome across batches") + coord_flip()
p




# percentage_agreement.csv is without saintpaul
input <- read.csv("percentage_agreement.csv", stringsAsFactors = FALSE)
# - and 0 ST are NA
data1 <- input %>%
  mutate_all(na_if, "-") %>%
  mutate_all(na_if, "0") 
# merge columns program_kmer into one
data2 <- transform(data1, program_kmer=paste(program, kmer, sep="_"))
# convert program_kmer row to column
data3 <- data2 %>%
  gather(key, value, st) %>%
  spread(program_kmer, value)
data4 <- aggregate(. ~ sra_id + batch, data=data3,
                   FUN=function(x) x[!is.na(x)][1], na.action=na.pass)
drops <- c("program","kmer")
data5 <- data4[ , !(names(data4) %in% drops)]

colnames(data5)[1] <- "sra_id"
data5$mlst_100 <- as.character(data5$mlst_100)
data5$stringmlst_10 <- as.character(data5$stringmlst_10)
data5$stringmlst_20 <- as.character(data5$stringmlst_20)
data5$stringmlst_30 <- as.character(data5$stringmlst_30)
data5$stringmlst_35 <- as.character(data5$stringmlst_35)
data5$stringmlst_45 <- as.character(data5$stringmlst_45)
data5$stringmlst_55 <- as.character(data5$stringmlst_55)
data5$stringmlst_65 <- as.character(data5$stringmlst_65)
data5$stringmlst_70 <- as.character(data5$stringmlst_70)
data5$stringmlst_80 <- as.character(data5$stringmlst_80)
data5$stringmlst_90 <- as.character(data5$stringmlst_90)
data5[is.na(data5)] <- -3
# same 1, different 0
df <- transform(data5, MLST_vs_stringMLST_kmer_10 = ifelse(data5$mlst_100==data5$stringmlst_10, 1, 0),
                MLST_vs_stringMLST_kmer_20 = ifelse(data5$mlst_100==data5$stringmlst_20, 1, 0),
                MLST_vs_stringMLST_kmer_30 = ifelse(data5$mlst_100==data5$stringmlst_30, 1, 0),
                MLST_vs_stringMLST_kmer_35 = ifelse(data5$mlst_100==data5$stringmlst_35, 1, 0),
                MLST_vs_stringMLST_kmer_45 = ifelse(data5$mlst_100==data5$stringmlst_45, 1, 0),
                MLST_vs_stringMLST_kmer_55 = ifelse(data5$mlst_100==data5$stringmlst_55, 1, 0),
                MLST_vs_stringMLST_kmer_65 = ifelse(data5$mlst_100==data5$stringmlst_65, 1, 0),
                MLST_vs_stringMLST_kmer_70 = ifelse(data5$mlst_100==data5$stringmlst_70, 1, 0),
                MLST_vs_stringMLST_kmer_80 = ifelse(data5$mlst_100==data5$stringmlst_80, 1, 0),
                MLST_vs_stringMLST_kmer_90 = ifelse(data5$mlst_100==data5$stringmlst_90, 1, 0))
dts <- df %>%
  group_by(species, batch) %>%
  do(data.frame(kmer_10=c(mean(.$MLST_vs_stringMLST_kmer_10)*100),
                kmer_20=c(mean(.$MLST_vs_stringMLST_kmer_20)*100),
                kmer_30=c(mean(.$MLST_vs_stringMLST_kmer_30)*100),
                kmer_35=c(mean(.$MLST_vs_stringMLST_kmer_35)*100),
                kmer_45=c(mean(.$MLST_vs_stringMLST_kmer_45)*100),
                kmer_55=c(mean(.$MLST_vs_stringMLST_kmer_55)*100),
                kmer_65=c(mean(.$MLST_vs_stringMLST_kmer_65)*100),
                kmer_70=c(mean(.$MLST_vs_stringMLST_kmer_70)*100),
                kmer_80=c(mean(.$MLST_vs_stringMLST_kmer_80)*100),
                kmer_90=c(mean(.$MLST_vs_stringMLST_kmer_90)*100)
  )
  )
dts <- dts[-1,]
DFtall <- dts %>% gather(key = kmer, value = Value, kmer_10:kmer_90)
DFtall$batch <- as.factor(DFtall$batch)
kmer_35_avg_data <- DFtall %>%
  filter(kmer == "kmer_35") %>%
  group_by(species) %>%
  summarise(avg = mean(Value))
# Fig 6
# https://community.rstudio.com/t/help-with-making-plot-with-multiple-columns/50763
p <- ggplot(data = DFtall, aes(y = kmer, x = Value, fill = kmer)) + geom_boxplot(position = "dodge") + xlim(0,100) +
# p + scale_color_manual(breaks = c("1", "2", "3"), values=c("darkred", "darkblue", "darkgreen"))
  facet_wrap(~species) + theme_bw() +
  theme(legend.position = "none") +
  scale_fill_manual(
                     values = c("gray", "orange", "darkblue", "darkgreen", "red", "pink", 
                                "black", "purple", "cyan", "steelblue")) +
  geom_vline(data = kmer_35_avg_data, aes(xintercept = avg), linetype = "longdash", color = "black") +
  theme(strip.text.x = element_text(size = 80, face = "italic")) +
  theme(axis.text.y = element_text(size = 80)) +
  theme(axis.title.y = element_text(size = 90, face = "bold")) +
  theme(axis.title.x = element_text(size = 90, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust=1, size = 80)) +
  theme(legend.title=element_text(size=25, face = "bold")) +
  theme(legend.text=element_text(size=20)) +
  labs(x = "% Agreement", y = "MLST vs. stringMLST with different kmer values")
p










# create dataframe with non classified )classified as NA,-3) for 100 and 1000
x <- c("Campylobacter jejuni", "Listeria monocytogenes", "Salmonella enterica", "Staphylococcus aureus",
       "Campylobacter jejuni", "Listeria monocytogenes", "Salmonella enterica", "Staphylococcus aureus",
       "Campylobacter jejuni", "Listeria monocytogenes", "Salmonella enterica", "Staphylococcus aureus",
       "Campylobacter jejuni", "Listeria monocytogenes", "Salmonella enterica", "Staphylococcus aureus")
y <- c("MLST", "MLST", "MLST", "MLST", 
       "stringMLST", "stringMLST", "stringMLST", "stringMLST",
       "MLST", "MLST", "MLST", "MLST", 
       "stringMLST", "stringMLST", "stringMLST", "stringMLST")
z <- c("100 genomes", "100 genomes", "100 genomes", "100 genomes", 
       "100 genomes", "100 genomes", "100 genomes", "100 genomes",
       "1000 genomes", "1000 genomes", "1000 genomes", "1000 genomes", 
       "1000 genomes", "1000 genomes", "1000 genomes", "1000 genomes")
w <- c(12, 10, 12, 1, 
       1, 2, 2, 1,
       130, 115, 74, 56,
       36, 50, 25, 12)
d <- c(100, 100, 89, 100,
       100, 100, 100, 100,
       999, 996, 938, 978,
       1000, 999, 993, 1000
)
dts <- data.frame("species"=x, "program"=y, "batch"=z, "non"=w, "total"=d)
dts$percentage <- (dts$non/dts$total)*100
# Fig 7-B
p <- ggplot(dts, aes(x=percentage, y=species, fill=program)) + 
  geom_bar(position="dodge", stat='identity') + facet_wrap(~ batch, ncol=2) +
  theme_bw() + theme(legend.position = "none") +
  theme(axis.text.x = element_text(size = 60),
        axis.title.x = element_text(size = 60, face = "bold"),
        axis.text=element_text(size=60),
        axis.title.y=element_blank(),
        axis.text.y = element_text(size = 60, face = "italic"),
        strip.text.y = element_text(size = 60, face = "italic"),
        strip.text.x = element_text(size = 60)) +
  labs(x = "% of non-classified STs") + 
  scale_y_discrete(labels = c("Campylobacter jejuni" = "Campylobacter\njejuni", "Listeria monocytogenes" = "Listeria\nmonocytogenes",
                              "Salmonella enterica" = "Salmonella\nTyphimurium", "Staphylococcus aureus" = "Staphylococcus\naureus"))
p


input <- read.csv("runtime_usage.csv", stringsAsFactors = FALSE)
input

input$program <- revalue(x = input$program, 
                           c("mlst" = "MLST", "stringmlst" = "stringMLST"))
input$species <- revalue(x = input$species, 
                         c("Salmonella enterica" = "Salmonella Typhimurium"))

input$batch <- as.character(input$batch)
input$batch <- revalue(x = input$batch, c("100" = "100 genomes", "1000" = "1000 genomes"))
# Fig 7-A
p <- ggplot(input , aes(x=runtime_total_min, y=species, fill=program)) + 
  geom_bar(position="dodge", stat='identity') + facet_wrap(~ batch, ncol=2) +
  theme_bw() + theme(legend.position = "none") +
  theme(axis.text.x = element_text(size = 60),
        axis.title.x = element_text(size = 60, face = "bold"),
        axis.text=element_text(size=60),
        axis.title.y=element_blank(),
        axis.text.y = element_text(size = 60, face = "italic"),
        strip.text.y = element_text(size = 60, face = "italic"),
        strip.text.x = element_text(size = 60)) +
  labs(x = "Total runtime in minutes") + 
  scale_y_discrete(labels = c("Campylobacter jejuni" = "Campylobacter\njejuni", "Listeria monocytogenes" = "Listeria\nmonocytogenes",
                              "Salmonella Typhimurium" = "Salmonella\nTyphimurium", "Staphylococcus aureus" = "Staphylococcus\naureus"))
p



x <- c("Campylobacter jejuni", "Listeria monocytogenes", "Salmonella enterica", "Staphylococcus aureus")
y <- c(89, 92, 82, 100)
z <- c(90.6, 93.3, 87.3, 95.6)
dts <- data.frame("species"=x, "batch_100"=y, "batch_1000"=z)
DFtall <- dts %>% gather(key = batch, value = Value, batch_100:batch_1000)
DFtall$species <- revalue(x = DFtall$species,
                          c("Salmonella enterica" = "Salmonella Typhimurium"))
DFtall$batch <- revalue(x = DFtall$batch, c("batch_100" = "100 genomes", "batch_1000" = "1000 genomes"))
# https://community.rstudio.com/t/help-with-making-plot-with-multiple-columns/50763
# Fig 7-C
p <- ggplot(data = DFtall, aes(y = Value, x = batch, fill = batch)) +
  # p + scale_color_manual(breaks = c("1", "2", "3"), values=c("darkred", "darkblue", "darkgreen"))
  facet_wrap(~species, nrow = 1) + theme_bw() +
  geom_bar(position = 'dodge', stat='identity') +
  geom_text(size = 21, aes(label = scales::percent(Value, accuracy = 0.01, scale = 1)), position=position_dodge(width=0.7), vjust=-0.21) +
  theme(legend.position = "none") +
  theme(strip.text.x = element_text(size = 80, face = "italic")) +
  theme(axis.text.y = element_text(size = 80)) +
  theme(axis.title.x=element_blank(),) +
  theme(axis.title.y = element_text(size = 90, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust=0.5, size = 70)) +
  theme(legend.title=element_text(size=25, face = "bold")) +
  theme(legend.text=element_text(size=20)) + scale_x_discrete() +
  labs(y = "% Agreement")
p








# S17 Fig, analyses from 23 Salmonella serovars, percentage of miscalls
in1 <- read.csv("mlst_stringmlst_2293_sub.txt", stringsAsFactors = FALSE)
in2 <- read.csv("sra_id_serovar.txt", stringsAsFactors = FALSE)
input <- merge(in1, in2, by = "sra_id")
# - and 0 ST are NA
data1 <- input %>%
  mutate_all(na_if, "-") %>%
  mutate_all(na_if, "0") 
# merge columns program_batch into one  
data2 <- transform(data1, program_kmer=paste(program, kmer, sep="_"))
# convert program_batch row to column
data3 <- data2 %>%
  spread(program_kmer, st)
data4 <- aggregate(. ~ sra_id + species, data=data3,
                   FUN=function(x) x[!is.na(x)][1], na.action=na.pass)
drops <- c("program","kmer")
data5 <- data4[ , !(names(data4) %in% drops)]
colnames(data5)[1] <- "sra_id"
data5$mlst_100 <- as.character(data5$mlst_100)
data5$stringmlst_20 <- as.character(data5$stringmlst_20)
data5$stringmlst_30 <- as.character(data5$stringmlst_30)
data5$stringmlst_35 <- as.character(data5$stringmlst_35)
data5$stringmlst_40 <- as.character(data5$stringmlst_40)
data5$stringmlst_45 <- as.character(data5$stringmlst_45)
data5$stringmlst_50 <- as.character(data5$stringmlst_50)
data5$stringmlst_55 <- as.character(data5$stringmlst_55)
data5$stringmlst_60 <- as.character(data5$stringmlst_60)
data5$stringmlst_65 <- as.character(data5$stringmlst_65)
data5$stringmlst_70 <- as.character(data5$stringmlst_70)
data5$stringmlst_80 <- as.character(data5$stringmlst_80)
data5$stringmlst_90 <- as.character(data5$stringmlst_90)
data6 <- data5 %>% 
  group_by(species) %>% 
  summarise(mlst100 = sum(is.na(mlst_100)), 
            kmer20 = sum(is.na(stringmlst_20)),
            kmer30 = sum(is.na(stringmlst_30)),
            kmer35 = sum(is.na(stringmlst_35)),
            kmer40 = sum(is.na(stringmlst_40)),
            kmer45 = sum(is.na(stringmlst_45)),
            kmer50 = sum(is.na(stringmlst_50)),
            kmer55 = sum(is.na(stringmlst_55)),
            kmer60 = sum(is.na(stringmlst_60)),
            kmer65 = sum(is.na(stringmlst_65)),
            kmer70 = sum(is.na(stringmlst_70)),
            kmer80 = sum(is.na(stringmlst_80)),
            kmer90 = sum(is.na(stringmlst_90)),
            total = n()
            )
data7 <- data6 %>% 
  group_by(species) %>% 
  summarise(mlst = (mlst100/total)*100, 
            kmer_20 = (kmer20/total)*100,
            kmer_30 = (kmer30/total)*100,
            kmer_35 = (kmer35/total)*100,
            kmer_40 = (kmer40/total)*100,
            kmer_45 = (kmer45/total)*100,
            kmer_50 = (kmer50/total)*100,
            kmer_55 = (kmer55/total)*100,
            kmer_60 = (kmer60/total)*100,
            kmer_65 = (kmer65/total)*100,
            kmer_70 = (kmer70/total)*100,
            kmer_80 = (kmer80/total)*100,
            kmer_90 = (kmer90/total)*100
            )
data_long <- gather(data7, combination, percentage, mlst:kmer_90, factor_key=TRUE)
data_long
p <- ggplot(data_long , aes(x=combination, y=percentage)) + ylim(0, 100) + 
  geom_bar(position="dodge", stat='identity') + facet_wrap(~ species, ncol=4) +
  theme_bw() + theme(legend.position = "none") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 40),
        axis.title.y = element_text(size = 60),
        axis.text=element_text(size=40),
        axis.text.y = element_text(size = 40),
        strip.text.y = element_text(size = 60),
        strip.text.x = element_text(size = 60)) +
  theme(axis.line = element_line(colour = "black")) +
  labs(y = "% of non-classified STs")
p

# S17 Fig, analyses from 23 Salmonella serovars, percentage of agreement
data5[is.na(data5)] <- -3
df <- transform(data5, MLST_vs_stringMLST_kmer_20 = ifelse(data5$mlst_100==data5$stringmlst_20, 1, 0),
                MLST_vs_stringMLST_kmer_30 = ifelse(data5$mlst_100==data5$stringmlst_30, 1, 0),
                MLST_vs_stringMLST_kmer_35 = ifelse(data5$mlst_100==data5$stringmlst_35, 1, 0),
                MLST_vs_stringMLST_kmer_40 = ifelse(data5$mlst_100==data5$stringmlst_40, 1, 0),
                MLST_vs_stringMLST_kmer_45 = ifelse(data5$mlst_100==data5$stringmlst_45, 1, 0),
                MLST_vs_stringMLST_kmer_50 = ifelse(data5$mlst_100==data5$stringmlst_50, 1, 0),
                MLST_vs_stringMLST_kmer_55 = ifelse(data5$mlst_100==data5$stringmlst_55, 1, 0),
                MLST_vs_stringMLST_kmer_60 = ifelse(data5$mlst_100==data5$stringmlst_60, 1, 0),
                MLST_vs_stringMLST_kmer_65 = ifelse(data5$mlst_100==data5$stringmlst_65, 1, 0),
                MLST_vs_stringMLST_kmer_70 = ifelse(data5$mlst_100==data5$stringmlst_70, 1, 0),
                MLST_vs_stringMLST_kmer_80 = ifelse(data5$mlst_100==data5$stringmlst_80, 1, 0),
                MLST_vs_stringMLST_kmer_90 = ifelse(data5$mlst_100==data5$stringmlst_90, 1, 0))
dts <- df %>%
  group_by(species) %>%
  do(data.frame(kmer_20=c(mean(.$MLST_vs_stringMLST_kmer_20)*100),
                kmer_30=c(mean(.$MLST_vs_stringMLST_kmer_30)*100),
                kmer_35=c(mean(.$MLST_vs_stringMLST_kmer_35)*100),
                kmer_40=c(mean(.$MLST_vs_stringMLST_kmer_40)*100),
                kmer_45=c(mean(.$MLST_vs_stringMLST_kmer_45)*100),
                kmer_50=c(mean(.$MLST_vs_stringMLST_kmer_50)*100),
                kmer_55=c(mean(.$MLST_vs_stringMLST_kmer_55)*100),
                kmer_60=c(mean(.$MLST_vs_stringMLST_kmer_60)*100),
                kmer_65=c(mean(.$MLST_vs_stringMLST_kmer_65)*100),
                kmer_70=c(mean(.$MLST_vs_stringMLST_kmer_70)*100),
                kmer_80=c(mean(.$MLST_vs_stringMLST_kmer_80)*100),
                kmer_90=c(mean(.$MLST_vs_stringMLST_kmer_90)*100)
  )
  )
DFtall <- dts %>% gather(key = kmer, value = Value, kmer_20:kmer_90)
kmer_35_avg_data <- DFtall %>%
  filter(kmer == "kmer_35") %>%
  group_by(species) %>%
  summarise(avg = mean(Value))
# https://community.rstudio.com/t/help-with-making-plot-with-multiple-columns/50763
p <- ggplot(data = DFtall, aes(y = kmer, x = Value, fill = kmer)) + geom_boxplot(position = "dodge") + xlim(0,100) +
  # p + scale_color_manual(breaks = c("1", "2", "3"), values=c("darkred", "darkblue", "darkgreen"))
  facet_wrap(~species, ncol = 4) + theme_bw() +
  theme(legend.position = "none") +
  geom_vline(data = kmer_35_avg_data, aes(xintercept = avg), linetype = "longdash", color = "black") +
  theme(axis.title.x = element_text(size = 60),
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 40),
        axis.title.y = element_text(size = 60),
        axis.text=element_text(size=40),
        axis.text.y = element_text(size = 30),
        strip.text.y = element_text(size = 60),
        strip.text.x = element_text(size = 60)) +
  theme(axis.line = element_line(colour = "black")) +
  labs(x = "% agreement", y = "MLST vs. stringMLST with different kmer values")
p

# S17 Fig, analyses from 23 Salmonella serovars, S. saintpaul, percentage of miscalls
in1 <- read.csv("saintpaul_out_all_100_kmers.txt", stringsAsFactors = FALSE)
in2 <- read.csv("sra_id_serovar.txt", stringsAsFactors = FALSE)
input <- merge(in1, in2, by = "sra_id")
# - and 0 ST are NA
data1 <- input %>%
  mutate_all(na_if, "-") %>%
  mutate_all(na_if, "0") 
# merge columns program_batch into one  
data2 <- transform(data1, program_kmer=paste(program, kmer, sep="_"))
# convert program_batch row to column
data3 <- data2 %>%
  spread(program_kmer, st)
data4 <- aggregate(. ~ sra_id + species, data=data3,
                   FUN=function(x) x[!is.na(x)][1], na.action=na.pass)
drops <- c("program","kmer")
data5 <- data4[ , !(names(data4) %in% drops)]
colnames(data5)[1] <- "sra_id"
data5$mlst_100 <- as.character(data5$mlst_100)
data5$stringmlst_20 <- as.character(data5$stringmlst_20)
data5$stringmlst_30 <- as.character(data5$stringmlst_30)
data5$stringmlst_35 <- as.character(data5$stringmlst_35)
data5$stringmlst_40 <- as.character(data5$stringmlst_40)
data5$stringmlst_45 <- as.character(data5$stringmlst_45)
data5$stringmlst_50 <- as.character(data5$stringmlst_50)
data5$stringmlst_55 <- as.character(data5$stringmlst_55)
data5$stringmlst_60 <- as.character(data5$stringmlst_60)
data5$stringmlst_65 <- as.character(data5$stringmlst_65)
data5$stringmlst_70 <- as.character(data5$stringmlst_70)
data5$stringmlst_80 <- as.character(data5$stringmlst_80)
data5$stringmlst_90 <- as.character(data5$stringmlst_90)
data5$stringmlst_100 <- as.character(data5$stringmlst_100)
data5$stringmlst_120 <- as.character(data5$stringmlst_120)
data5$stringmlst_140 <- as.character(data5$stringmlst_140)
data5$stringmlst_160 <- as.character(data5$stringmlst_160)
data5$stringmlst_180 <- as.character(data5$stringmlst_180)
data5$stringmlst_200 <- as.character(data5$stringmlst_200)
data5$stringmlst_220 <- as.character(data5$stringmlst_220)
data5$stringmlst_240 <- as.character(data5$stringmlst_240)
data6 <- data5 %>% 
  group_by(species) %>% 
  summarise(mlst100 = sum(is.na(mlst_100)), 
            kmer20 = sum(is.na(stringmlst_20)),
            kmer30 = sum(is.na(stringmlst_30)),
            kmer35 = sum(is.na(stringmlst_35)),
            kmer40 = sum(is.na(stringmlst_40)),
            kmer45 = sum(is.na(stringmlst_45)),
            kmer50 = sum(is.na(stringmlst_50)),
            kmer55 = sum(is.na(stringmlst_55)),
            kmer60 = sum(is.na(stringmlst_60)),
            kmer65 = sum(is.na(stringmlst_65)),
            kmer70 = sum(is.na(stringmlst_70)),
            kmer80 = sum(is.na(stringmlst_80)),
            kmer90 = sum(is.na(stringmlst_90)),
            kmer100 = sum(is.na(stringmlst_100)),
            kmer120 = sum(is.na(stringmlst_120)),
            kmer140 = sum(is.na(stringmlst_140)),
            kmer160 = sum(is.na(stringmlst_160)),
            kmer180 = sum(is.na(stringmlst_180)),
            kmer200 = sum(is.na(stringmlst_200)),
            kmer220 = sum(is.na(stringmlst_220)),
            kmer240 = sum(is.na(stringmlst_240)),
            total = n()
  )
data7 <- data6 %>% 
  group_by(species) %>% 
  summarise(mlst = (mlst100/total)*100, 
            kmer_20 = (kmer20/total)*100,
            kmer_30 = (kmer30/total)*100,
            kmer_35 = (kmer35/total)*100,
            kmer_40 = (kmer40/total)*100,
            kmer_45 = (kmer45/total)*100,
            kmer_50 = (kmer50/total)*100,
            kmer_55 = (kmer55/total)*100,
            kmer_60 = (kmer60/total)*100,
            kmer_65 = (kmer65/total)*100,
            kmer_70 = (kmer70/total)*100,
            kmer_80 = (kmer80/total)*100,
            kmer_90 = (kmer90/total)*100,
            kmer_100 = (kmer100/total)*100,
            kmer_120 = (kmer120/total)*100,
            kmer_140 = (kmer140/total)*100,
            kmer_160 = (kmer160/total)*100,
            kmer_180 = (kmer180/total)*100,
            kmer_200 = (kmer200/total)*100,
            kmer_220 = (kmer220/total)*100,
            kmer_240 = (kmer240/total)*100
  )
out <- merge(input, data7, by = "species")
data_long <- gather(data7, combination, percentage, mlst:kmer_240, factor_key=TRUE)
data_long$species[data_long$species == "Saintpaul"] <- "S. Saintpaul"
p <- ggplot(data_long , aes(x=combination, y=percentage)) + ylim(0, 100) + 
  geom_bar(position="dodge", stat='identity') + facet_wrap(~ species, ncol=1) +
  theme_bw() + theme(legend.position = "none") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 80),
        axis.title.y = element_text(size = 90),
        axis.text=element_text(size=80),
        axis.text.y = element_text(size = 80),
        strip.text.y = element_text(size = 90),
        strip.text.x = element_text(size = 90)) +
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +
  labs(y = "% of non-classified STs")
p

# S17 Fig, analyses from 23 Salmonella serovars, percentage of agreement
# same 1, different 0
data5[is.na(data5)] <- -3
df <- transform(data5, MLST_vs_stringMLST_kmer_20 = ifelse(data5$mlst_100==data5$stringmlst_20, 1, 0),
                MLST_vs_stringMLST_kmer_30 = ifelse(data5$mlst_100==data5$stringmlst_30, 1, 0),
                MLST_vs_stringMLST_kmer_35 = ifelse(data5$mlst_100==data5$stringmlst_35, 1, 0),
                MLST_vs_stringMLST_kmer_40 = ifelse(data5$mlst_100==data5$stringmlst_40, 1, 0),
                MLST_vs_stringMLST_kmer_45 = ifelse(data5$mlst_100==data5$stringmlst_45, 1, 0),
                MLST_vs_stringMLST_kmer_50 = ifelse(data5$mlst_100==data5$stringmlst_50, 1, 0),
                MLST_vs_stringMLST_kmer_55 = ifelse(data5$mlst_100==data5$stringmlst_55, 1, 0),
                MLST_vs_stringMLST_kmer_60 = ifelse(data5$mlst_100==data5$stringmlst_60, 1, 0),
                MLST_vs_stringMLST_kmer_65 = ifelse(data5$mlst_100==data5$stringmlst_65, 1, 0),
                MLST_vs_stringMLST_kmer_70 = ifelse(data5$mlst_100==data5$stringmlst_70, 1, 0),
                MLST_vs_stringMLST_kmer_80 = ifelse(data5$mlst_100==data5$stringmlst_80, 1, 0),
                MLST_vs_stringMLST_kmer_90 = ifelse(data5$mlst_100==data5$stringmlst_90, 1, 0),
                MLST_vs_stringMLST_kmer_100 = ifelse(data5$mlst_100==data5$stringmlst_100, 1, 0),
                MLST_vs_stringMLST_kmer_120 = ifelse(data5$mlst_100==data5$stringmlst_120, 1, 0),
                MLST_vs_stringMLST_kmer_140 = ifelse(data5$mlst_100==data5$stringmlst_140, 1, 0),
                MLST_vs_stringMLST_kmer_160 = ifelse(data5$mlst_100==data5$stringmlst_160, 1, 0),
                MLST_vs_stringMLST_kmer_180 = ifelse(data5$mlst_100==data5$stringmlst_180, 1, 0),
                MLST_vs_stringMLST_kmer_200 = ifelse(data5$mlst_100==data5$stringmlst_200, 1, 0),
                MLST_vs_stringMLST_kmer_220 = ifelse(data5$mlst_100==data5$stringmlst_220, 1, 0),
                MLST_vs_stringMLST_kmer_240 = ifelse(data5$mlst_100==data5$stringmlst_240, 1, 0)
                )
dts <- df %>%
  group_by(species) %>%
  do(data.frame(kmer_20=c(mean(.$MLST_vs_stringMLST_kmer_20)*100),
                kmer_30=c(mean(.$MLST_vs_stringMLST_kmer_30)*100),
                kmer_35=c(mean(.$MLST_vs_stringMLST_kmer_35)*100),
                kmer_40=c(mean(.$MLST_vs_stringMLST_kmer_40)*100),
                kmer_45=c(mean(.$MLST_vs_stringMLST_kmer_45)*100),
                kmer_50=c(mean(.$MLST_vs_stringMLST_kmer_50)*100),
                kmer_55=c(mean(.$MLST_vs_stringMLST_kmer_55)*100),
                kmer_60=c(mean(.$MLST_vs_stringMLST_kmer_60)*100),
                kmer_65=c(mean(.$MLST_vs_stringMLST_kmer_65)*100),
                kmer_70=c(mean(.$MLST_vs_stringMLST_kmer_70)*100),
                kmer_80=c(mean(.$MLST_vs_stringMLST_kmer_80)*100),
                kmer_90=c(mean(.$MLST_vs_stringMLST_kmer_90)*100),
                kmer_100=c(mean(.$MLST_vs_stringMLST_kmer_100)*100),
                kmer_120=c(mean(.$MLST_vs_stringMLST_kmer_120)*100),
                kmer_140=c(mean(.$MLST_vs_stringMLST_kmer_140)*100),
                kmer_160=c(mean(.$MLST_vs_stringMLST_kmer_160)*100),
                kmer_180=c(mean(.$MLST_vs_stringMLST_kmer_180)*100),
                kmer_200=c(mean(.$MLST_vs_stringMLST_kmer_200)*100),
                kmer_220=c(mean(.$MLST_vs_stringMLST_kmer_220)*100),
                kmer_240=c(mean(.$MLST_vs_stringMLST_kmer_240)*100)
  )
  )
DFtall <- dts %>% gather(key = kmer, value = Value, kmer_20:kmer_240)
DFtall$species[DFtall$species == "Saintpaul"] <- "S. Saintpaul"
kmer_35_avg_data <- DFtall %>%
  filter(kmer == "kmer_35") %>%
  group_by(species) %>%
  summarise(avg = mean(Value))
level_order <- c("kmer_20", "kmer_30", "kmer_35", "kmer_40", "kmer_45", "kmer_50", "kmer_55", "kmer_60", "kmer_65", "kmer_70", "kmer_80", "kmer_90", "kmer_100", "kmer_120", "kmer_140", "kmer_160", "kmer_180", "kmer_200", "kmer_220", "kmer_240")
# https://community.rstudio.com/t/help-with-making-plot-with-multiple-columns/50763
p <- ggplot(data = DFtall, aes(y = factor(kmer, level = level_order), x = Value, fill = kmer)) + geom_boxplot(position = "dodge") + xlim(0,100) +
  # p + scale_color_manual(breaks = c("1", "2", "3"), values=c("darkred", "darkblue", "darkgreen"))
  facet_wrap(~species, ncol = 4) + theme_bw() + 
  theme(legend.position = "none") +
  geom_vline(data = kmer_35_avg_data, aes(xintercept = avg), linetype = "longdash", color = "black") +
  theme(axis.title.x = element_text(size = 60),
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 40),
        axis.title.y = element_text(size = 60),
        axis.text=element_text(size=40),
        axis.text.y = element_text(size = 30),
        strip.text.y = element_text(size = 60),
        strip.text.x = element_text(size = 60)) +
  theme(axis.line = element_line(colour = "black")) +
  labs(x = "% agreement", y = "MLST vs. stringMLST with different kmer values")
p





# S17 Fig, analyses from 14 pathogens, percentage of miscalls
input <- read.csv("stringmlst_mlst_12313_all.txt", stringsAsFactors = FALSE)
# - and 0 ST are NA
data1 <- input %>%
  mutate_all(na_if, "-") %>%
  mutate_all(na_if, "0") 
# merge columns program_batch into one  
data2 <- transform(data1, program_kmer=paste(program, kmer, sep="_"))
# convert program_batch row to column
data3 <- data2 %>%
  spread(program_kmer, st)
data4 <- aggregate(. ~ sra_id + species, data=data3,
                   FUN=function(x) x[!is.na(x)][1], na.action=na.pass)
drops <- c("program","kmer")
data5 <- data4[ , !(names(data4) %in% drops)]
colnames(data5)[1] <- "sra_id"
data5$mlst_100 <- as.character(data5$mlst_100)
data5$stringmlst_20 <- as.character(data5$stringmlst_20)
data5$stringmlst_30 <- as.character(data5$stringmlst_30)
data5$stringmlst_35 <- as.character(data5$stringmlst_35)
data5$stringmlst_45 <- as.character(data5$stringmlst_45)
data5$stringmlst_55 <- as.character(data5$stringmlst_55)
data5$stringmlst_65 <- as.character(data5$stringmlst_65)
data5$stringmlst_70 <- as.character(data5$stringmlst_70)
data5$stringmlst_80 <- as.character(data5$stringmlst_80)
data5$stringmlst_90 <- as.character(data5$stringmlst_90)
data6 <- data5 %>% 
  group_by(species) %>% 
  summarise(mlst100 = sum(is.na(mlst_100)),
            kmer20 = sum(is.na(stringmlst_20)),
            kmer30 = sum(is.na(stringmlst_30)),
            kmer35 = sum(is.na(stringmlst_35)),
            kmer45 = sum(is.na(stringmlst_45)),
            kmer55 = sum(is.na(stringmlst_55)),
            kmer65 = sum(is.na(stringmlst_65)),
            kmer70 = sum(is.na(stringmlst_70)),
            kmer80 = sum(is.na(stringmlst_80)),
            kmer90 = sum(is.na(stringmlst_90)),
            total = n()
            )
data7 <- data6 %>% 
  group_by(species) %>% 
  summarise(mlst = (mlst100/total)*100,
            kmer_20 = (kmer20/total)*100,
            kmer_30 = (kmer30/total)*100,
            kmer_35 = (kmer35/total)*100,
            kmer_45 = (kmer45/total)*100,
            kmer_55 = (kmer55/total)*100,
            kmer_65 = (kmer65/total)*100,
            kmer_70 = (kmer70/total)*100,
            kmer_80 = (kmer80/total)*100,
            kmer_90 = (kmer90/total)*100
            )
data_long <- gather(data7, combination, percentage, mlst:kmer_90, factor_key=TRUE)
data_long
p <- ggplot(data_long , aes(x=combination, y=percentage)) + ylim(0, 100) + 
  geom_bar(position="dodge", stat='identity') + facet_wrap(~ species, ncol=4) +
  theme_bw() + theme(legend.position = "none") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 40),
        axis.title.y = element_text(size = 60),
        axis.text=element_text(size=40),
        axis.text.y = element_text(size = 40),
        strip.text.y = element_text(size = 60),
        strip.text.x = element_text(size = 60)) +
  theme(axis.line = element_line(colour = "black")) +
  labs(y = "% of non-classified STs")
p

# S17 Fig, analyses from 14 pathogens, percentage of agreement
# same 1, different 0
data5[is.na(data5)] <- -3
df <- transform(data5, MLST_vs_stringMLST_kmer_20 = ifelse(data5$mlst_100==data5$stringmlst_20, 1, 0),
                MLST_vs_stringMLST_kmer_30 = ifelse(data5$mlst_100==data5$stringmlst_30, 1, 0),
                MLST_vs_stringMLST_kmer_35 = ifelse(data5$mlst_100==data5$stringmlst_35, 1, 0),
                MLST_vs_stringMLST_kmer_45 = ifelse(data5$mlst_100==data5$stringmlst_45, 1, 0),
                MLST_vs_stringMLST_kmer_55 = ifelse(data5$mlst_100==data5$stringmlst_55, 1, 0),
                MLST_vs_stringMLST_kmer_65 = ifelse(data5$mlst_100==data5$stringmlst_65, 1, 0),
                MLST_vs_stringMLST_kmer_70 = ifelse(data5$mlst_100==data5$stringmlst_70, 1, 0),
                MLST_vs_stringMLST_kmer_80 = ifelse(data5$mlst_100==data5$stringmlst_80, 1, 0),
                MLST_vs_stringMLST_kmer_90 = ifelse(data5$mlst_100==data5$stringmlst_90, 1, 0))
dts <- df %>%
  group_by(species) %>%
  do(data.frame(kmer_20=c(mean(.$MLST_vs_stringMLST_kmer_20)*100),
                kmer_30=c(mean(.$MLST_vs_stringMLST_kmer_30)*100),
                kmer_35=c(mean(.$MLST_vs_stringMLST_kmer_35)*100),
                kmer_45=c(mean(.$MLST_vs_stringMLST_kmer_45)*100),
                kmer_55=c(mean(.$MLST_vs_stringMLST_kmer_55)*100),
                kmer_65=c(mean(.$MLST_vs_stringMLST_kmer_65)*100),
                kmer_70=c(mean(.$MLST_vs_stringMLST_kmer_70)*100),
                kmer_80=c(mean(.$MLST_vs_stringMLST_kmer_80)*100),
                kmer_90=c(mean(.$MLST_vs_stringMLST_kmer_90)*100)
  )
  )
DFtall <- dts %>% gather(key = kmer, value = Value, kmer_20:kmer_90)
kmer_35_avg_data <- DFtall %>%
  filter(kmer == "kmer_35") %>%
  group_by(species) %>%
  summarise(avg = mean(Value))
# https://community.rstudio.com/t/help-with-making-plot-with-multiple-columns/50763
p <- ggplot(data = DFtall, aes(y = kmer, x = Value, fill = kmer)) + geom_boxplot(position = "dodge") + xlim(0,100) +
  # p + scale_color_manual(breaks = c("1", "2", "3"), values=c("darkred", "darkblue", "darkgreen"))
  facet_wrap(~species, ncol = 4) + theme_bw() +
  theme(legend.position = "none") +
  geom_vline(data = kmer_35_avg_data, aes(xintercept = avg), linetype = "longdash", color = "black") +
  theme(axis.title.x = element_text(size = 60),
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 40),
        axis.title.y = element_text(size = 60),
        axis.text=element_text(size=40),
        axis.text.y = element_text(size = 30),
        strip.text.y = element_text(size = 60),
        strip.text.x = element_text(size = 60)) +
  theme(axis.line = element_line(colour = "black")) +
  labs(x = "% agreement", y = "MLST vs. stringMLST with different kmer values")
p



# S17 Fig, analyses from 14 pathogens, Enterococcus_faecalis 100, percentage of miscalls
input <- read.csv("Enterococcus_faecalis_all_92.txt", stringsAsFactors = FALSE)
# - and 0 ST are NA
data1 <- input %>%
  mutate_all(na_if, "-") %>%
  mutate_all(na_if, "0") 
# merge columns program_batch into one  
data2 <- transform(data1, program_kmer=paste(program, kmer, sep="_"))
# convert program_batch row to column
data3 <- data2 %>%
  spread(program_kmer, st)
data4 <- aggregate(. ~ sra_id + species, data=data3,
                   FUN=function(x) x[!is.na(x)][1], na.action=na.pass)
drops <- c("program","kmer")
data5 <- data4[ , !(names(data4) %in% drops)]
colnames(data5)[1] <- "sra_id"
data5$mlst_100 <- as.character(data5$mlst_100)
data5$stringmlst_20 <- as.character(data5$stringmlst_20)
data5$stringmlst_30 <- as.character(data5$stringmlst_30)
data5$stringmlst_35 <- as.character(data5$stringmlst_35)
data5$stringmlst_45 <- as.character(data5$stringmlst_45)
data5$stringmlst_55 <- as.character(data5$stringmlst_55)
data5$stringmlst_65 <- as.character(data5$stringmlst_65)
data5$stringmlst_70 <- as.character(data5$stringmlst_70)
data5$stringmlst_80 <- as.character(data5$stringmlst_80)
data5$stringmlst_90 <- as.character(data5$stringmlst_90)
data5$stringmlst_100 <- as.character(data5$stringmlst_100)
data5$stringmlst_110 <- as.character(data5$stringmlst_110)
data5$stringmlst_120 <- as.character(data5$stringmlst_120)
data5$stringmlst_140 <- as.character(data5$stringmlst_140)
data6 <- data5 %>% 
  group_by(species) %>% 
  summarise(mlst = sum(is.na(mlst_100)),
            kmer20 = sum(is.na(stringmlst_20)),
            kmer30 = sum(is.na(stringmlst_30)),
            kmer35 = sum(is.na(stringmlst_35)),
            kmer45 = sum(is.na(stringmlst_45)),
            kmer55 = sum(is.na(stringmlst_55)),
            kmer65 = sum(is.na(stringmlst_65)),
            kmer70 = sum(is.na(stringmlst_70)),
            kmer80 = sum(is.na(stringmlst_80)),
            kmer90 = sum(is.na(stringmlst_90)),
            kmer100 = sum(is.na(stringmlst_100)),
            kmer110 = sum(is.na(stringmlst_110)),
            kmer120 = sum(is.na(stringmlst_120)),
            kmer140 = sum(is.na(stringmlst_140)),
            total = n()
  )
data7 <- data6 %>% 
  group_by(species) %>% 
  summarise(mlst = (mlst/total)*100,
            kmer_20 = (kmer20/total)*100,
            kmer_30 = (kmer30/total)*100,
            kmer_35 = (kmer35/total)*100,
            kmer_45 = (kmer45/total)*100,
            kmer_55 = (kmer55/total)*100,
            kmer_65 = (kmer65/total)*100,
            kmer_70 = (kmer70/total)*100,
            kmer_80 = (kmer80/total)*100,
            kmer_90 = (kmer90/total)*100,
            kmer_100 = (kmer100/total)*100,
            kmer_110 = (kmer110/total)*100,
            kmer_120 = (kmer120/total)*100,
            kmer_140 = (kmer140/total)*100
  )
data_long <- gather(data7, combination, percentage, mlst:kmer_140, factor_key=TRUE)
data_long$species[data_long$species == "Enterococcus_faecalis"] <- "E. faecalis"
p <- ggplot(data_long , aes(x=combination, y=percentage)) + ylim(0, 100) + 
  geom_bar(position="dodge", stat='identity') + facet_wrap(~ species, ncol=1) +
  theme_bw() + theme(legend.position = "none") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 80),
        axis.title.y = element_text(size = 90),
        axis.text=element_text(size=80),
        axis.text.y = element_text(size = 80),
        strip.text.y = element_text(size = 90),
        strip.text.x = element_text(size = 90)) +
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +
  labs(y = "% of non-classified STs")
p

# S17 Fig, analyses from 14 pathogens, Enterococcus_faecalis 100, percentage of agreement
# same 1, different 0
input <- read.csv("Enterococcus_faecalis_all_92.txt", stringsAsFactors = FALSE)
# - and 0 ST are NA
data1 <- input %>%
  mutate_all(na_if, "-") %>%
  mutate_all(na_if, "0") 
# merge columns program_batch into one  
data2 <- transform(data1, program_kmer=paste(program, kmer, sep="_"))
# convert program_batch row to column
data3 <- data2 %>%
  spread(program_kmer, st)
data4 <- aggregate(. ~ sra_id + species, data=data3,
                   FUN=function(x) x[!is.na(x)][1], na.action=na.pass)
drops <- c("program","kmer")
data5 <- data4[ , !(names(data4) %in% drops)]
colnames(data5)[1] <- "sra_id"
data5$mlst_100 <- as.character(data5$mlst_100)
data5$stringmlst_20 <- as.character(data5$stringmlst_20)
data5$stringmlst_30 <- as.character(data5$stringmlst_30)
data5$stringmlst_35 <- as.character(data5$stringmlst_35)
data5$stringmlst_45 <- as.character(data5$stringmlst_45)
data5$stringmlst_55 <- as.character(data5$stringmlst_55)
data5$stringmlst_65 <- as.character(data5$stringmlst_65)
data5$stringmlst_70 <- as.character(data5$stringmlst_70)
data5$stringmlst_80 <- as.character(data5$stringmlst_80)
data5$stringmlst_90 <- as.character(data5$stringmlst_90)
data5$stringmlst_100 <- as.character(data5$stringmlst_100)
data5$stringmlst_110 <- as.character(data5$stringmlst_110)
data5$stringmlst_120 <- as.character(data5$stringmlst_120)
data5$stringmlst_140 <- as.character(data5$stringmlst_140)
data5[is.na(data5)] <- -3
df <- transform(data5, MLST_vs_stringMLST_kmer_20 = ifelse(data5$mlst_100==data5$stringmlst_20, 1, 0),
                MLST_vs_stringMLST_kmer_30 = ifelse(data5$mlst_100==data5$stringmlst_30, 1, 0),
                MLST_vs_stringMLST_kmer_35 = ifelse(data5$mlst_100==data5$stringmlst_35, 1, 0),
                MLST_vs_stringMLST_kmer_45 = ifelse(data5$mlst_100==data5$stringmlst_45, 1, 0),
                MLST_vs_stringMLST_kmer_55 = ifelse(data5$mlst_100==data5$stringmlst_55, 1, 0),
                MLST_vs_stringMLST_kmer_65 = ifelse(data5$mlst_100==data5$stringmlst_65, 1, 0),
                MLST_vs_stringMLST_kmer_70 = ifelse(data5$mlst_100==data5$stringmlst_70, 1, 0),
                MLST_vs_stringMLST_kmer_80 = ifelse(data5$mlst_100==data5$stringmlst_80, 1, 0),
                MLST_vs_stringMLST_kmer_90 = ifelse(data5$mlst_100==data5$stringmlst_90, 1, 0),
                MLST_vs_stringMLST_kmer_100 = ifelse(data5$mlst_100==data5$stringmlst_100, 1, 0),
                MLST_vs_stringMLST_kmer_110 = ifelse(data5$mlst_100==data5$stringmlst_110, 1, 0),
                MLST_vs_stringMLST_kmer_120 = ifelse(data5$mlst_100==data5$stringmlst_120, 1, 0),
                MLST_vs_stringMLST_kmer_140 = ifelse(data5$mlst_100==data5$stringmlst_140, 1, 0))
dts <- df %>%
  group_by(species) %>%
  do(data.frame(kmer_20=c(mean(.$MLST_vs_stringMLST_kmer_20)*100),
                kmer_30=c(mean(.$MLST_vs_stringMLST_kmer_30)*100),
                kmer_35=c(mean(.$MLST_vs_stringMLST_kmer_35)*100),
                kmer_45=c(mean(.$MLST_vs_stringMLST_kmer_45)*100),
                kmer_55=c(mean(.$MLST_vs_stringMLST_kmer_55)*100),
                kmer_65=c(mean(.$MLST_vs_stringMLST_kmer_65)*100),
                kmer_70=c(mean(.$MLST_vs_stringMLST_kmer_70)*100),
                kmer_80=c(mean(.$MLST_vs_stringMLST_kmer_80)*100),
                kmer_90=c(mean(.$MLST_vs_stringMLST_kmer_90)*100),
                kmer_100=c(mean(.$MLST_vs_stringMLST_kmer_100)*100),
                kmer_110=c(mean(.$MLST_vs_stringMLST_kmer_110)*100),
                kmer_120=c(mean(.$MLST_vs_stringMLST_kmer_120)*100),
                kmer_140=c(mean(.$MLST_vs_stringMLST_kmer_140)*100)
  ))
DFtall <- dts %>% gather(key = kmer, value = Value, kmer_20:kmer_140)
DFtall$species[DFtall$species == "Enterococcus_faecalis"] <- "E. faecalis"
kmer_35_avg_data <- DFtall %>%
  filter(kmer == "kmer_35") %>%
  group_by(species) %>%
  summarise(avg = mean(Value))
level_order <- c("kmer_20", "kmer_30", "kmer_35", "kmer_45", "kmer_55", "kmer_65", "kmer_70", "kmer_80", "kmer_90", "kmer_100", "kmer_110", "kmer_120", "kmer_140")
# https://community.rstudio.com/t/help-with-making-plot-with-multiple-columns/50763
p <- ggplot(data = DFtall, aes(y = factor(kmer, level = level_order), x = Value, fill = kmer)) + geom_boxplot(position = "dodge") + xlim(0,100) +
  # p + scale_color_manual(breaks = c("1", "2", "3"), values=c("darkred", "darkblue", "darkgreen"))
  facet_wrap(~species, ncol = 4) + theme_bw() +
  theme(legend.position = "none") +
  geom_vline(data = kmer_35_avg_data, aes(xintercept = avg), linetype = "longdash", color = "black") +
  theme(axis.title.x = element_text(size = 60),
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 40),
        axis.title.y = element_text(size = 60),
        axis.text=element_text(size=40),
        axis.text.y = element_text(size = 30),
        strip.text.y = element_text(size = 60),
        strip.text.x = element_text(size = 60)) +
  theme(axis.line = element_line(colour = "black")) +
  labs(x = "% agreement", y = "MLST vs. stringMLST with different kmer values")
p



# S17 Fig, analyses from 14 pathogens, Enterococcus_faecium, percentage of miscalls
input <- read.csv("Enterococcus_faecium_all_913.txt", stringsAsFactors = FALSE)
# - and 0 ST are NA
data1 <- input %>%
  mutate_all(na_if, "-") %>%
  mutate_all(na_if, "0") 
# merge columns program_batch into one  
data2 <- transform(data1, program_kmer=paste(program, kmer, sep="_"))
# convert program_batch row to column
data3 <- data2 %>%
  spread(program_kmer, st)
data4 <- aggregate(. ~ sra_id + species, data=data3,
                   FUN=function(x) x[!is.na(x)][1], na.action=na.pass)
drops <- c("program","kmer")
data5 <- data4[ , !(names(data4) %in% drops)]
colnames(data5)[1] <- "sra_id"
data5$mlst_100 <- as.character(data5$mlst_100)
data5$stringmlst_20 <- as.character(data5$stringmlst_20)
data5$stringmlst_30 <- as.character(data5$stringmlst_30)
data5$stringmlst_35 <- as.character(data5$stringmlst_35)
data5$stringmlst_45 <- as.character(data5$stringmlst_45)
data5$stringmlst_55 <- as.character(data5$stringmlst_55)
data5$stringmlst_65 <- as.character(data5$stringmlst_65)
data5$stringmlst_70 <- as.character(data5$stringmlst_70)
data5$stringmlst_80 <- as.character(data5$stringmlst_80)
data5$stringmlst_90 <- as.character(data5$stringmlst_90)
data5$stringmlst_100 <- as.character(data5$stringmlst_100)
data5$stringmlst_110 <- as.character(data5$stringmlst_110)
data5$stringmlst_120 <- as.character(data5$stringmlst_120)
data5$stringmlst_140 <- as.character(data5$stringmlst_140)
data6 <- data5 %>% 
  group_by(species) %>% 
  summarise(mlst100 = sum(is.na(mlst_100)),
            kmer20 = sum(is.na(stringmlst_20)),
            kmer30 = sum(is.na(stringmlst_30)),
            kmer35 = sum(is.na(stringmlst_35)),
            kmer45 = sum(is.na(stringmlst_45)),
            kmer55 = sum(is.na(stringmlst_55)),
            kmer65 = sum(is.na(stringmlst_65)),
            kmer70 = sum(is.na(stringmlst_70)),
            kmer80 = sum(is.na(stringmlst_80)),
            kmer90 = sum(is.na(stringmlst_90)),
            kmer100 = sum(is.na(stringmlst_100)),
            kmer110 = sum(is.na(stringmlst_110)),
            kmer120 = sum(is.na(stringmlst_120)),
            kmer140 = sum(is.na(stringmlst_140)),
            total = n()
  )
data7 <- data6 %>% 
  group_by(species) %>% 
  summarise(mlst = (mlst100/total)*100,
            kmer_20 = (kmer20/total)*100,
            kmer_30 = (kmer30/total)*100,
            kmer_35 = (kmer35/total)*100,
            kmer_45 = (kmer45/total)*100,
            kmer_55 = (kmer55/total)*100,
            kmer_65 = (kmer65/total)*100,
            kmer_70 = (kmer70/total)*100,
            kmer_80 = (kmer80/total)*100,
            kmer_90 = (kmer90/total)*100,
            kmer_100 = (kmer100/total)*100,
            kmer_110 = (kmer110/total)*100,
            kmer_120 = (kmer120/total)*100,
            kmer_140 = (kmer140/total)*100
  )
data_long <- gather(data7, combination, percentage, mlst:kmer_140, factor_key=TRUE)
data_long$species[data_long$species == "Enterococcus_faecium"] <- "E. faecium"
p <- ggplot(data_long , aes(x=combination, y=percentage)) + ylim(0, 100) + 
  geom_bar(position="dodge", stat='identity') + facet_wrap(~ species, ncol=1) +
  theme_bw() + theme(legend.position = "none") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 80),
        axis.title.y = element_text(size = 90),
        axis.text=element_text(size=80),
        axis.text.y = element_text(size = 80),
        strip.text.y = element_text(size = 90),
        strip.text.x = element_text(size = 90)) +
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +
  labs(y = "% of non-classified STs")
p

# S17 Fig, analyses from 14 pathogens, Enterococcus_faecium, percentage of agreement
# same 1, different 0
data5[is.na(data5)] <- -3
df <- transform(data5, MLST_vs_stringMLST_kmer_20 = ifelse(data5$mlst_100==data5$stringmlst_20, 1, 0),
                MLST_vs_stringMLST_kmer_30 = ifelse(data5$mlst_100==data5$stringmlst_30, 1, 0),
                MLST_vs_stringMLST_kmer_35 = ifelse(data5$mlst_100==data5$stringmlst_35, 1, 0),
                MLST_vs_stringMLST_kmer_45 = ifelse(data5$mlst_100==data5$stringmlst_45, 1, 0),
                MLST_vs_stringMLST_kmer_55 = ifelse(data5$mlst_100==data5$stringmlst_55, 1, 0),
                MLST_vs_stringMLST_kmer_65 = ifelse(data5$mlst_100==data5$stringmlst_65, 1, 0),
                MLST_vs_stringMLST_kmer_70 = ifelse(data5$mlst_100==data5$stringmlst_70, 1, 0),
                MLST_vs_stringMLST_kmer_80 = ifelse(data5$mlst_100==data5$stringmlst_80, 1, 0),
                MLST_vs_stringMLST_kmer_90 = ifelse(data5$mlst_100==data5$stringmlst_90, 1, 0),
                MLST_vs_stringMLST_kmer_100 = ifelse(data5$mlst_100==data5$stringmlst_100, 1, 0),
                MLST_vs_stringMLST_kmer_110 = ifelse(data5$mlst_100==data5$stringmlst_110, 1, 0),
                MLST_vs_stringMLST_kmer_120 = ifelse(data5$mlst_100==data5$stringmlst_120, 1, 0),
                MLST_vs_stringMLST_kmer_140 = ifelse(data5$mlst_100==data5$stringmlst_140, 1, 0))
dts <- df %>%
  group_by(species) %>%
  do(data.frame(kmer_20=c(mean(.$MLST_vs_stringMLST_kmer_20)*100),
                kmer_30=c(mean(.$MLST_vs_stringMLST_kmer_30)*100),
                kmer_35=c(mean(.$MLST_vs_stringMLST_kmer_35)*100),
                kmer_45=c(mean(.$MLST_vs_stringMLST_kmer_45)*100),
                kmer_55=c(mean(.$MLST_vs_stringMLST_kmer_55)*100),
                kmer_65=c(mean(.$MLST_vs_stringMLST_kmer_65)*100),
                kmer_70=c(mean(.$MLST_vs_stringMLST_kmer_70)*100),
                kmer_80=c(mean(.$MLST_vs_stringMLST_kmer_80)*100),
                kmer_90=c(mean(.$MLST_vs_stringMLST_kmer_90)*100),
                kmer_100=c(mean(.$MLST_vs_stringMLST_kmer_100)*100),
                kmer_110=c(mean(.$MLST_vs_stringMLST_kmer_110)*100),
                kmer_120=c(mean(.$MLST_vs_stringMLST_kmer_120)*100),
                kmer_140=c(mean(.$MLST_vs_stringMLST_kmer_140)*100)
  )
  )
DFtall <- dts %>% gather(key = kmer, value = Value, kmer_20:kmer_140)
DFtall$species[DFtall$species == "Enterococcus_faecium"] <- "E. faecium"
kmer_35_avg_data <- DFtall %>%
  filter(kmer == "kmer_35") %>%
  group_by(species) %>%
  summarise(avg = mean(Value))
level_order <- c("kmer_20", "kmer_30", "kmer_35", "kmer_45", "kmer_55", "kmer_65", "kmer_70", "kmer_80", "kmer_90", "kmer_100", "kmer_110", "kmer_120", "kmer_140")
# https://community.rstudio.com/t/help-with-making-plot-with-multiple-columns/50763
p <- ggplot(data = DFtall, aes(y = factor(kmer, level = level_order), x = Value, fill = kmer)) + geom_boxplot(position = "dodge") + xlim(0,100) +
  # p + scale_color_manual(breaks = c("1", "2", "3"), values=c("darkred", "darkblue", "darkgreen"))
  facet_wrap(~species, ncol = 4) + theme_bw() +
  theme(legend.position = "none") +
  geom_vline(data = kmer_35_avg_data, aes(xintercept = avg), linetype = "longdash", color = "black") +
  theme(axis.title.x = element_text(size = 60),
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 40),
        axis.title.y = element_text(size = 60),
        axis.text=element_text(size=40),
        axis.text.y = element_text(size = 30),
        strip.text.y = element_text(size = 60),
        strip.text.x = element_text(size = 60)) +
  theme(axis.line = element_line(colour = "black")) +
  labs(x = "% agreement", y = "MLST vs. stringMLST with different kmer values")
p



# S17 Fig, analyses from 14 pathogens, Enterococcus_faecium 100, percentage of miscalls
input <- read.csv("Enterococcus_faecium_all_92.txt", stringsAsFactors = FALSE)
# - and 0 ST are NA
data1 <- input %>%
  mutate_all(na_if, "-") %>%
  mutate_all(na_if, "0") 
# merge columns program_batch into one  
data2 <- transform(data1, program_kmer=paste(program, kmer, sep="_"))
# convert program_batch row to column
data3 <- data2 %>%
  spread(program_kmer, st)
data4 <- aggregate(. ~ sra_id + species, data=data3,
                   FUN=function(x) x[!is.na(x)][1], na.action=na.pass)
drops <- c("program","kmer")
data5 <- data4[ , !(names(data4) %in% drops)]
colnames(data5)[1] <- "sra_id"
data5$mlst_100 <- as.character(data5$mlst_100)
data5$stringmlst_20 <- as.character(data5$stringmlst_20)
data5$stringmlst_30 <- as.character(data5$stringmlst_30)
data5$stringmlst_35 <- as.character(data5$stringmlst_35)
data5$stringmlst_45 <- as.character(data5$stringmlst_45)
data5$stringmlst_55 <- as.character(data5$stringmlst_55)
data5$stringmlst_65 <- as.character(data5$stringmlst_65)
data5$stringmlst_70 <- as.character(data5$stringmlst_70)
data5$stringmlst_80 <- as.character(data5$stringmlst_80)
data5$stringmlst_90 <- as.character(data5$stringmlst_90)
data5$stringmlst_100 <- as.character(data5$stringmlst_100)
data5$stringmlst_110 <- as.character(data5$stringmlst_110)
data5$stringmlst_120 <- as.character(data5$stringmlst_120)
data5$stringmlst_140 <- as.character(data5$stringmlst_140)
data6 <- data5 %>% 
  group_by(species) %>% 
  summarise(mlst = sum(is.na(mlst_100)),
            kmer20 = sum(is.na(stringmlst_20)),
            kmer30 = sum(is.na(stringmlst_30)),
            kmer35 = sum(is.na(stringmlst_35)),
            kmer45 = sum(is.na(stringmlst_45)),
            kmer55 = sum(is.na(stringmlst_55)),
            kmer65 = sum(is.na(stringmlst_65)),
            kmer70 = sum(is.na(stringmlst_70)),
            kmer80 = sum(is.na(stringmlst_80)),
            kmer90 = sum(is.na(stringmlst_90)),
            kmer100 = sum(is.na(stringmlst_100)),
            kmer110 = sum(is.na(stringmlst_110)),
            kmer120 = sum(is.na(stringmlst_120)),
            kmer140 = sum(is.na(stringmlst_140)),
            total = n()
  )
data7 <- data6 %>% 
  group_by(species) %>% 
  summarise(mlst = (mlst/total)*100,
            kmer_20 = (kmer20/total)*100,
            kmer_30 = (kmer30/total)*100,
            kmer_35 = (kmer35/total)*100,
            kmer_45 = (kmer45/total)*100,
            kmer_55 = (kmer55/total)*100,
            kmer_65 = (kmer65/total)*100,
            kmer_70 = (kmer70/total)*100,
            kmer_80 = (kmer80/total)*100,
            kmer_90 = (kmer90/total)*100,
            kmer_100 = (kmer100/total)*100,
            kmer_110 = (kmer110/total)*100,
            kmer_120 = (kmer120/total)*100,
            kmer_140 = (kmer140/total)*100
  )
data_long <- gather(data7, combination, percentage, mlst:kmer_140, factor_key=TRUE)
data_long$species[data_long$species == "Enterococcus_faecium"] <- "E. faecium"
p <- ggplot(data_long , aes(x=combination, y=percentage)) + ylim(0, 100) + 
  geom_bar(position="dodge", stat='identity') + facet_wrap(~ species, ncol=1) +
  theme_bw() + theme(legend.position = "none") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 80),
        axis.title.y = element_text(size = 90),
        axis.text=element_text(size=80),
        axis.text.y = element_text(size = 80),
        strip.text.y = element_text(size = 90),
        strip.text.x = element_text(size = 90)) +
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +
  labs(y = "% of non-classified STs")
p

# S17 Fig, analyses from 14 pathogens, Enterococcus_faecium 100, percentage of agreement
# same 1, different 0
input <- read.csv("Enterococcus_faecium_all_92.txt", stringsAsFactors = FALSE)
# - and 0 ST are NA
data1 <- input %>%
  mutate_all(na_if, "-") %>%
  mutate_all(na_if, "0") 
# merge columns program_batch into one  
data2 <- transform(data1, program_kmer=paste(program, kmer, sep="_"))
# convert program_batch row to column
data3 <- data2 %>%
  spread(program_kmer, st)
data4 <- aggregate(. ~ sra_id + species, data=data3,
                   FUN=function(x) x[!is.na(x)][1], na.action=na.pass)
drops <- c("program","kmer")
data5 <- data4[ , !(names(data4) %in% drops)]
colnames(data5)[1] <- "sra_id"
data5$mlst_100 <- as.character(data5$mlst_100)
data5$stringmlst_20 <- as.character(data5$stringmlst_20)
data5$stringmlst_30 <- as.character(data5$stringmlst_30)
data5$stringmlst_35 <- as.character(data5$stringmlst_35)
data5$stringmlst_45 <- as.character(data5$stringmlst_45)
data5$stringmlst_55 <- as.character(data5$stringmlst_55)
data5$stringmlst_65 <- as.character(data5$stringmlst_65)
data5$stringmlst_70 <- as.character(data5$stringmlst_70)
data5$stringmlst_80 <- as.character(data5$stringmlst_80)
data5$stringmlst_90 <- as.character(data5$stringmlst_90)
data5$stringmlst_100 <- as.character(data5$stringmlst_100)
data5$stringmlst_110 <- as.character(data5$stringmlst_110)
data5$stringmlst_120 <- as.character(data5$stringmlst_120)
data5$stringmlst_140 <- as.character(data5$stringmlst_140)
data5[is.na(data5)] <- -3
df <- transform(data5, MLST_vs_stringMLST_kmer_20 = ifelse(data5$mlst_100==data5$stringmlst_20, 1, 0),
                MLST_vs_stringMLST_kmer_30 = ifelse(data5$mlst_100==data5$stringmlst_30, 1, 0),
                MLST_vs_stringMLST_kmer_35 = ifelse(data5$mlst_100==data5$stringmlst_35, 1, 0),
                MLST_vs_stringMLST_kmer_45 = ifelse(data5$mlst_100==data5$stringmlst_45, 1, 0),
                MLST_vs_stringMLST_kmer_55 = ifelse(data5$mlst_100==data5$stringmlst_55, 1, 0),
                MLST_vs_stringMLST_kmer_65 = ifelse(data5$mlst_100==data5$stringmlst_65, 1, 0),
                MLST_vs_stringMLST_kmer_70 = ifelse(data5$mlst_100==data5$stringmlst_70, 1, 0),
                MLST_vs_stringMLST_kmer_80 = ifelse(data5$mlst_100==data5$stringmlst_80, 1, 0),
                MLST_vs_stringMLST_kmer_90 = ifelse(data5$mlst_100==data5$stringmlst_90, 1, 0),
                MLST_vs_stringMLST_kmer_100 = ifelse(data5$mlst_100==data5$stringmlst_100, 1, 0),
                MLST_vs_stringMLST_kmer_110 = ifelse(data5$mlst_100==data5$stringmlst_110, 1, 0),
                MLST_vs_stringMLST_kmer_120 = ifelse(data5$mlst_100==data5$stringmlst_120, 1, 0),
                MLST_vs_stringMLST_kmer_140 = ifelse(data5$mlst_100==data5$stringmlst_140, 1, 0))
dts <- df %>%
  group_by(species) %>%
  do(data.frame(kmer_20=c(mean(.$MLST_vs_stringMLST_kmer_20)*100),
                kmer_30=c(mean(.$MLST_vs_stringMLST_kmer_30)*100),
                kmer_35=c(mean(.$MLST_vs_stringMLST_kmer_35)*100),
                kmer_45=c(mean(.$MLST_vs_stringMLST_kmer_45)*100),
                kmer_55=c(mean(.$MLST_vs_stringMLST_kmer_55)*100),
                kmer_65=c(mean(.$MLST_vs_stringMLST_kmer_65)*100),
                kmer_70=c(mean(.$MLST_vs_stringMLST_kmer_70)*100),
                kmer_80=c(mean(.$MLST_vs_stringMLST_kmer_80)*100),
                kmer_90=c(mean(.$MLST_vs_stringMLST_kmer_90)*100),
                kmer_100=c(mean(.$MLST_vs_stringMLST_kmer_100)*100),
                kmer_110=c(mean(.$MLST_vs_stringMLST_kmer_110)*100),
                kmer_120=c(mean(.$MLST_vs_stringMLST_kmer_120)*100),
                kmer_140=c(mean(.$MLST_vs_stringMLST_kmer_140)*100)
  ))
DFtall <- dts %>% gather(key = kmer, value = Value, kmer_20:kmer_140)
DFtall$species[DFtall$species == "Enterococcus_faecium"] <- "E. faecium"
kmer_35_avg_data <- DFtall %>%
  filter(kmer == "kmer_35") %>%
  group_by(species) %>%
  summarise(avg = mean(Value))
level_order <- c("kmer_20", "kmer_30", "kmer_35", "kmer_45", "kmer_55", "kmer_65", "kmer_70", "kmer_80", "kmer_90", "kmer_100", "kmer_110", "kmer_120", "kmer_140")
# https://community.rstudio.com/t/help-with-making-plot-with-multiple-columns/50763
p <- ggplot(data = DFtall, aes(y = factor(kmer, level = level_order), x = Value, fill = kmer)) + geom_boxplot(position = "dodge") + xlim(0,100) +
  # p + scale_color_manual(breaks = c("1", "2", "3"), values=c("darkred", "darkblue", "darkgreen"))
  facet_wrap(~species, ncol = 4) + theme_bw() +
  theme(legend.position = "none") +
  geom_vline(data = kmer_35_avg_data, aes(xintercept = avg), linetype = "longdash", color = "black") +
  theme(axis.title.x = element_text(size = 60),
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 40),
        axis.title.y = element_text(size = 60),
        axis.text=element_text(size=40),
        axis.text.y = element_text(size = 30),
        strip.text.y = element_text(size = 60),
        strip.text.x = element_text(size = 60)) +
  theme(axis.line = element_line(colour = "black")) +
  labs(x = "% agreement", y = "MLST vs. stringMLST with different kmer values")
p



# S17 Fig, analyses from 14 pathogens, Helicobacter_pylori, percentage of miscalls
input <- read.csv("Helicobacter_pylori_all_874.txt", stringsAsFactors = FALSE)
# - and 0 ST are NA
data1 <- input %>%
  mutate_all(na_if, "-") %>%
  mutate_all(na_if, "0") 
# merge columns program_batch into one  
data2 <- transform(data1, program_kmer=paste(program, kmer, sep="_"))
# convert program_batch row to column
data3 <- data2 %>%
  spread(program_kmer, st)
data4 <- aggregate(. ~ sra_id + species, data=data3,
                   FUN=function(x) x[!is.na(x)][1], na.action=na.pass)
drops <- c("program","kmer")
data5 <- data4[ , !(names(data4) %in% drops)]
colnames(data5)[1] <- "sra_id"
data5$mlst_100 <- as.character(data5$mlst_100)
data5$stringmlst_20 <- as.character(data5$stringmlst_20)
data5$stringmlst_30 <- as.character(data5$stringmlst_30)
data5$stringmlst_35 <- as.character(data5$stringmlst_35)
data5$stringmlst_45 <- as.character(data5$stringmlst_45)
data5$stringmlst_55 <- as.character(data5$stringmlst_55)
data5$stringmlst_65 <- as.character(data5$stringmlst_65)
data5$stringmlst_70 <- as.character(data5$stringmlst_70)
data5$stringmlst_80 <- as.character(data5$stringmlst_80)
data5$stringmlst_90 <- as.character(data5$stringmlst_90)
data5$stringmlst_100 <- as.character(data5$stringmlst_100)
data5$stringmlst_110 <- as.character(data5$stringmlst_110)
data5$stringmlst_120 <- as.character(data5$stringmlst_120)
data5$stringmlst_140 <- as.character(data5$stringmlst_140)
data6 <- data5 %>% 
  group_by(species) %>% 
  summarise(mlst100 = sum(is.na(mlst_100)),
            kmer20 = sum(is.na(stringmlst_20)),
            kmer30 = sum(is.na(stringmlst_30)),
            kmer35 = sum(is.na(stringmlst_35)),
            kmer45 = sum(is.na(stringmlst_45)),
            kmer55 = sum(is.na(stringmlst_55)),
            kmer65 = sum(is.na(stringmlst_65)),
            kmer70 = sum(is.na(stringmlst_70)),
            kmer80 = sum(is.na(stringmlst_80)),
            kmer90 = sum(is.na(stringmlst_90)),
            kmer100 = sum(is.na(stringmlst_100)),
            kmer110 = sum(is.na(stringmlst_110)),
            kmer120 = sum(is.na(stringmlst_120)),
            kmer140 = sum(is.na(stringmlst_140)),
            total = n()
  )
data7 <- data6 %>% 
  group_by(species) %>% 
  summarise(mlst = (mlst100/total)*100,
            kmer_20 = (kmer20/total)*100,
            kmer_30 = (kmer30/total)*100,
            kmer_35 = (kmer35/total)*100,
            kmer_45 = (kmer45/total)*100,
            kmer_55 = (kmer55/total)*100,
            kmer_65 = (kmer65/total)*100,
            kmer_70 = (kmer70/total)*100,
            kmer_80 = (kmer80/total)*100,
            kmer_90 = (kmer90/total)*100,
            kmer_100 = (kmer100/total)*100,
            kmer_110 = (kmer110/total)*100,
            kmer_120 = (kmer120/total)*100,
            kmer_140 = (kmer140/total)*100
  )
data_long <- gather(data7, combination, percentage, mlst:kmer_140, factor_key=TRUE)
data_long$species[data_long$species == "Helicobacter_pylori"] <- "H. pylori"
p <- ggplot(data_long , aes(x=combination, y=percentage)) + ylim(0, 100) + 
  geom_bar(position="dodge", stat='identity') + facet_wrap(~ species, ncol=1) +
  theme_bw() + theme(legend.position = "none") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 80),
        axis.title.y = element_text(size = 90),
        axis.text=element_text(size=80),
        axis.text.y = element_text(size = 80),
        strip.text.y = element_text(size = 90),
        strip.text.x = element_text(size = 90)) +
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +
  labs(y = "% of non-classified STs")
p

# S17 Fig, analyses from 14 pathogens, Helicobacter_pylori, percentage of agreement
# same 1, different 0
data5[is.na(data5)] <- -3
df <- transform(data5, MLST_vs_stringMLST_kmer_20 = ifelse(data5$mlst_100==data5$stringmlst_20, 1, 0),
                MLST_vs_stringMLST_kmer_30 = ifelse(data5$mlst_100==data5$stringmlst_30, 1, 0),
                MLST_vs_stringMLST_kmer_35 = ifelse(data5$mlst_100==data5$stringmlst_35, 1, 0),
                MLST_vs_stringMLST_kmer_45 = ifelse(data5$mlst_100==data5$stringmlst_45, 1, 0),
                MLST_vs_stringMLST_kmer_55 = ifelse(data5$mlst_100==data5$stringmlst_55, 1, 0),
                MLST_vs_stringMLST_kmer_65 = ifelse(data5$mlst_100==data5$stringmlst_65, 1, 0),
                MLST_vs_stringMLST_kmer_70 = ifelse(data5$mlst_100==data5$stringmlst_70, 1, 0),
                MLST_vs_stringMLST_kmer_80 = ifelse(data5$mlst_100==data5$stringmlst_80, 1, 0),
                MLST_vs_stringMLST_kmer_90 = ifelse(data5$mlst_100==data5$stringmlst_90, 1, 0),
                MLST_vs_stringMLST_kmer_100 = ifelse(data5$mlst_100==data5$stringmlst_100, 1, 0),
                MLST_vs_stringMLST_kmer_110 = ifelse(data5$mlst_100==data5$stringmlst_110, 1, 0),
                MLST_vs_stringMLST_kmer_120 = ifelse(data5$mlst_100==data5$stringmlst_120, 1, 0),
                MLST_vs_stringMLST_kmer_140 = ifelse(data5$mlst_100==data5$stringmlst_140, 1, 0))
dts <- df %>%
  group_by(species) %>%
  do(data.frame(kmer_20=c(mean(.$MLST_vs_stringMLST_kmer_20)*100),
                kmer_30=c(mean(.$MLST_vs_stringMLST_kmer_30)*100),
                kmer_35=c(mean(.$MLST_vs_stringMLST_kmer_35)*100),
                kmer_45=c(mean(.$MLST_vs_stringMLST_kmer_45)*100),
                kmer_55=c(mean(.$MLST_vs_stringMLST_kmer_55)*100),
                kmer_65=c(mean(.$MLST_vs_stringMLST_kmer_65)*100),
                kmer_70=c(mean(.$MLST_vs_stringMLST_kmer_70)*100),
                kmer_80=c(mean(.$MLST_vs_stringMLST_kmer_80)*100),
                kmer_90=c(mean(.$MLST_vs_stringMLST_kmer_90)*100),
                kmer_100=c(mean(.$MLST_vs_stringMLST_kmer_100)*100),
                kmer_110=c(mean(.$MLST_vs_stringMLST_kmer_110)*100),
                kmer_120=c(mean(.$MLST_vs_stringMLST_kmer_120)*100),
                kmer_140=c(mean(.$MLST_vs_stringMLST_kmer_140)*100)
  )
  )
DFtall <- dts %>% gather(key = kmer, value = Value, kmer_20:kmer_140)
DFtall$species[DFtall$species == "Helicobacter_pylori"] <- "H. pylori"
kmer_35_avg_data <- DFtall %>%
  filter(kmer == "kmer_35") %>%
  group_by(species) %>%
  summarise(avg = mean(Value))
level_order <- c("kmer_20", "kmer_30", "kmer_35", "kmer_45", "kmer_55", "kmer_65", "kmer_70", "kmer_80", "kmer_90", "kmer_100", "kmer_110", "kmer_120", "kmer_140")
# https://community.rstudio.com/t/help-with-making-plot-with-multiple-columns/50763
p <- ggplot(data = DFtall, aes(y = factor(kmer, level = level_order), x = Value, fill = kmer)) + geom_boxplot(position = "dodge") + xlim(0,100) +
  # p + scale_color_manual(breaks = c("1", "2", "3"), values=c("darkred", "darkblue", "darkgreen"))
  facet_wrap(~species, ncol = 4) + theme_bw() +
  theme(legend.position = "none") +
  geom_vline(data = kmer_35_avg_data, aes(xintercept = avg), linetype = "longdash", color = "black") +
  theme(axis.title.x = element_text(size = 60),
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 40),
        axis.title.y = element_text(size = 60),
        axis.text=element_text(size=40),
        axis.text.y = element_text(size = 30),
        strip.text.y = element_text(size = 60),
        strip.text.x = element_text(size = 60)) +
  theme(axis.line = element_line(colour = "black")) +
  labs(x = "% agreement", y = "MLST vs. stringMLST with different kmer values")
p



# S17 Fig, analyses from 14 pathogens, Helicobacter_pylori 100, percentage of miscalls
input <- read.csv("Helicobacter_pylori_all_81.txt", stringsAsFactors = FALSE)
# - and 0 ST are NA
data1 <- input %>%
  mutate_all(na_if, "-") %>%
  mutate_all(na_if, "0") 
# merge columns program_batch into one  
data2 <- transform(data1, program_kmer=paste(program, kmer, sep="_"))
# convert program_batch row to column
data3 <- data2 %>%
  spread(program_kmer, st)
data4 <- aggregate(. ~ sra_id + species, data=data3,
                   FUN=function(x) x[!is.na(x)][1], na.action=na.pass)
drops <- c("program","kmer")
data5 <- data4[ , !(names(data4) %in% drops)]
colnames(data5)[1] <- "sra_id"
data5$mlst_100 <- as.character(data5$mlst_100)
data5$stringmlst_20 <- as.character(data5$stringmlst_20)
data5$stringmlst_30 <- as.character(data5$stringmlst_30)
data5$stringmlst_35 <- as.character(data5$stringmlst_35)
data5$stringmlst_45 <- as.character(data5$stringmlst_45)
data5$stringmlst_55 <- as.character(data5$stringmlst_55)
data5$stringmlst_65 <- as.character(data5$stringmlst_65)
data5$stringmlst_70 <- as.character(data5$stringmlst_70)
data5$stringmlst_80 <- as.character(data5$stringmlst_80)
data5$stringmlst_90 <- as.character(data5$stringmlst_90)
data5$stringmlst_100 <- as.character(data5$stringmlst_100)
data5$stringmlst_110 <- as.character(data5$stringmlst_110)
data5$stringmlst_120 <- as.character(data5$stringmlst_120)
data5$stringmlst_140 <- as.character(data5$stringmlst_140)
data6 <- data5 %>% 
  group_by(species) %>% 
  summarise(mlst = sum(is.na(mlst_100)),
            kmer20 = sum(is.na(stringmlst_20)),
            kmer30 = sum(is.na(stringmlst_30)),
            kmer35 = sum(is.na(stringmlst_35)),
            kmer45 = sum(is.na(stringmlst_45)),
            kmer55 = sum(is.na(stringmlst_55)),
            kmer65 = sum(is.na(stringmlst_65)),
            kmer70 = sum(is.na(stringmlst_70)),
            kmer80 = sum(is.na(stringmlst_80)),
            kmer90 = sum(is.na(stringmlst_90)),
            kmer100 = sum(is.na(stringmlst_100)),
            kmer110 = sum(is.na(stringmlst_110)),
            kmer120 = sum(is.na(stringmlst_120)),
            kmer140 = sum(is.na(stringmlst_140)),
            total = n()
  )
data7 <- data6 %>% 
  group_by(species) %>% 
  summarise(mlst = (mlst/total)*100,
            kmer_20 = (kmer20/total)*100,
            kmer_30 = (kmer30/total)*100,
            kmer_35 = (kmer35/total)*100,
            kmer_45 = (kmer45/total)*100,
            kmer_55 = (kmer55/total)*100,
            kmer_65 = (kmer65/total)*100,
            kmer_70 = (kmer70/total)*100,
            kmer_80 = (kmer80/total)*100,
            kmer_90 = (kmer90/total)*100,
            kmer_100 = (kmer100/total)*100,
            kmer_110 = (kmer110/total)*100,
            kmer_120 = (kmer120/total)*100,
            kmer_140 = (kmer140/total)*100
  )
data_long <- gather(data7, combination, percentage, mlst:kmer_140, factor_key=TRUE)
data_long$species[data_long$species == "Helicobacter_pylori"] <- "H. pylori"
p <- ggplot(data_long , aes(x=combination, y=percentage)) + ylim(0, 100) + 
  geom_bar(position="dodge", stat='identity') + facet_wrap(~ species, ncol=1) +
  theme_bw() + theme(legend.position = "none") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 80),
        axis.title.y = element_text(size = 90),
        axis.text=element_text(size=80),
        axis.text.y = element_text(size = 80),
        strip.text.y = element_text(size = 90),
        strip.text.x = element_text(size = 90)) +
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(), axis.line = element_line(colour = "black")) +
  labs(y = "% of non-classified STs")
p

# S17 Fig, analyses from 14 pathogens, Helicobacter_pylori 100, percentage of agreement
# same 1, different 0
input <- read.csv("Helicobacter_pylori_all_81.txt", stringsAsFactors = FALSE)
# - and 0 ST are NA
data1 <- input %>%
  mutate_all(na_if, "-") %>%
  mutate_all(na_if, "0") 
# merge columns program_batch into one  
data2 <- transform(data1, program_kmer=paste(program, kmer, sep="_"))
# convert program_batch row to column
data3 <- data2 %>%
  spread(program_kmer, st)
data4 <- aggregate(. ~ sra_id + species, data=data3,
                   FUN=function(x) x[!is.na(x)][1], na.action=na.pass)
drops <- c("program","kmer")
data5 <- data4[ , !(names(data4) %in% drops)]
colnames(data5)[1] <- "sra_id"
data5$mlst_100 <- as.character(data5$mlst_100)
data5$stringmlst_20 <- as.character(data5$stringmlst_20)
data5$stringmlst_30 <- as.character(data5$stringmlst_30)
data5$stringmlst_35 <- as.character(data5$stringmlst_35)
data5$stringmlst_45 <- as.character(data5$stringmlst_45)
data5$stringmlst_55 <- as.character(data5$stringmlst_55)
data5$stringmlst_65 <- as.character(data5$stringmlst_65)
data5$stringmlst_70 <- as.character(data5$stringmlst_70)
data5$stringmlst_80 <- as.character(data5$stringmlst_80)
data5$stringmlst_90 <- as.character(data5$stringmlst_90)
data5$stringmlst_100 <- as.character(data5$stringmlst_100)
data5$stringmlst_110 <- as.character(data5$stringmlst_110)
data5$stringmlst_120 <- as.character(data5$stringmlst_120)
data5$stringmlst_140 <- as.character(data5$stringmlst_140)
data5[is.na(data5)] <- -3
df <- transform(data5, MLST_vs_stringMLST_kmer_20 = ifelse(data5$mlst_100==data5$stringmlst_20, 1, 0),
                MLST_vs_stringMLST_kmer_30 = ifelse(data5$mlst_100==data5$stringmlst_30, 1, 0),
                MLST_vs_stringMLST_kmer_35 = ifelse(data5$mlst_100==data5$stringmlst_35, 1, 0),
                MLST_vs_stringMLST_kmer_45 = ifelse(data5$mlst_100==data5$stringmlst_45, 1, 0),
                MLST_vs_stringMLST_kmer_55 = ifelse(data5$mlst_100==data5$stringmlst_55, 1, 0),
                MLST_vs_stringMLST_kmer_65 = ifelse(data5$mlst_100==data5$stringmlst_65, 1, 0),
                MLST_vs_stringMLST_kmer_70 = ifelse(data5$mlst_100==data5$stringmlst_70, 1, 0),
                MLST_vs_stringMLST_kmer_80 = ifelse(data5$mlst_100==data5$stringmlst_80, 1, 0),
                MLST_vs_stringMLST_kmer_90 = ifelse(data5$mlst_100==data5$stringmlst_90, 1, 0),
                MLST_vs_stringMLST_kmer_100 = ifelse(data5$mlst_100==data5$stringmlst_100, 1, 0),
                MLST_vs_stringMLST_kmer_110 = ifelse(data5$mlst_100==data5$stringmlst_110, 1, 0),
                MLST_vs_stringMLST_kmer_120 = ifelse(data5$mlst_100==data5$stringmlst_120, 1, 0),
                MLST_vs_stringMLST_kmer_140 = ifelse(data5$mlst_100==data5$stringmlst_140, 1, 0))
dts <- df %>%
  group_by(species) %>%
  do(data.frame(kmer_20=c(mean(.$MLST_vs_stringMLST_kmer_20)*100),
                kmer_30=c(mean(.$MLST_vs_stringMLST_kmer_30)*100),
                kmer_35=c(mean(.$MLST_vs_stringMLST_kmer_35)*100),
                kmer_45=c(mean(.$MLST_vs_stringMLST_kmer_45)*100),
                kmer_55=c(mean(.$MLST_vs_stringMLST_kmer_55)*100),
                kmer_65=c(mean(.$MLST_vs_stringMLST_kmer_65)*100),
                kmer_70=c(mean(.$MLST_vs_stringMLST_kmer_70)*100),
                kmer_80=c(mean(.$MLST_vs_stringMLST_kmer_80)*100),
                kmer_90=c(mean(.$MLST_vs_stringMLST_kmer_90)*100),
                kmer_100=c(mean(.$MLST_vs_stringMLST_kmer_100)*100),
                kmer_110=c(mean(.$MLST_vs_stringMLST_kmer_110)*100),
                kmer_120=c(mean(.$MLST_vs_stringMLST_kmer_120)*100),
                kmer_140=c(mean(.$MLST_vs_stringMLST_kmer_140)*100)
  ))
DFtall <- dts %>% gather(key = kmer, value = Value, kmer_20:kmer_140)
DFtall$species[DFtall$species == "Helicobacter_pylori"] <- "H. pylori"
kmer_35_avg_data <- DFtall %>%
  filter(kmer == "kmer_35") %>%
  group_by(species) %>%
  summarise(avg = mean(Value))
level_order <- c("kmer_20", "kmer_30", "kmer_35", "kmer_45", "kmer_55", "kmer_65", "kmer_70", "kmer_80", "kmer_90", "kmer_100", "kmer_110", "kmer_120", "kmer_140")
# https://community.rstudio.com/t/help-with-making-plot-with-multiple-columns/50763
p <- ggplot(data = DFtall, aes(y = factor(kmer, level = level_order), x = Value, fill = kmer)) + geom_boxplot(position = "dodge") + xlim(0,100) +
  # p + scale_color_manual(breaks = c("1", "2", "3"), values=c("darkred", "darkblue", "darkgreen"))
  facet_wrap(~species, ncol = 4) + theme_bw() +
  theme(legend.position = "none") +
  geom_vline(data = kmer_35_avg_data, aes(xintercept = avg), linetype = "longdash", color = "black") +
  theme(axis.title.x = element_text(size = 60),
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 40),
        axis.title.y = element_text(size = 60),
        axis.text=element_text(size=40),
        axis.text.y = element_text(size = 30),
        strip.text.y = element_text(size = 60),
        strip.text.x = element_text(size = 60)) +
  theme(axis.line = element_line(colour = "black")) +
  labs(x = "% agreement", y = "MLST vs. stringMLST with different kmer values")
p
